<!DOCTYPE html>
<html lang="en" itemscope itemtype="http://schema.org/WebPage">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>GDC Notebook - xiaoni&#39;s blog</title>
  

<meta name="renderer" content="webkit" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>

<meta name="MobileOptimized" content="width"/>
<meta name="HandheldFriendly" content="true"/>


<meta name="applicable-device" content="pc,mobile">

<meta name="theme-color" content="#f8f5ec" />
<meta name="msapplication-navbutton-color" content="#f8f5ec">
<meta name="apple-mobile-web-app-capable" content="yes">
<meta name="apple-mobile-web-app-status-bar-style" content="#f8f5ec">

<meta name="mobile-web-app-capable" content="yes">

<meta name="author" content="Xiaoni" />
  <meta name="description" content="The Genomic Data Commons (GDC), is a research program of the National Cancer Institute (NCI).
The mission of the GDC is to provide the cancer research community with a unified data repository that enables data sharing across cancer genomic studies in support of precision medicine.
" />

  <meta name="keywords" content="xiaonili, xiaonilee, xiaoni" />






<meta name="generator" content="Hugo 0.110.0" />


<link rel="canonical" href="/post/gdc/" />





<link rel="icon" href="/favicon.ico" />











<link rel="stylesheet" href="/sass/jane.min.fa4b2b9f31b5c6d0b683db81157a9226e17b06e61911791ab547242a4a0556f2.css" integrity="sha256-&#43;ksrnzG1xtC2g9uBFXqSJuF7BuYZEXkatUckKkoFVvI=" media="screen" crossorigin="anonymous">



<link href="https://cdn.bootcdn.net/ajax/libs/font-awesome/5.15.1/css/all.css" rel="stylesheet">
<script src="https://cdn.bootcdn.net/ajax/libs/font-awesome/5.15.1/js/all.js"></script>




<meta property="og:title" content="GDC Notebook" />
<meta property="og:description" content="The Genomic Data Commons (GDC), is a research program of the National Cancer Institute (NCI).
The mission of the GDC is to provide the cancer research community with a unified data repository that enables data sharing across cancer genomic studies in support of precision medicine." />
<meta property="og:type" content="article" />
<meta property="og:url" content="/post/gdc/" /><meta property="article:section" content="post" />
<meta property="article:published_time" content="2020-09-25T00:00:00+00:00" />
<meta property="article:modified_time" content="2020-09-28T00:00:00+00:00" />
<meta itemprop="name" content="GDC Notebook">
<meta itemprop="description" content="The Genomic Data Commons (GDC), is a research program of the National Cancer Institute (NCI).
The mission of the GDC is to provide the cancer research community with a unified data repository that enables data sharing across cancer genomic studies in support of precision medicine."><meta itemprop="datePublished" content="2020-09-25T00:00:00+00:00" />
<meta itemprop="dateModified" content="2020-09-28T00:00:00+00:00" />
<meta itemprop="wordCount" content="1037">
<meta itemprop="keywords" content="TCGA,Bioinformatics,GDC,R," /><meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="GDC Notebook"/>
<meta name="twitter:description" content="The Genomic Data Commons (GDC), is a research program of the National Cancer Institute (NCI).
The mission of the GDC is to provide the cancer research community with a unified data repository that enables data sharing across cancer genomic studies in support of precision medicine."/>

<!--[if lte IE 9]>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/classlist/1.1.20170427/classList.min.js"></script>
<![endif]-->

<!--[if lt IE 9]>
  <script src="https://cdn.jsdelivr.net/npm/html5shiv@3.7.3/dist/html5shiv.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/respond.js@1.4.2/dest/respond.min.js"></script>
<![endif]-->


<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'UA-177860608-1', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>



</head>
<body>
  <div id="mobile-navbar" class="mobile-navbar">
  <div class="mobile-header-logo">
    <a href="/" class="logo">Xiaoni's Blog</a>
  </div>
  <div class="mobile-navbar-icon">
    <span></span>
    <span></span>
    <span></span>
  </div>
</div>
<nav id="mobile-menu" class="mobile-menu slideout-menu">
  <ul class="mobile-menu-list">
    <li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/">Home</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/post/">Archives</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/categories/">Categories</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/tags/">Tags</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/about/">About Me</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          <div class="mobile-menu-parent">
            <span class="mobile-submenu-open"></span>
            <a href="/">
              docs
            </a>
          </div>
          <ul class="mobile-submenu-list">
            
              <li>
                <a href="/post/tensorflow2/"> Machine Learning: TensorFlow2 Notebook</a>
              </li>
            
              <li>
                <a href="/post/bioscirep/">Bioinformatics analysis of prognostic value of PITX1 gene in breast cancer</a>
              </li>
            
              <li>
                <a href="/post/rinactionmap/">Book: R in Action</a>
              </li>
            
              <li>
                <a href="/post/tumorimmune/">Comprehensive Bioinformatics Analysis of Tumor-Infiltrating Immune Cells</a>
              </li>
            
              <li>
                <a href="/post/aavcapsid/">Deep diversification of an AAV capsid protein by machine learning</a>
              </li>
            
              <li>
                <a href="/post/firebrowse/">FireBrowse Notebook</a>
              </li>
            
              <li>
                <a href="/post/gdc/">GDC Notebook</a>
              </li>
            
              <li>
                <a href="/post/git/">Git - Common commands</a>
              </li>
            
              <li>
                <a href="/post/gscalite/">GSCALite: Gene Set Cancer Analysis</a>
              </li>
            
              <li>
                <a href="/post/frontoncol/">Identification of Therapeutic Targets and Prognostic Biomarkers Among CXC Chemokines in the Renal Cell Carcinoma Microenvironment</a>
              </li>
            
              <li>
                <a href="/post/condacolab/">Installing Conda in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/oncolnc/">OncoLnc: A straightforward Tool</a>
              </li>
            
              <li>
                <a href="/post/webcrawler/">Python Spider</a>
              </li>
            
              <li>
                <a href="/post/rnasequenceanalysis/">RNA-seq Bioinformatics - RNA sequence analysis</a>
              </li>
            
              <li>
                <a href="/post/scrnaknn/">scRNA-sequencing analysis - PCA and KNN</a>
              </li>
            
              <li>
                <a href="/post/survival/">Survival Analysis with data from GSE database</a>
              </li>
            
              <li>
                <a href="/post/rincolab/">The way to use R in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/tissgdb/">TissGDB: tissue-specific gene database in cancer</a>
              </li>
            
              <li>
                <a href="/post/ucscmap/">UCSC Topics</a>
              </li>
            
              <li>
                <a href="/post/webfrontend/">web front end</a>
              </li>
            
          </ul>
        
      </li>
    

    
  </ul>
</nav>


  
    






  <link rel="stylesheet" href="/lib/photoswipe/photoswipe.min.css" />
  <link rel="stylesheet" href="/lib/photoswipe/default-skin/default-skin.min.css" />




<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

<div class="pswp__bg"></div>

<div class="pswp__scroll-wrap">
    
    <div class="pswp__container">
      <div class="pswp__item"></div>
      <div class="pswp__item"></div>
      <div class="pswp__item"></div>
    </div>
    
    <div class="pswp__ui pswp__ui--hidden">
    <div class="pswp__top-bar">
      
      <div class="pswp__counter"></div>
      <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>
      <button class="pswp__button pswp__button--share" title="Share"></button>
      <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>
      <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>
      
      
      <div class="pswp__preloader">
        <div class="pswp__preloader__icn">
          <div class="pswp__preloader__cut">
            <div class="pswp__preloader__donut"></div>
          </div>
        </div>
      </div>
    </div>
    <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
      <div class="pswp__share-tooltip"></div>
    </div>
    <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
    </button>
    <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
    </button>
    <div class="pswp__caption">
      <div class="pswp__caption__center"></div>
    </div>
    </div>
    </div>
</div>

  

  

  

  <header id="header" class="header container">
    <div class="logo-wrapper">
  <a href="/" class="logo">
    
      Xiaoni's Blog
    
  </a>
</div>

<nav class="site-navbar">
  <ul id="menu" class="menu">
    
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/">Home</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/post/">Archives</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/categories/">Categories</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/tags/">Tags</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/about/">About Me</a>
          

        

      </li>
    
        <li class="menu-item active">
        
          
          <a class="menu-item-link menu-parent" href="/">docs</a>
          <ul class="submenu" >
            
              <li>
                <a href="/post/tensorflow2/"> Machine Learning: TensorFlow2 Notebook</a>
              </li>
            
              <li>
                <a href="/post/bioscirep/">Bioinformatics analysis of prognostic value of PITX1 gene in breast cancer</a>
              </li>
            
              <li>
                <a href="/post/rinactionmap/">Book: R in Action</a>
              </li>
            
              <li>
                <a href="/post/tumorimmune/">Comprehensive Bioinformatics Analysis of Tumor-Infiltrating Immune Cells</a>
              </li>
            
              <li>
                <a href="/post/aavcapsid/">Deep diversification of an AAV capsid protein by machine learning</a>
              </li>
            
              <li>
                <a href="/post/firebrowse/">FireBrowse Notebook</a>
              </li>
            
              <li>
                <a href="/post/gdc/">GDC Notebook</a>
              </li>
            
              <li>
                <a href="/post/git/">Git - Common commands</a>
              </li>
            
              <li>
                <a href="/post/gscalite/">GSCALite: Gene Set Cancer Analysis</a>
              </li>
            
              <li>
                <a href="/post/frontoncol/">Identification of Therapeutic Targets and Prognostic Biomarkers Among CXC Chemokines in the Renal Cell Carcinoma Microenvironment</a>
              </li>
            
              <li>
                <a href="/post/condacolab/">Installing Conda in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/oncolnc/">OncoLnc: A straightforward Tool</a>
              </li>
            
              <li>
                <a href="/post/webcrawler/">Python Spider</a>
              </li>
            
              <li>
                <a href="/post/rnasequenceanalysis/">RNA-seq Bioinformatics - RNA sequence analysis</a>
              </li>
            
              <li>
                <a href="/post/scrnaknn/">scRNA-sequencing analysis - PCA and KNN</a>
              </li>
            
              <li>
                <a href="/post/survival/">Survival Analysis with data from GSE database</a>
              </li>
            
              <li>
                <a href="/post/rincolab/">The way to use R in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/tissgdb/">TissGDB: tissue-specific gene database in cancer</a>
              </li>
            
              <li>
                <a href="/post/ucscmap/">UCSC Topics</a>
              </li>
            
              <li>
                <a href="/post/webfrontend/">web front end</a>
              </li>
            
          </ul>

        

      </li>
    

    
    

    
  </ul>
</nav>

  </header>

  <div id="mobile-panel">
    <main id="main" class="main bg-llight">
      <div class="content-wrapper">
        <div id="content" class="content container">
          <article class="post bg-white">
    
    <header class="post-header">
      <h1 class="post-title">GDC Notebook</h1>
      
      <div class="post-meta">
        <time datetime="2020-09-25" class="post-time">
          2020-09-25
        </time>
        <div class="post-category">
            <a href="/categories/cancer-research/"> Cancer Research </a>
            <a href="/categories/database/"> Database </a>
            
          </div>
        

        
        

        
        
      </div>
    </header>

    
    
<div class="post-toc" id="post-toc">
  <h2 class="post-toc-title">Table of Contents</h2>
  <div class="post-toc-content">
    <nav id="TableOfContents">
  <ul>
    <li>
      <ul>
        <li><a href="#in-brief">In Brief</a></li>
        <li><a href="#example-1-standarized-download">Example 1 Standarized Download</a></li>
        <li><a href="#data-wrangling">Data Wrangling</a></li>
        <li><a href="#r-scripts-for-single-sample">R Scripts For Single Sample</a></li>
        <li><a href="#run-r-scripts-for-all-files">Run R Scripts For All files</a></li>
        <li><a href="#example-2-fast-download">Example 2 Fast Download</a></li>
        <li><a href="#run-rscripts">Run R.Scripts</a></li>
        <li><a href="#in-summary">In summary</a></li>
      </ul>
    </li>
  </ul>
</nav>
  </div>
</div>

    
    <div class="post-content">
      <p>The Genomic Data Commons (<a href="https://portal.gdc.cancer.gov/">GDC</a>), is a research program of the National Cancer Institute (NCI).</p>
<p>The <strong>mission</strong> of the GDC is to provide the cancer research community with a <strong>unified</strong> data repository that enables data sharing across cancer genomic studies in support of <strong>precision medicine</strong>.</p>
<h3 id="in-brief">In Brief</h3>
<ul>
<li>
<p>Download Data with different methods</p>
</li>
<li>
<p>Data Wrangling</p>
</li>
</ul>
<h3 id="example-1-standarized-download">Example 1 Standarized Download</h3>
<ul>
<li>
<p>Click on <code>Repository</code> and choose <code>Cases</code> to setup interested data: TCGA-LUNG.</p>
<p><img src="caseset.png" alt="caseset"></p>
</li>
<li>
<p>Choose <code>Files</code> and select interested demand, and then click on <code>Manifest</code> to download data1.</p>
<p><img src="filesset1.png" alt="filesset1"></p>
</li>
<li>
<p>Continue, select <code>Isoform Expression Quantification</code> to change <strong>Data Type</strong> and click on <code>Manifest</code> to download data2.</p>
<p><img src="filesset2.png" alt="filesset2"></p>
</li>
<li>
<p>Keep <code>Cases</code> selection, remove all <strong>previously</strong> selection on the page of <code>Files</code>.</p>
</li>
<li>
<p>Select clinical from <code>Data Category</code> and <strong>bcr xml</strong> from <code>Data Format</code>, then Download data3.</p>
<p><img src="filesset3.png" alt="filesset3"></p>
</li>
<li>
<p>Count number of lines in all downloaded data.</p>
</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-markdown" data-lang="markdown"><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">$ wc -l gdc_manifest.2020-09-27-*
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">Output
</span></span><span class="line"><span class="cl">     568 gdc_manifest.2020-09-27-LUNG-miRNA-isoform.txt
</span></span><span class="line"><span class="cl">     568 gdc_manifest.2020-09-27-LUNG-miRNA-seq.txt
</span></span><span class="line"><span class="cl">     523 gdc_manifest.2020-09-27-clinical.txt
</span></span><span class="line"><span class="cl">    1659 total
</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>Download the <a href="https://gdc.cancer.gov/files/public/file/gdc-client_v1.6.0_OSX_x64_1.zip">GDC Data Transfer Tool Client</a>.</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-markdown" data-lang="markdown"><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">wget https://gdc.cancer.gov/files/public/file/gdc-client_v1.6.0_OSX_x64_1.zip
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">unzip gdc-client_v1.6.0_OSX_x64_1.zip
</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>Check the usage of gdc-client.</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-markdown" data-lang="markdown"><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">$ ./gdc-client --help
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">output
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">commands:
</span></span><span class="line"><span class="cl">  {download,upload,settings}
</span></span><span class="line"><span class="cl">                        for more information, specify -h after a command
</span></span><span class="line"><span class="cl">    download            download data from the GDC
</span></span><span class="line"><span class="cl">    upload              upload data to the GDC
</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>Check usage of gdc-client download.</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span><span class="lnt">42
</span><span class="lnt">43
</span><span class="lnt">44
</span><span class="lnt">45
</span><span class="lnt">46
</span><span class="lnt">47
</span><span class="lnt">48
</span><span class="lnt">49
</span><span class="lnt">50
</span><span class="lnt">51
</span><span class="lnt">52
</span><span class="lnt">53
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-markdown" data-lang="markdown"><span class="line"><span class="cl">$ ./gdc-client download --help
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">output
</span></span><span class="line"><span class="cl">usage: gdc-client download [-h] [--debug] [--log-file LOG_FILE] [--color_off]
</span></span><span class="line"><span class="cl">                           [-t TOKEN_FILE] [-d DIR] [-s server]
</span></span><span class="line"><span class="cl">                           [--no-segment-md5sums] [--no-file-md5sum]
</span></span><span class="line"><span class="cl">                           [-n N_PROCESSES]
</span></span><span class="line"><span class="cl">                           [--http-chunk-size HTTP_CHUNK_SIZE]
</span></span><span class="line"><span class="cl">                           [--save-interval SAVE_INTERVAL] [-k]
</span></span><span class="line"><span class="cl">                           [--no-related-files] [--no-annotations]
</span></span><span class="line"><span class="cl">                           [--no-auto-retry] [--retry-amount RETRY_AMOUNT]
</span></span><span class="line"><span class="cl">                           [--wait-time WAIT_TIME] [--latest] [--config FILE]
</span></span><span class="line"><span class="cl">                           [-m MANIFEST]
</span></span><span class="line"><span class="cl">                           [file_id [file_id ...]]
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">positional arguments:
</span></span><span class="line"><span class="cl">  file_id               The GDC UUID of the file(s) to download
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">optional arguments:
</span></span><span class="line"><span class="cl">  -h, --help            show this help message and exit
</span></span><span class="line"><span class="cl">  --debug               Enable debug logging. If a failure occurs, the program
</span></span><span class="line"><span class="cl">                        will stop.
</span></span><span class="line"><span class="cl">  --log-file LOG_FILE   Save logs to file. Amount logged affected by --debug
</span></span><span class="line"><span class="cl">  --color_off           Disable colored output
</span></span><span class="line"><span class="cl">  -t TOKEN_FILE, --token-file TOKEN_FILE
</span></span><span class="line"><span class="cl">                        GDC API auth token file
</span></span><span class="line"><span class="cl">  -d DIR, --dir DIR     Directory to download files to. Defaults to current
</span></span><span class="line"><span class="cl">                        directory
</span></span><span class="line"><span class="cl">  -s server, --server server
</span></span><span class="line"><span class="cl">                        The TCP server address server[:port]
</span></span><span class="line"><span class="cl">  --no-segment-md5sums  Do not calculate inbound segment md5sums and/or do not
</span></span><span class="line"><span class="cl">                        verify md5sums on restart
</span></span><span class="line"><span class="cl">  --no-file-md5sum      Do not verify file md5sum after download
</span></span><span class="line"><span class="cl">  -n N_PROCESSES, --n-processes N_PROCESSES
</span></span><span class="line"><span class="cl">                        Number of client connections.
</span></span><span class="line"><span class="cl">  --http-chunk-size HTTP_CHUNK_SIZE, -c HTTP_CHUNK_SIZE
</span></span><span class="line"><span class="cl">                        Size in bytes of standard HTTP block size.
</span></span><span class="line"><span class="cl">  --save-interval SAVE_INTERVAL
</span></span><span class="line"><span class="cl">                        The number of chunks after which to flush state file.
</span></span><span class="line"><span class="cl">                        A lower save interval will result in more frequent
</span></span><span class="line"><span class="cl">                        printout but lower performance.
</span></span><span class="line"><span class="cl">  -k, --no-verify       Perform insecure SSL connection and transfer
</span></span><span class="line"><span class="cl">  --no-related-files    Do not download related files.
</span></span><span class="line"><span class="cl">  --no-annotations      Do not download annotations.
</span></span><span class="line"><span class="cl">  --no-auto-retry       Ask before retrying to download a file
</span></span><span class="line"><span class="cl">  --retry-amount RETRY_AMOUNT
</span></span><span class="line"><span class="cl">                        Number of times to retry a download
</span></span><span class="line"><span class="cl">  --wait-time WAIT_TIME
</span></span><span class="line"><span class="cl">                        Amount of seconds to wait before retrying
</span></span><span class="line"><span class="cl">  --latest              Download latest version of a file if it exists
</span></span><span class="line"><span class="cl">  --config FILE         Path to INI-type config file
</span></span><span class="line"><span class="cl">  -m MANIFEST, --manifest MANIFEST
</span></span><span class="line"><span class="cl">                        GDC download manifest file
</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>Download Manifest files.</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-markdown" data-lang="markdown"><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">./gdc-client download -m gdc_manifest.2020-09-27-LUNG-miRNA-clinical.txt -d clinical/
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">./gdc-client download -m gdc_manifest.2020-09-27-LUNG-miRNA-isoform.txt -d isoform/
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">./gdc-client download -m gdc_manifest.2020-09-27-LUNG-miRNA-seq.txt -d miRNAseq/
</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>Check the information of files downloaded.</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-markdown" data-lang="markdown"><span class="line"><span class="cl">$ cd clinical
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="gh"># count total number of files downloaded
</span></span></span><span class="line"><span class="cl"><span class="gh"></span>$ ls|wc
</span></span><span class="line"><span class="cl">output
</span></span><span class="line"><span class="cl">    522     522   19314
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="gh"># count alive number of clinical information
</span></span></span><span class="line"><span class="cl"><span class="gh"></span>$ grep -i vital_status */*xml|grep Alive|wc
</span></span><span class="line"><span class="cl">output
</span></span><span class="line"><span class="cl">    840     840   10920
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="gh"># count dead number of clinical information
</span></span></span><span class="line"><span class="cl"><span class="gh"></span>$ grep -i vital_status */*xml|grep -v Alive|wc
</span></span><span class="line"><span class="cl">output
</span></span><span class="line"><span class="cl">     298    3580  104444
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="gh"># get sampes name/id:
</span></span></span><span class="line"><span class="cl"><span class="gh"></span>$ grep -i vital_status */*xml|grep Alive |cut -d&#34;.&#34; -f 3
</span></span><span class="line"><span class="cl">output
</span></span><span class="line"><span class="cl">TCGA-J2-8192
</span></span><span class="line"><span class="cl">TCGA-J2-8192
</span></span><span class="line"><span class="cl">TCGA-91-8499
</span></span><span class="line"><span class="cl">TCGA-91-8499
</span></span><span class="line"><span class="cl">TCGA-55-6986
</span></span><span class="line"><span class="cl">TCGA-55-6986
</span></span><span class="line"><span class="cl">TCGA-NJ-A4YG
</span></span><span class="line"><span class="cl">TCGA-NJ-A4YG
</span></span><span class="line"><span class="cl">......
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="gh"># count total number of alive samples
</span></span></span><span class="line"><span class="cl"><span class="gh"></span>$ grep -i vital_status */*xml|grep Alive |cut -d&#34;.&#34; -f 3|sort -u|wc
</span></span><span class="line"><span class="cl">output
</span></span><span class="line"><span class="cl">    395     395    5135
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="data-wrangling">Data Wrangling</h3>
<h4 id="data-of-overall-information">Data of overall Information</h4>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-markdown" data-lang="markdown"><span class="line"><span class="cl"><span class="gh"># count files number for clinical, isoform and miRNAseq
</span></span></span><span class="line"><span class="cl"><span class="gh"></span>$ ls clinical|wc
</span></span><span class="line"><span class="cl">output
</span></span><span class="line"><span class="cl">     522     522   19314
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">$ ls isoform|wc
</span></span><span class="line"><span class="cl">output
</span></span><span class="line"><span class="cl">     567     567   20979
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">$ ls miRNAseq|wc
</span></span><span class="line"><span class="cl">output
</span></span><span class="line"><span class="cl">     567     567   20979
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="r-scripts-for-single-sample">R Scripts For Single Sample</h3>
<ul>
<li>Choose one file of clinical randomly to get the format of sample.</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-markdown" data-lang="markdown"><span class="line"><span class="cl">$ ls clinical/57d733f6-2f8e-4f0c-b391-48d3cf7c1f87
</span></span><span class="line"><span class="cl">output
</span></span><span class="line"><span class="cl">nationwidechildrens.org_clinical.TCGA-75-7030.xml
</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li>
<p>Follow the usage of <a href="https://www.tutorialspoint.com/r/r_xml_files.htm">R - XML Files</a>.</p>
<ul>
<li>
<p>Open R.</p>
<p>The xml file is read by R using the function xmlParse(). It is stored as a list in R.</p>
</li>
<li>
<p>Reading XML File.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="c1"># Load the package required to read XML files.</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="s">&#34;XML&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Also load the other required package.</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="s">&#34;methods&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Give the input file name to the function.</span>
</span></span><span class="line"><span class="cl"><span class="n">result</span> <span class="o">&lt;-</span> <span class="nf">xmlParse</span><span class="p">(</span><span class="n">file</span> <span class="o">=</span> <span class="s">&#34;nationwidechildrens.org_clinical.TCGA-75-7030.xml&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Print the result.</span>
</span></span><span class="line"><span class="cl"><span class="nf">print</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
</span></span></code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>Get Number of Nodes Present in XML File.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="c1"># Exract the root node form the xml file.</span>
</span></span><span class="line"><span class="cl"><span class="n">rootnode</span> <span class="o">&lt;-</span> <span class="nf">xmlRoot</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Find number of nodes in the root.</span>
</span></span><span class="line"><span class="cl"><span class="n">rootsize</span> <span class="o">&lt;-</span> <span class="nf">xmlSize</span><span class="p">(</span><span class="n">rootnode</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Print the result.</span>
</span></span><span class="line"><span class="cl"><span class="nf">print</span><span class="p">(</span><span class="n">rootsize</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">output</span>
</span></span><span class="line"><span class="cl"><span class="n">[1]</span> <span class="m">2</span>
</span></span></code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>Details of the First Node and second Node.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="c1"># Exract the root node form the xml file.</span>
</span></span><span class="line"><span class="cl"><span class="n">rootnode</span> <span class="o">&lt;-</span> <span class="nf">xmlRoot</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># Print the result.</span>
</span></span><span class="line"><span class="cl"><span class="nf">print</span><span class="p">(</span><span class="n">rootnode[1]</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">print</span><span class="p">(</span><span class="n">rootnode[2]</span><span class="p">)</span>
</span></span></code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>XML to Data Frame.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="cl"><span class="c1"># Convert the input xml file to a data frame.</span>
</span></span><span class="line"><span class="cl"><span class="n">xmldataframe</span> <span class="o">&lt;-</span> <span class="nf">xmlToDataFrame</span><span class="p">(</span><span class="n">rootnode[2]</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">print</span><span class="p">(</span><span class="n">xmldataframe</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">t</span><span class="p">(</span><span class="n">xmldataframe</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">write.table</span><span class="p">(</span><span class="nf">t</span><span class="p">(</span><span class="n">xmldataframe</span><span class="p">),</span><span class="s">&#39;tmp&#39;</span><span class="p">)</span>
</span></span></code></pre></td></tr></table>
</div>
</div></li>
</ul>
</li>
</ul>
<h3 id="run-r-scripts-for-all-files">Run R Scripts For All files</h3>
<ul>
<li>
<p>Complete scripts according to that of single sample.</p>
</li>
<li>
<p>Run <a href="gdc.Rmd">Scripts</a>.</p>
<p><img src="RScript1.png" alt="RScript1"></p>
</li>
</ul>
<h3 id="example-2-fast-download">Example 2 Fast Download</h3>
<ul>
<li>
<p>Enter Key words in Google and search: <em><strong>tcga gdc lusc</strong></em>.</p>
<p><img src="google.png" alt="google"></p>
</li>
<li>
<p>Click on <a href="https://portal.gdc.cancer.gov/projects/TCGA-LUSC">site 1</a>, and download data of files.</p>
<p><img src="downloadfile.png" alt="downloadfile"></p>
</li>
<li>
<p>Choose and download interested clinical data.</p>
<p><img src="set2.png" alt="set2"></p>
</li>
<li>
<p>Choose and download interested Transcript data.</p>
<p><img src="transcript1.png" alt="transcript1"></p>
<hr>
<p><img src="transcript2.png" alt="transcript2"></p>
</li>
</ul>
<h3 id="run-rscripts">Run R.Scripts</h3>
<ul>
<li>
<p>Follow the usage of <a href="https://www.tutorialspoint.com/r/r_json_files.htm">R-JSON Files</a> and <a href="https://www.tutorialspoint.com/r/r_xml_files.htm">R - XML Files</a>.</p>
</li>
<li>
<p>Run <a href="Rjsonfiles.Rmd">Rjson Script</a> and get matched case id with file names.</p>
<p><img src="RJson.png" alt="RJson"></p>
</li>
</ul>
<h3 id="in-summary">In summary</h3>
<ul>
<li>Anyway, here is the official GDC Documentation <a href="https://docs.gdc.cancer.gov/">on line</a>.</li>
</ul>
    </div>

    
    


    
    

    <footer class="post-footer">
      <div class="post-tags">
          <a href="/tags/tcga/">TCGA</a>
          <a href="/tags/bioinformatics/">Bioinformatics</a>
          <a href="/tags/gdc/">GDC</a>
          <a href="/tags/r/">R</a>
          
        </div>

      
      <nav class="post-nav">
        
          <a class="prev" href="/post/oncolnc/">
            
            <i class="iconfont">
              <svg  class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="18" height="18">
  <path d="M691.908486 949.511495l75.369571-89.491197c10.963703-12.998035 10.285251-32.864502-1.499144-44.378743L479.499795 515.267417 757.434875 204.940602c11.338233-12.190647 11.035334-32.285311-0.638543-44.850487l-80.46666-86.564541c-11.680017-12.583596-30.356378-12.893658-41.662889-0.716314L257.233596 494.235404c-11.332093 12.183484-11.041474 32.266891 0.657986 44.844348l80.46666 86.564541c1.772366 1.910513 3.706415 3.533476 5.750981 4.877077l306.620399 321.703933C662.505829 963.726242 680.945807 962.528973 691.908486 949.511495z"></path>
</svg>

            </i>
            <span class="prev-text nav-default">OncoLnc: A straightforward Tool</span>
            <span class="prev-text nav-mobile">Prev</span>
          </a>
        
          <a class="next" href="/post/firebrowse/">
            <span class="next-text nav-default">FireBrowse Notebook</span>
            <span class="prev-text nav-mobile">Next</span>
            
            <i class="iconfont">
              <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="18" height="18">
  <path d="M332.091514 74.487481l-75.369571 89.491197c-10.963703 12.998035-10.285251 32.864502 1.499144 44.378743l286.278095 300.375162L266.565125 819.058374c-11.338233 12.190647-11.035334 32.285311 0.638543 44.850487l80.46666 86.564541c11.680017 12.583596 30.356378 12.893658 41.662889 0.716314l377.434212-421.426145c11.332093-12.183484 11.041474-32.266891-0.657986-44.844348l-80.46666-86.564541c-1.772366-1.910513-3.706415-3.533476-5.750981-4.877077L373.270379 71.774697C361.493148 60.273758 343.054193 61.470003 332.091514 74.487481z"></path>
</svg>

            </i>
          </a>
      </nav>
    </footer>
  </article>

  
  

  
  

  

  
  

  

  

  

    

  

        </div>
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="icon-links">
  
  
    <a href="mailto:xiaoni0601@gmail.com" rel="me noopener" class="iconfont"
      title="email" >
      <svg class="icon" viewBox="0 0 1451 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="36" height="36">
  <path d="M664.781909 681.472759 0 97.881301C0 3.997201 71.046997 0 71.046997 0L474.477909 0 961.649408 0 1361.641813 0C1361.641813 0 1432.688811 3.997201 1432.688811 97.881301L771.345323 681.472759C771.345323 681.472759 764.482731 685.154773 753.594283 688.65053L753.594283 688.664858C741.602731 693.493018 729.424896 695.068979 718.077952 694.839748 706.731093 695.068979 694.553173 693.493018 682.561621 688.664858L682.561621 688.65053C671.644501 685.140446 664.781909 681.472759 664.781909 681.472759L664.781909 681.472759ZM718.063616 811.603883C693.779541 811.016482 658.879232 802.205449 619.10784 767.734955 542.989056 701.759633 0 212.052267 0 212.052267L0 942.809523C0 942.809523 0 1024 83.726336 1024L682.532949 1024 753.579947 1024 1348.948139 1024C1432.688811 1024 1432.688811 942.809523 1432.688811 942.809523L1432.688811 212.052267C1432.688811 212.052267 893.138176 701.759633 817.019477 767.734955 777.248 802.205449 742.347691 811.03081 718.063616 811.603883L718.063616 811.603883Z"></path>
</svg>

    </a>
  
    <a href="https://github.com/xiaonilee" rel="me noopener" class="iconfont"
      title="github"  target="_blank"
      >
      <svg class="icon" style="" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="36" height="36">
  <path d="M512 12.672c-282.88 0-512 229.248-512 512 0 226.261333 146.688 418.133333 350.08 485.76 25.6 4.821333 34.986667-11.008 34.986667-24.618667 0-12.16-0.426667-44.373333-0.64-87.04-142.421333 30.890667-172.458667-68.693333-172.458667-68.693333C188.672 770.986667 155.008 755.2 155.008 755.2c-46.378667-31.744 3.584-31.104 3.584-31.104 51.413333 3.584 78.421333 52.736 78.421333 52.736 45.653333 78.293333 119.850667 55.68 149.12 42.581333 4.608-33.109333 17.792-55.68 32.426667-68.48-113.706667-12.8-233.216-56.832-233.216-253.013333 0-55.893333 19.84-101.546667 52.693333-137.386667-5.76-12.928-23.04-64.981333 4.48-135.509333 0 0 42.88-13.738667 140.8 52.48 40.96-11.392 84.48-17.024 128-17.28 43.52 0.256 87.04 5.888 128 17.28 97.28-66.218667 140.16-52.48 140.16-52.48 27.52 70.528 10.24 122.581333 5.12 135.509333 32.64 35.84 52.48 81.493333 52.48 137.386667 0 196.693333-119.68 240-233.6 252.586667 17.92 15.36 34.56 46.762667 34.56 94.72 0 68.522667-0.64 123.562667-0.64 140.202666 0 13.44 8.96 29.44 35.2 24.32C877.44 942.592 1024 750.592 1024 524.672c0-282.752-229.248-512-512-512"></path>
</svg>

    </a>


<a href="/index.xml" rel="noopener alternate" type="application/rss&#43;xml"
    class="iconfont" title="rss" target="_blank">
    <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="30" height="30">
  <path d="M819.157333 1024C819.157333 574.592 449.408 204.8 0 204.8V0c561.706667 0 1024 462.293333 1024 1024h-204.842667zM140.416 743.04a140.8 140.8 0 0 1 140.501333 140.586667A140.928 140.928 0 0 1 140.074667 1024C62.72 1024 0 961.109333 0 883.626667s62.933333-140.544 140.416-140.586667zM678.784 1024h-199.04c0-263.210667-216.533333-479.786667-479.744-479.786667V345.173333c372.352 0 678.784 306.517333 678.784 678.826667z"></path>
</svg>

  </a>
   
</div>

<div class="copyright">

  <span class="copyright-year">
    &copy;
    
      2020 -
    2023
    <span class="heart">
      
      <i class="iconfont">
        <svg class="icon" viewBox="0 0 1025 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="14" height="14">
  <path d="M1000.1 247.9c-15.5-37.3-37.6-70.6-65.7-98.9-54.4-54.8-125.8-85-201-85-85.7 0-166 39-221.4 107.4C456.6 103 376.3 64 290.6 64c-75.1 0-146.5 30.4-201.1 85.6-28.2 28.5-50.4 61.9-65.8 99.3-16 38.8-24 79.9-23.6 122.2 0.7 91.7 40.1 177.2 108.1 234.8 3.1 2.6 6 5.1 8.9 7.8 14.9 13.4 58 52.8 112.6 102.7 93.5 85.5 209.9 191.9 257.5 234.2 7 6.1 15.8 9.5 24.9 9.5 9.2 0 18.1-3.4 24.9-9.5 34.5-30.7 105.8-95.9 181.4-165 74.2-67.8 150.9-138 195.8-178.2 69.5-57.9 109.6-144.4 109.9-237.3 0.1-42.5-8-83.6-24-122.2z"
   fill="#8a8a8a"></path>
</svg>

      </i>
    </span><span class="author">
        xiaonilee
        
      </span></span>

  
  

  
</div>

    </footer>

    <div class="back-to-top" id="back-to-top">
      <i class="iconfont">
        
        <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="35" height="35">
  <path d="M510.866688 227.694839 95.449397 629.218702l235.761562 0-2.057869 328.796468 362.40389 0L691.55698 628.188232l241.942331-3.089361L510.866688 227.694839zM63.840492 63.962777l894.052392 0 0 131.813095L63.840492 195.775872 63.840492 63.962777 63.840492 63.962777zM63.840492 63.962777"></path>
</svg>

      </i>
    </div>
  </div>
  
<script type="text/javascript" src="/lib/jquery/jquery-3.2.1.min.js"></script>
  <script type="text/javascript" src="/lib/slideout/slideout-1.0.1.min.js"></script>




<script type="text/javascript" src="/js/main.dee43230127a73d039a734510fa896c89c3c7ce0cf0be0c7a7433f8fd69b76dc.js" integrity="sha256-3uQyMBJ6c9A5pzRRD6iWyJw8fODPC&#43;DHp0M/j9abdtw=" crossorigin="anonymous"></script>



  <script type="text/javascript">
    window.MathJax = {
      showProcessingMessages: false,
      messageStyle: 'none'
    };
  </script>
  <script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-MML-AM_CHTML' async></script>









  
    <script type="text/javascript" src="/js/load-photoswipe.js"></script>
    <script type="text/javascript" src="/lib/photoswipe/photoswipe.min.js"></script>
    <script type="text/javascript" src="/lib/photoswipe/photoswipe-ui-default.min.js"></script>
  















</body>
</html>
