<!DOCTYPE html>
<html lang="en" itemscope itemtype="http://schema.org/WebPage">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>UCSC gene browser with R package Gviz - part 1 - xiaoni&#39;s blog</title>
  

<meta name="renderer" content="webkit" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>

<meta name="MobileOptimized" content="width"/>
<meta name="HandheldFriendly" content="true"/>


<meta name="applicable-device" content="pc,mobile">

<meta name="theme-color" content="#f8f5ec" />
<meta name="msapplication-navbutton-color" content="#f8f5ec">
<meta name="apple-mobile-web-app-capable" content="yes">
<meta name="apple-mobile-web-app-status-bar-style" content="#f8f5ec">

<meta name="mobile-web-app-capable" content="yes">

<meta name="author" content="Xiaoni" />
  <meta name="description" content="Gviz, Visualize genomic data. Let&amp;rsquo;s dive into it to explore useful functions.
" />

  <meta name="keywords" content="xiaonili, xiaonilee, xiaoni" />






<meta name="generator" content="Hugo 0.79.0" />


<link rel="canonical" href="/post/ucscgenebrowser/" />





<link rel="icon" href="/favicon.ico" />











<link rel="stylesheet" href="/sass/jane.min.f1e506a781bf25d33ffc18aa6b4e972a965c58049d27d4f92b7db2e9bf28e4bf.css" integrity="sha256-8eUGp4G/JdM//Biqa06XKpZcWASdJ9T5K32y6b8o5L8=" media="screen" crossorigin="anonymous">



<link href="https://cdn.bootcdn.net/ajax/libs/font-awesome/5.15.1/css/all.css" rel="stylesheet">
<script src="https://cdn.bootcdn.net/ajax/libs/font-awesome/5.15.1/js/all.js"></script>




<meta property="og:title" content="UCSC gene browser with R package Gviz - part 1" />
<meta property="og:description" content="Gviz, Visualize genomic data. Let&rsquo;s dive into it to explore useful functions." />
<meta property="og:type" content="article" />
<meta property="og:url" content="/post/ucscgenebrowser/" />
<meta property="article:published_time" content="2021-02-13T00:00:00+00:00" />
<meta property="article:modified_time" content="2021-02-19T00:00:00+00:00" />
<meta itemprop="name" content="UCSC gene browser with R package Gviz - part 1">
<meta itemprop="description" content="Gviz, Visualize genomic data. Let&rsquo;s dive into it to explore useful functions.">
<meta itemprop="datePublished" content="2021-02-13T00:00:00+00:00" />
<meta itemprop="dateModified" content="2021-02-19T00:00:00+00:00" />
<meta itemprop="wordCount" content="1664">



<meta itemprop="keywords" content="R,Bioinformatics,UCSC,R Package," />
<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="UCSC gene browser with R package Gviz - part 1"/>
<meta name="twitter:description" content="Gviz, Visualize genomic data. Let&rsquo;s dive into it to explore useful functions."/>

<!--[if lte IE 9]>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/classlist/1.1.20170427/classList.min.js"></script>
<![endif]-->

<!--[if lt IE 9]>
  <script src="https://cdn.jsdelivr.net/npm/html5shiv@3.7.3/dist/html5shiv.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/respond.js@1.4.2/dest/respond.min.js"></script>
<![endif]-->


<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'UA-177860608-1', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>



</head>
<body>
  <div id="mobile-navbar" class="mobile-navbar">
  <div class="mobile-header-logo">
    <a href="/" class="logo">Xiaoni's Blog</a>
  </div>
  <div class="mobile-navbar-icon">
    <span></span>
    <span></span>
    <span></span>
  </div>
</div>
<nav id="mobile-menu" class="mobile-menu slideout-menu">
  <ul class="mobile-menu-list">
    <li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/">Home</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/post/">Archives</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/categories/">Categories</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/tags/">Tags</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/about/">About Me</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          <div class="mobile-menu-parent">
            <span class="mobile-submenu-open"></span>
            <a href="/">
              docs
            </a>
          </div>
          <ul class="mobile-submenu-list">
            
              <li>
                <a href="/post/tensorflow2/"> Machine Learning: TensorFlow2 Notebook</a>
              </li>
            
              <li>
                <a href="/post/topics/"> Topics in my blog</a>
              </li>
            
              <li>
                <a href="/post/bioscirep/">Bioinformatics analysis of prognostic value of PITX1 gene in breast cancer</a>
              </li>
            
              <li>
                <a href="/post/rinactionmap/">Book: R in Action</a>
              </li>
            
              <li>
                <a href="/post/tumorimmune/">Comprehensive Bioinformatics Analysis of Tumor-Infiltrating Immune Cells</a>
              </li>
            
              <li>
                <a href="/post/firebrowse/">FireBrowse Notebook</a>
              </li>
            
              <li>
                <a href="/post/gdc/">GDC Notebook</a>
              </li>
            
              <li>
                <a href="/post/gscalite/">GSCALite: Gene Set Cancer Analysis</a>
              </li>
            
              <li>
                <a href="/post/frontoncol/">Identification of Therapeutic Targets and Prognostic Biomarkers Among CXC Chemokines in the Renal Cell Carcinoma Microenvironment</a>
              </li>
            
              <li>
                <a href="/post/condacolab/">Installing Conda in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/oncolnc/">OncoLnc: A straightforward Tool</a>
              </li>
            
              <li>
                <a href="/post/webcrawler/">Python Spider</a>
              </li>
            
              <li>
                <a href="/post/rnasequenceanalysis/">RNA-seq Bioinformatics - RNA sequence analysis</a>
              </li>
            
              <li>
                <a href="/post/scrnaknn/">scRNA-sequencing analysis - PCA and KNN</a>
              </li>
            
              <li>
                <a href="/post/survival/">Survival Analysis with data from GSE database</a>
              </li>
            
              <li>
                <a href="/post/rincolab/">The way to use R in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/tissgdb/">TissGDB: tissue-specific gene database in cancer</a>
              </li>
            
              <li>
                <a href="/post/ucscmap/">UCSC Topics</a>
              </li>
            
              <li>
                <a href="/post/webfrontend/">web front end</a>
              </li>
            
          </ul>
        
      </li>
    

    
  </ul>
</nav>


  
    






  <link rel="stylesheet" href="/lib/photoswipe/photoswipe.min.css" />
  <link rel="stylesheet" href="/lib/photoswipe/default-skin/default-skin.min.css" />




<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

<div class="pswp__bg"></div>

<div class="pswp__scroll-wrap">
    
    <div class="pswp__container">
      <div class="pswp__item"></div>
      <div class="pswp__item"></div>
      <div class="pswp__item"></div>
    </div>
    
    <div class="pswp__ui pswp__ui--hidden">
    <div class="pswp__top-bar">
      
      <div class="pswp__counter"></div>
      <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>
      <button class="pswp__button pswp__button--share" title="Share"></button>
      <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>
      <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>
      
      
      <div class="pswp__preloader">
        <div class="pswp__preloader__icn">
          <div class="pswp__preloader__cut">
            <div class="pswp__preloader__donut"></div>
          </div>
        </div>
      </div>
    </div>
    <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
      <div class="pswp__share-tooltip"></div>
    </div>
    <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
    </button>
    <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
    </button>
    <div class="pswp__caption">
      <div class="pswp__caption__center"></div>
    </div>
    </div>
    </div>
</div>

  

  

  

  <header id="header" class="header container">
    <div class="logo-wrapper">
  <a href="/" class="logo">
    
      Xiaoni's Blog
    
  </a>
</div>

<nav class="site-navbar">
  <ul id="menu" class="menu">
    
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/">Home</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/post/">Archives</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/categories/">Categories</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/tags/">Tags</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/about/">About Me</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          <a class="menu-item-link menu-parent" href="/">docs</a>
          <ul class="submenu" >
            
              <li>
                <a href="/post/tensorflow2/"> Machine Learning: TensorFlow2 Notebook</a>
              </li>
            
              <li>
                <a href="/post/topics/"> Topics in my blog</a>
              </li>
            
              <li>
                <a href="/post/bioscirep/">Bioinformatics analysis of prognostic value of PITX1 gene in breast cancer</a>
              </li>
            
              <li>
                <a href="/post/rinactionmap/">Book: R in Action</a>
              </li>
            
              <li>
                <a href="/post/tumorimmune/">Comprehensive Bioinformatics Analysis of Tumor-Infiltrating Immune Cells</a>
              </li>
            
              <li>
                <a href="/post/firebrowse/">FireBrowse Notebook</a>
              </li>
            
              <li>
                <a href="/post/gdc/">GDC Notebook</a>
              </li>
            
              <li>
                <a href="/post/gscalite/">GSCALite: Gene Set Cancer Analysis</a>
              </li>
            
              <li>
                <a href="/post/frontoncol/">Identification of Therapeutic Targets and Prognostic Biomarkers Among CXC Chemokines in the Renal Cell Carcinoma Microenvironment</a>
              </li>
            
              <li>
                <a href="/post/condacolab/">Installing Conda in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/oncolnc/">OncoLnc: A straightforward Tool</a>
              </li>
            
              <li>
                <a href="/post/webcrawler/">Python Spider</a>
              </li>
            
              <li>
                <a href="/post/rnasequenceanalysis/">RNA-seq Bioinformatics - RNA sequence analysis</a>
              </li>
            
              <li>
                <a href="/post/scrnaknn/">scRNA-sequencing analysis - PCA and KNN</a>
              </li>
            
              <li>
                <a href="/post/survival/">Survival Analysis with data from GSE database</a>
              </li>
            
              <li>
                <a href="/post/rincolab/">The way to use R in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/tissgdb/">TissGDB: tissue-specific gene database in cancer</a>
              </li>
            
              <li>
                <a href="/post/ucscmap/">UCSC Topics</a>
              </li>
            
              <li>
                <a href="/post/webfrontend/">web front end</a>
              </li>
            
          </ul>

        

      </li>
    

    
    

    
  </ul>
</nav>

  </header>

  <div id="mobile-panel">
    <main id="main" class="main bg-llight">
      <div class="content-wrapper">
        <div id="content" class="content container">
          <article class="post bg-white">
    
    <header class="post-header">
      <h1 class="post-title">UCSC gene browser with R package Gviz - part 1</h1>
      
      <div class="post-meta">
        <time datetime="2021-02-13" class="post-time">
          2021-02-13
        </time>
        <div class="post-category">
            <a href="/categories/r/"> R </a>
            <a href="/categories/bioinformatics/"> Bioinformatics </a>
            <a href="/categories/ucsc/"> UCSC </a>
            <a href="/categories/r-package/"> R Package </a>
            
          </div>
        

        
        

        
        
      </div>
    </header>

    
    
<div class="post-toc" id="post-toc">
  <h2 class="post-toc-title">Table of Contents</h2>
  <div class="post-toc-content">
    <nav id="TableOfContents">
  <ul>
    <li><a href="#introduction-to-gviz-package">Introduction to Gviz package</a>
      <ul>
        <li><a href="#plot-annotation-track">Plot annotation track</a></li>
        <li><a href="#add-genome-axis-track">Add genome axis track</a></li>
        <li><a href="#add-chromosome-ideogram">Add chromosome ideogram</a></li>
        <li><a href="#add-gene-model">Add gene model</a></li>
        <li><a href="#zoom-the-plot">Zoom the plot</a></li>
        <li><a href="#add-sequence-track-and-zoom-to-view-sequence">Add sequence track and zoom to view sequence</a></li>
        <li><a href="#add-data-track">Add data track</a></li>
      </ul>
    </li>
    <li><a href="#plotting-parameters">Plotting parameters</a>
      <ul>
        <li><a href="#plotting-direction">Plotting direction</a></li>
      </ul>
    </li>
    <li><a href="#track-classes">Track classes</a>
      <ul>
        <li><a href="#ideogramtrack">IdeogramTrack</a></li>
        <li><a href="#datatrack">DataTrack</a></li>
        <li><a href="#example-of-datatrack-plots">Example of DataTrack plots</a></li>
        <li><a href="#data-grouping">Data grouping</a></li>
        <li><a href="#building-datatrack-objects-from-files">Building DataTrack objects from files</a></li>
        <li><a href="#annotationtrack">AnnotationTrack</a></li>
        <li><a href="#add-id-information">Add id information</a></li>
        <li><a href="#change-fontcolor">change fontcolor</a></li>
        <li><a href="#generegiontrack">GeneRegionTrack</a></li>
        <li><a href="#building-generegiontrack-objects-from-transcriptdbs">Building GeneRegionTrack objects from TranscriptDbs</a></li>
        <li><a href="#biomartgeneregiontrack">BiomartGeneRegionTrack</a></li>
        <li><a href="#sequence-track">Sequence Track</a></li>
        <li><a href="#alignmentstrack">AlignmentsTrack</a></li>
        <li><a href="#zoom-in-a-bit-further-to-check-out-the-details-of-the-pile-ups-section">Zoom in a bit further to check out the details of the pile-ups section</a></li>
        <li><a href="#dnaseq-experiment">DNAseq experiment</a></li>
      </ul>
    </li>
    <li><a href="#track-highlighting-and-overlays">Track highlighting and overlays</a>
      <ul>
        <li><a href="#overlays-multiple-tracks-on-the-same-area-of-the-plot">overlays multiple tracks on the same area of the plot</a></li>
      </ul>
    </li>
  </ul>
</nav>
  </div>
</div>

    
    <div class="post-content">
      <p>Gviz, Visualize genomic data. Let&rsquo;s dive into it to explore useful functions.</p>
<h2 id="introduction-to-gviz-package">Introduction to Gviz package</h2>
<ul>
<li>Provide a structured visualization framework to plot any type of data along genomic coordinates.</li>
<li>Integrate publicly available genomic annotation data from sources like UCSC or ENSEMBL.</li>
</ul>
<h3 id="plot-annotation-track">Plot annotation track</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">library</span><span class="p">(</span><span class="n">Gviz</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">GenomicRanges</span><span class="p">)</span>
<span class="c1">#Load data : class = GRanges</span>
<span class="nf">data</span><span class="p">(</span><span class="n">cpgIslands</span><span class="p">)</span>
<span class="n">cpgIslands</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p0.png" alt="p0"></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#Annotation track, title =&#34;CpG&#34;</span>
<span class="n">atrack</span> <span class="o">&lt;-</span> <span class="nf">AnnotationTrack</span><span class="p">(</span><span class="n">cpgIslands</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;CpG&#34;</span><span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">atrack</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p1.png" alt="p1"></p>
<h3 id="add-genome-axis-track">Add genome axis track</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">## genomic coordinates</span>
<span class="n">gtrack</span> <span class="o">&lt;-</span> <span class="nf">GenomeAxisTrack</span><span class="p">()</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack</span><span class="p">))</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p2.png" alt="p2"></p>
<h3 id="add-chromosome-ideogram">Add chromosome ideogram</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#genome : &#34;hg19&#34; </span>
<span class="n">gen</span><span class="o">&lt;-</span><span class="nf">genome</span><span class="p">(</span><span class="n">cpgIslands</span><span class="p">)</span>
<span class="c1">#Chromosme name : &#34;chr7&#34;</span>
<span class="n">chr</span> <span class="o">&lt;-</span> <span class="nf">as.character</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="nf">seqnames</span><span class="p">(</span><span class="n">cpgIslands</span><span class="p">)))</span>
<span class="c1">#Ideogram track</span>
<span class="n">itrack</span> <span class="o">&lt;-</span> <span class="nf">IdeogramTrack</span><span class="p">(</span><span class="n">genome</span> <span class="o">=</span> <span class="n">gen</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">chr</span><span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span> <span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack</span><span class="p">))</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p3.png" alt="p3"></p>
<h3 id="add-gene-model">Add gene model</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#Load data</span>
<span class="nf">data</span><span class="p">(</span><span class="n">geneModels</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">geneModels</span><span class="p">)</span>
<span class="c1">#Plot</span>
<span class="n">grtrack</span> <span class="o">&lt;-</span> <span class="nf">GeneRegionTrack</span><span class="p">(</span><span class="n">geneModels</span><span class="p">,</span> <span class="n">genome</span> <span class="o">=</span> <span class="n">gen</span><span class="p">,</span>
                           <span class="n">chromosome</span> <span class="o">=</span> <span class="n">chr</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;Gene Model&#34;</span><span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span> <span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack</span><span class="p">,</span> <span class="n">grtrack</span><span class="p">))</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p4.png" alt="p4"></p>
<h3 id="zoom-the-plot">Zoom the plot</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#Use from and to arguments to zoom</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span> <span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack</span><span class="p">,</span> <span class="n">grtrack</span><span class="p">),</span>
           <span class="n">from</span> <span class="o">=</span> <span class="m">26700000</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="m">26750000</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p5.png" alt="p5"></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1"># Use extend.left and extend.right to zoom</span>
<span class="c1">#those arguments are relative to the currently displayed ranges, </span>
<span class="c1">#and can be used to quickly extend the view on one or both ends of the plot.</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span> <span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack</span><span class="p">,</span> <span class="n">grtrack</span><span class="p">),</span>
           <span class="n">extend.left</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> <span class="n">extend.right</span> <span class="o">=</span> <span class="m">1000000</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p6.png" alt="p6"></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1"># to drop the bounding borders of the exons and </span>
<span class="c1"># to have a nice plot</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span> <span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack</span><span class="p">,</span> <span class="n">grtrack</span><span class="p">),</span>
           <span class="n">extend.left</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> <span class="n">extend.right</span> <span class="o">=</span> <span class="m">1000000</span><span class="p">,</span> <span class="n">col</span> <span class="o">=</span> <span class="kc">NULL</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p7.png" alt="p7"></p>
<h3 id="add-sequence-track-and-zoom-to-view-sequence">Add sequence track and zoom to view sequence</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1"># if (!requireNamespace(&#34;BiocManager&#34;, quietly = TRUE))</span>
<span class="c1">#     install.packages(&#34;BiocManager&#34;)</span>
<span class="c1"># </span>
<span class="c1"># BiocManager::install(&#34;BSgenome.Hsapiens.UCSC.hg19&#34;)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">BSgenome.Hsapiens.UCSC.hg19</span><span class="p">)</span>
<span class="n">strack</span> <span class="o">&lt;-</span> <span class="nf">SequenceTrack</span><span class="p">(</span><span class="n">Hsapiens</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">chr</span><span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span> <span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack</span><span class="p">,</span> <span class="n">grtrack</span><span class="p">,</span>
                <span class="n">strack</span><span class="p">),</span> <span class="n">from</span> <span class="o">=</span> <span class="m">26591822</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="m">26591852</span><span class="p">,</span> <span class="n">cex</span> <span class="o">=</span> <span class="m">0.8</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p8.png" alt="p8"></p>
<h3 id="add-data-track">Add data track</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#For demonstration purposes we can create a simple DataTrack object from</span>
<span class="c1">#randomly sampled data.</span>
<span class="nf">set.seed</span><span class="p">(</span><span class="m">255</span><span class="p">)</span>
<span class="n">lim</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="m">26700000</span><span class="p">,</span> <span class="m">26750000</span><span class="p">)</span>
<span class="n">coords</span> <span class="o">&lt;-</span> <span class="nf">sort</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">lim[1]</span><span class="p">,</span> <span class="nf">sample</span><span class="p">(</span><span class="nf">seq</span><span class="p">(</span><span class="n">from</span> <span class="o">=</span> <span class="n">lim[1]</span><span class="p">,</span> 
                                    <span class="n">to</span> <span class="o">=</span> <span class="n">lim[2]</span><span class="p">),</span> <span class="m">99</span><span class="p">),</span> <span class="n">lim[2]</span><span class="p">))</span>
<span class="n">dat</span> <span class="o">&lt;-</span> <span class="nf">runif</span><span class="p">(</span><span class="m">100</span><span class="p">,</span> <span class="n">min</span> <span class="o">=</span> <span class="m">-10</span><span class="p">,</span> <span class="n">max</span> <span class="o">=</span> <span class="m">10</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>

<span class="c1">##data track</span>
<span class="n">dtrack</span> <span class="o">&lt;-</span> <span class="nf">DataTrack</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">dat</span><span class="p">,</span> <span class="n">start</span> <span class="o">=</span> <span class="n">coords[</span><span class="o">-</span><span class="nf">length</span><span class="p">(</span><span class="n">coords</span><span class="p">)</span><span class="n">]</span><span class="p">,</span>
                    <span class="n">end</span> <span class="o">=</span> <span class="n">coords[</span><span class="m">-1</span><span class="n">]</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">chr</span><span class="p">,</span> <span class="n">genome</span> <span class="o">=</span> <span class="n">gen</span><span class="p">,</span>
                    <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;Uniform&#34;</span><span class="p">)</span>
<span class="c1">##Plot data track</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span> <span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack</span><span class="p">,</span> <span class="n">grtrack</span><span class="p">,</span>
                <span class="n">dtrack</span><span class="p">),</span> <span class="n">from</span> <span class="o">=</span> <span class="n">lim[1]</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="n">lim[2]</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p9.png" alt="p9"></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#Change plot type to histogram</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span> <span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack</span><span class="p">,</span> <span class="n">grtrack</span><span class="p">,</span><span class="n">dtrack</span><span class="p">),</span>
           <span class="n">from</span> <span class="o">=</span> <span class="n">lim[1]</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="n">lim[2]</span><span class="p">,</span> <span class="n">type</span> <span class="o">=</span> <span class="s">&#34;histogram&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p10.png" alt="p10"></p>
<h2 id="plotting-parameters">Plotting parameters</h2>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span><span class="lnt">9
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#setting parameters</span>
<span class="c1">#Annotation of transcript</span>
<span class="c1">#Change panel and title background color</span>
<span class="n">grtrack</span> <span class="o">&lt;-</span> <span class="nf">GeneRegionTrack</span><span class="p">(</span><span class="n">geneModels</span><span class="p">,</span> <span class="n">genome</span> <span class="o">=</span> <span class="n">gen</span><span class="p">,</span>
                           <span class="n">chromosome</span> <span class="o">=</span> <span class="n">chr</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;Gene Model&#34;</span><span class="p">,</span> 
                           <span class="n">transcriptAnnotation</span> <span class="o">=</span> <span class="s">&#34;symbol&#34;</span><span class="p">,</span>
                           <span class="n">background.panel</span> <span class="o">=</span> <span class="s">&#34;#FFFEDB&#34;</span><span class="p">,</span>
                           <span class="n">background.title</span> <span class="o">=</span> <span class="s">&#34;darkblue&#34;</span><span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span> <span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack</span><span class="p">,</span> <span class="n">grtrack</span><span class="p">))</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p11.png" alt="p11"></p>
<h3 id="plotting-direction">Plotting direction</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#By default all tracks will be plotted in a 5’ -&gt; 3’ direction. </span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span> <span class="n">gtrack</span><span class="p">,</span> <span class="n">atrack</span><span class="p">,</span> <span class="n">grtrack</span><span class="p">),</span>
           <span class="n">reverseStrand</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="p12.png" alt="p12"></p>
<h2 id="track-classes">Track classes</h2>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span><span class="lnt">9
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#Set the position of labels to below, show IDs, change color</span>


<span class="n">axisTrack</span> <span class="o">&lt;-</span> <span class="nf">GenomeAxisTrack</span><span class="p">(</span><span class="n">range</span> <span class="o">=</span> <span class="nf">IRanges</span><span class="p">(</span><span class="n">start</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">2000000</span><span class="p">,</span><span class="m">4000000</span><span class="p">),</span> 
                                             <span class="n">end</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">3000000</span><span class="p">,</span> <span class="m">7000000</span><span class="p">),</span>
                                             <span class="n">names</span> <span class="o">=</span> <span class="nf">rep</span><span class="p">(</span><span class="s">&#34;N-stretch&#34;</span><span class="p">,</span> <span class="m">2</span><span class="p">))</span>
                             <span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">axisTrack</span><span class="p">,</span> <span class="n">from</span> <span class="o">=</span> <span class="m">1000000</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="m">9000000</span><span class="p">,</span> 
           <span class="n">labelPos</span> <span class="o">=</span> <span class="s">&#34;below&#34;</span><span class="p">,</span><span class="n">showId</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> <span class="n">col</span><span class="o">=</span><span class="s">&#34;red&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p13.png" alt="p13"></p>
<h3 id="ideogramtrack">IdeogramTrack</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#Ideogram</span>
<span class="n">ideoTrack</span> <span class="o">&lt;-</span> <span class="nf">IdeogramTrack</span><span class="p">(</span><span class="n">genome</span> <span class="o">=</span> <span class="s">&#34;hg19&#34;</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="s">&#34;chrX&#34;</span><span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">ideoTrack</span><span class="p">,</span> <span class="n">from</span> <span class="o">=</span> <span class="m">85000000</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="m">129000000</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p14.png" alt="p14"></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#Show chromosome band ID</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">ideoTrack</span><span class="p">,</span> <span class="n">from</span> <span class="o">=</span> <span class="m">85000000</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="m">129000000</span><span class="p">,</span>
           <span class="n">showId</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">showBandId</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">cex.bands</span> <span class="o">=</span> <span class="m">0.4</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p15.png" alt="p15"></p>
<h3 id="datatrack">DataTrack</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#Load data</span>
<span class="nf">data</span><span class="p">(</span><span class="n">twoGroups</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">twoGroups</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p160.png" alt="p160"></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#Plot data track</span>
<span class="n">dTrack</span> <span class="o">&lt;-</span> <span class="nf">DataTrack</span><span class="p">(</span><span class="n">twoGroups</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;uniform&#34;</span><span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">dTrack</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p16.png" alt="p16"></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#The different plot types</span>
<span class="c1">#dotplot</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">DataTrack</span><span class="p">(</span><span class="n">twoGroups</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;p&#34;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&#34;p&#34;</span><span class="p">)</span>
<span class="c1">#lines plot</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">DataTrack</span><span class="p">(</span><span class="n">twoGroups</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;l&#34;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&#34;l&#34;</span><span class="p">)</span>
<span class="c1">#line and dot plot</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">DataTrack</span><span class="p">(</span><span class="n">twoGroups</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;b&#34;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&#34;b&#34;</span><span class="p">)</span>
<span class="c1">#lines plot of average</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">DataTrack</span><span class="p">(</span><span class="n">twoGroups</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;a&#34;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&#34;a&#34;</span><span class="p">)</span>
<span class="c1">#histogram lines</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">DataTrack</span><span class="p">(</span><span class="n">twoGroups</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;h&#34;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&#34;h&#34;</span><span class="p">)</span>
<span class="c1">#histogram histogram (bar width equal to range with)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">DataTrack</span><span class="p">(</span><span class="n">twoGroups</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;histogram&#34;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&#34;histogram&#34;</span><span class="p">)</span>
<span class="c1">#&#39;polygon-type&#39; plot relative to a baseline</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">DataTrack</span><span class="p">(</span><span class="n">twoGroups</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;polygon&#34;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&#34;polygon&#34;</span><span class="p">)</span>
<span class="c1">#box and whisker plot</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">DataTrack</span><span class="p">(</span><span class="n">twoGroups</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;boxplot&#34;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&#34;boxplot&#34;</span><span class="p">)</span>
<span class="c1">#false color image of the individual values</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">DataTrack</span><span class="p">(</span><span class="n">twoGroups</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;heatmap&#34;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&#34;heatmap&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p17new.png" alt="p17"></p>
<h3 id="example-of-datatrack-plots">Example of DataTrack plots</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#Combine a boxplot with an average line and a data grid (g):</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">dTrack</span><span class="p">,</span> <span class="n">type</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;boxplot&#34;</span><span class="p">,</span> <span class="s">&#34;a&#34;</span><span class="p">,</span> <span class="s">&#34;g&#34;</span><span class="p">))</span>
<span class="c1">#Heatmap and show sample names</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">dTrack</span><span class="p">,</span> <span class="n">type</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;heatmap&#34;</span><span class="p">),</span> <span class="n">showSampleNames</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span>
           <span class="n">cex.sampleNames</span> <span class="o">=</span> <span class="m">0.6</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p18new.png" alt="p18"></p>
<h3 id="data-grouping">Data grouping</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">plotTracks</span><span class="p">(</span><span class="n">dTrack</span><span class="p">,</span> <span class="n">groups</span> <span class="o">=</span> <span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&#34;control&#34;</span><span class="p">,</span> <span class="s">&#34;treated&#34;</span><span class="p">),</span> <span class="n">each</span> <span class="o">=</span> <span class="m">3</span><span class="p">),</span>
           <span class="n">type</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;a&#34;</span><span class="p">,</span> <span class="s">&#34;p&#34;</span><span class="p">),</span> <span class="n">legend</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="c1">#Boxplot</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">dTrack</span><span class="p">,</span> <span class="n">groups</span> <span class="o">=</span> <span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&#34;control&#34;</span><span class="p">,</span> <span class="s">&#34;treated&#34;</span><span class="p">),</span> <span class="n">each</span> <span class="o">=</span> <span class="m">3</span><span class="p">),</span>
           <span class="n">type</span> <span class="o">=</span> <span class="s">&#34;boxplot&#34;</span><span class="p">)</span>
<span class="c1">#Aggregate group. aggregation can be  mean, median, extreme,</span>
<span class="c1">#sum, min and max</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">dTrack</span><span class="p">,</span> <span class="n">groups</span> <span class="o">=</span> <span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&#34;control&#34;</span><span class="p">,</span> <span class="s">&#34;treated&#34;</span><span class="p">),</span><span class="n">each</span> <span class="o">=</span> <span class="m">3</span><span class="p">),</span>
           <span class="n">type</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;b&#34;</span><span class="p">),</span> <span class="n">aggregateGroups</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span>
           <span class="n">aggregation</span> <span class="o">=</span> <span class="s">&#34;max&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p19.png" alt="p19"></p>
<h3 id="building-datatrack-objects-from-files">Building DataTrack objects from files</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span><span class="lnt">9
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">bgFile</span> <span class="o">&lt;-</span> <span class="nf">system.file</span><span class="p">(</span><span class="s">&#34;extdata/test.bedGraph&#34;</span><span class="p">,</span> <span class="n">package</span> <span class="o">=</span> <span class="s">&#34;Gviz&#34;</span><span class="p">)</span>
<span class="n">dTrack2</span> <span class="o">&lt;-</span> <span class="nf">DataTrack</span><span class="p">(</span><span class="n">range</span> <span class="o">=</span> <span class="n">bgFile</span><span class="p">,</span> <span class="n">genome</span> <span class="o">=</span> <span class="s">&#34;hg19&#34;</span><span class="p">,</span>
                     <span class="n">type</span> <span class="o">=</span> <span class="s">&#34;l&#34;</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="s">&#34;chr19&#34;</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;bedGraph&#34;</span><span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">dTrack2</span><span class="p">)</span>

<span class="n">bamFile</span> <span class="o">&lt;-</span> <span class="nf">system.file</span><span class="p">(</span><span class="s">&#34;extdata/test.bam&#34;</span><span class="p">,</span> <span class="n">package</span> <span class="o">=</span> <span class="s">&#34;Gviz&#34;</span><span class="p">)</span>
<span class="n">dTrack4</span> <span class="o">&lt;-</span> <span class="nf">DataTrack</span><span class="p">(</span><span class="n">range</span> <span class="o">=</span> <span class="n">bamFile</span><span class="p">,</span> <span class="n">genome</span> <span class="o">=</span> <span class="s">&#34;hg19&#34;</span><span class="p">,</span>
                     <span class="n">type</span> <span class="o">=</span> <span class="s">&#34;l&#34;</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;Coverage&#34;</span><span class="p">,</span> <span class="n">window</span> <span class="o">=</span> <span class="m">-1</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="s">&#34;chr1&#34;</span><span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">dTrack4</span><span class="p">,</span> <span class="n">from</span> <span class="o">=</span> <span class="m">189990000</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="m">190000000</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p20new.png" alt="p20"></p>
<h3 id="annotationtrack">AnnotationTrack</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">aTrack</span> <span class="o">&lt;-</span> <span class="nf">AnnotationTrack</span><span class="p">(</span><span class="n">start</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">10</span><span class="p">,</span> <span class="m">40</span><span class="p">,</span> <span class="m">120</span><span class="p">),</span>
                          <span class="n">width</span> <span class="o">=</span> <span class="m">15</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="s">&#34;chrX&#34;</span><span class="p">,</span> <span class="n">strand</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;+&#34;</span><span class="p">,</span><span class="s">&#34;*&#34;</span><span class="p">,</span> <span class="s">&#34;-&#34;</span><span class="p">),</span>
                          <span class="n">id</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;Huey&#34;</span><span class="p">,</span> <span class="s">&#34;Dewey&#34;</span><span class="p">,</span> <span class="s">&#34;Louie&#34;</span><span class="p">),</span>
                          <span class="n">genome</span> <span class="o">=</span> <span class="s">&#34;hg19&#34;</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;foo&#34;</span><span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">aTrack</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p21.png" alt="p21"></p>
<h3 id="add-id-information">Add id information</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">plotTracks</span><span class="p">(</span><span class="n">aTrack</span><span class="p">,</span> <span class="n">shape</span> <span class="o">=</span> <span class="s">&#34;box&#34;</span><span class="p">,</span> <span class="n">featureAnnotation</span> <span class="o">=</span> <span class="s">&#34;id&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p210.png" alt="p210"></p>
<h3 id="change-fontcolor">change fontcolor</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">plotTracks</span><span class="p">(</span><span class="n">aTrack</span><span class="p">,</span> <span class="n">shape</span> <span class="o">=</span> <span class="s">&#34;ellipse&#34;</span><span class="p">,</span> <span class="n">featureAnnotation</span> <span class="o">=</span> <span class="s">&#34;id&#34;</span><span class="p">,</span> 
          <span class="n">fontcolor.feature</span> <span class="o">=</span> <span class="s">&#34;darkblue&#34;</span><span class="p">)</span>
<span class="c1">### Building AnnotationTrack objects from files</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p211.png" alt="p211"></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#Annotation track</span>
<span class="n">aTrack2</span> <span class="o">&lt;-</span> <span class="nf">AnnotationTrack</span><span class="p">(</span><span class="n">range</span> <span class="o">=</span> <span class="n">bamFile</span><span class="p">,</span> <span class="n">genome</span> <span class="o">=</span> <span class="s">&#34;hg19&#34;</span><span class="p">,</span>
                           <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;Reads&#34;</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="s">&#34;chr1&#34;</span><span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">aTrack2</span><span class="p">,</span> <span class="n">from</span> <span class="o">=</span> <span class="m">189995000</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="m">190000000</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p22.png" alt="p22"></p>
<h4 id="plot-both-the-datatrack-representation-as-well-as-the-annotationtrack-representation-of-the-bam-file-together">plot both the DataTrack representation as well as the AnnotationTrack representation of the bam file together</h4>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r">
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">dTrack4</span><span class="p">,</span> <span class="n">aTrack2</span><span class="p">),</span> <span class="n">from</span> <span class="o">=</span> <span class="m">189990000</span><span class="p">,</span>
           <span class="n">to</span> <span class="o">=</span> <span class="m">190000000</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p23.png" alt="p23"></p>
<h3 id="generegiontrack">GeneRegionTrack</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">data</span><span class="p">(</span><span class="n">geneModels</span><span class="p">)</span>
<span class="n">grtrack</span> <span class="o">&lt;-</span> <span class="nf">GeneRegionTrack</span><span class="p">(</span><span class="n">geneModels</span><span class="p">,</span> <span class="n">genome</span> <span class="o">=</span> <span class="n">gen</span><span class="p">,</span>
                           <span class="n">chromosome</span> <span class="o">=</span> <span class="n">chr</span><span class="p">,</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;foo&#34;</span><span class="p">,</span> 
                           <span class="n">transcriptAnnotation</span> <span class="o">=</span> <span class="s">&#34;symbol&#34;</span><span class="p">)</span>

</code></pre></td></tr></table>
</div>
</div><h3 id="building-generegiontrack-objects-from-transcriptdbs">Building GeneRegionTrack objects from TranscriptDbs</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">library</span><span class="p">(</span><span class="n">GenomicFeatures</span><span class="p">)</span>
<span class="n">samplefile</span> <span class="o">&lt;-</span> <span class="nf">system.file</span><span class="p">(</span><span class="s">&#34;extdata/&#34;</span><span class="p">,</span> <span class="s">&#34;UCSC_knownGene_sample.sqlite&#34;</span><span class="p">,</span>
                          <span class="n">package</span> <span class="o">=</span> <span class="s">&#34;GenomicFeatures&#34;</span><span class="p">)</span>
<span class="n">txdb</span> <span class="o">&lt;-</span> <span class="nf">loadDb</span><span class="p">(</span><span class="n">samplefile</span><span class="p">,</span><span class="n">packageName</span> <span class="o">=</span> <span class="s">&#34;GenomicFeatures&#34;</span><span class="p">)</span>
<span class="n">txTr</span> <span class="o">&lt;-</span> <span class="nf">GeneRegionTrack</span><span class="p">(</span><span class="n">txdb</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="s">&#34;chr6&#34;</span><span class="p">,</span> <span class="n">start</span> <span class="o">=</span> <span class="m">300000</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="m">350000</span><span class="p">)</span>
<span class="c1">#feature(txTr)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">txTr</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><h3 id="biomartgeneregiontrack">BiomartGeneRegionTrack</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">biomTrack</span> <span class="o">&lt;-</span> <span class="nf">BiomartGeneRegionTrack</span><span class="p">(</span><span class="n">genome</span> <span class="o">=</span> <span class="s">&#34;hg19&#34;</span><span class="p">,</span>
                                    <span class="n">chromosome</span> <span class="o">=</span> <span class="n">chr</span><span class="p">,</span> <span class="n">start</span> <span class="o">=</span> <span class="m">20000000</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span>
                                      <span class="m">21000000</span><span class="p">,</span><span class="n">name</span> <span class="o">=</span> <span class="s">&#34;ENSEMBL&#34;</span><span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">biomTrack</span><span class="p">,</span> <span class="n">col.line</span> <span class="o">=</span> <span class="kc">NULL</span><span class="p">,</span> <span class="n">col</span> <span class="o">=</span> <span class="kc">NULL</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p24.png" alt="p24"></p>
<h3 id="sequence-track">Sequence Track</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">library</span><span class="p">(</span><span class="n">BSgenome.Hsapiens.UCSC.hg19</span><span class="p">)</span>
<span class="n">sTrack</span> <span class="o">&lt;-</span> <span class="nf">SequenceTrack</span><span class="p">(</span><span class="n">Hsapiens</span><span class="p">)</span>
<span class="c1">#sequence track : add 5&#39;-&gt;3&#39;</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">sTrack</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="m">1</span><span class="p">,</span> <span class="n">from</span> <span class="o">=</span> <span class="m">20000</span><span class="p">,</span><span class="n">to</span> <span class="o">=</span> <span class="m">20050</span><span class="p">,</span>
           <span class="n">add53</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="c1">#The complement</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="n">sTrack</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="m">1</span><span class="p">,</span> <span class="n">from</span> <span class="o">=</span> <span class="m">20000</span><span class="p">,</span><span class="n">to</span> <span class="o">=</span> <span class="m">20050</span><span class="p">,</span>
           <span class="n">add53</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> <span class="n">complement</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p25.png" alt="p25"></p>
<h3 id="alignmentstrack">AlignmentsTrack</h3>
<h4 id="rnaseq-experiment">RNAseq experiment</h4>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">afrom</span><span class="o">=</span><span class="m">2960000</span>
<span class="n">ato</span><span class="o">=</span><span class="m">3160000</span>
<span class="c1">#bam file</span>
<span class="n">alTrack</span> <span class="o">&lt;-</span> <span class="nf">AlignmentsTrack</span><span class="p">(</span><span class="nf">system.file</span><span class="p">(</span><span class="n">package</span> <span class="o">=</span> <span class="s">&#34;Gviz&#34;</span><span class="p">,</span> <span class="s">&#34;extdata&#34;</span><span class="p">,</span> <span class="s">&#34;gapped.bam&#34;</span><span class="p">),</span> <span class="n">isPaired</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="n">bmt</span> <span class="o">&lt;-</span> <span class="nf">BiomartGeneRegionTrack</span><span class="p">(</span><span class="n">genome</span> <span class="o">=</span> <span class="s">&#34;hg19&#34;</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="s">&#34;chr12&#34;</span><span class="p">,</span>
                              <span class="n">start</span> <span class="o">=</span> <span class="n">afrom</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="n">ato</span><span class="p">,</span> <span class="n">filter</span> <span class="o">=</span> <span class="nf">list</span><span class="p">(</span><span class="n">with_ox_refseq_mrna</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">),</span>
                              <span class="n">stacking</span> <span class="o">=</span> <span class="s">&#34;dense&#34;</span><span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">bmt</span><span class="p">,</span> <span class="n">alTrack</span><span class="p">),</span> <span class="n">from</span> <span class="o">=</span> <span class="n">afrom</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="n">ato</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="s">&#34;chr12&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p26.png" alt="p26"></p>
<ul>
<li>To reduce the size of the coverage section by setting the coverageHeight or the minCoverageHeight parameters.</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">plotTracks</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">bmt</span><span class="p">,</span> <span class="n">alTrack</span><span class="p">),</span> <span class="n">from</span> <span class="o">=</span> <span class="n">afrom</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="n">ato</span><span class="p">,</span>
           <span class="n">chromosome</span> <span class="o">=</span> <span class="s">&#34;chr12&#34;</span><span class="p">,</span> <span class="n">min.height</span> <span class="o">=</span> <span class="m">0</span><span class="p">,</span> <span class="n">coverageHeight</span> <span class="o">=</span> <span class="m">0.08</span><span class="p">,</span>
           <span class="n">minCoverageHeight</span> <span class="o">=</span> <span class="m">0</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p27.png" alt="p27"></p>
<ul>
<li>Turn off the pile-up view of the individual reads by setting the type display parameter.</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">plotTracks</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">alTrack</span><span class="p">,</span> <span class="n">bmt</span><span class="p">),</span> <span class="n">from</span> <span class="o">=</span> <span class="n">afrom</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="n">ato</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="s">&#34;chr12&#34;</span><span class="p">,</span> <span class="n">type</span> <span class="o">=</span> <span class="s">&#34;coverage&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p28.png" alt="p28"></p>
<h3 id="zoom-in-a-bit-further-to-check-out-the-details-of-the-pile-ups-section">Zoom in a bit further to check out the details of the pile-ups section</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">plotTracks</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">bmt</span><span class="p">,</span> <span class="n">alTrack</span><span class="p">),</span> <span class="n">from</span> <span class="o">=</span> <span class="n">afrom</span> <span class="o">+</span> <span class="m">12700</span><span class="p">,</span>
           <span class="n">to</span> <span class="o">=</span> <span class="n">afrom</span> <span class="o">+</span> <span class="m">15200</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="s">&#34;chr12&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p29.png" alt="p29"></p>
<ul>
<li>in single end mode by setting the isPaired argument in the constructor.</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">alTrack</span> <span class="o">&lt;-</span> <span class="nf">AlignmentsTrack</span><span class="p">(</span><span class="nf">system.file</span><span class="p">(</span><span class="n">package</span> <span class="o">=</span> <span class="s">&#34;Gviz&#34;</span><span class="p">,</span>
                                       <span class="s">&#34;extdata&#34;</span><span class="p">,</span> <span class="s">&#34;gapped.bam&#34;</span><span class="p">),</span> <span class="n">isPaired</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">bmt</span><span class="p">,</span> <span class="n">alTrack</span><span class="p">),</span> <span class="n">from</span> <span class="o">=</span> <span class="n">afrom</span> <span class="o">+</span> <span class="m">12700</span><span class="p">,</span>
           <span class="n">to</span> <span class="o">=</span> <span class="n">afrom</span> <span class="o">+</span> <span class="m">15200</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="s">&#34;chr12&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p30.png" alt="p30"></p>
<h3 id="dnaseq-experiment">DNAseq experiment</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">afrom</span> <span class="o">&lt;-</span> <span class="m">44945200</span>
<span class="n">ato</span> <span class="o">&lt;-</span> <span class="m">44947200</span>
<span class="n">alTrack</span> <span class="o">&lt;-</span> <span class="nf">AlignmentsTrack</span><span class="p">(</span><span class="nf">system.file</span><span class="p">(</span><span class="n">package</span> <span class="o">=</span> <span class="s">&#34;Gviz&#34;</span><span class="p">,</span><span class="s">&#34;extdata&#34;</span><span class="p">,</span> <span class="s">&#34;snps.bam&#34;</span><span class="p">),</span> <span class="n">isPaired</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">alTrack</span><span class="p">,</span> <span class="n">sTrack</span><span class="p">),</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="s">&#34;chr21&#34;</span><span class="p">,</span> <span class="n">from</span> <span class="o">=</span> <span class="n">afrom</span><span class="p">,</span><span class="n">to</span> <span class="o">=</span> <span class="n">ato</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p31.png" alt="p31"></p>
<ul>
<li>Zoom in to one of the obvious heterozygous SNP positions</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">plotTracks</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">alTrack</span><span class="p">,</span> <span class="n">sTrack</span><span class="p">),</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="s">&#34;chr21&#34;</span><span class="p">,</span> <span class="n">from</span> <span class="o">=</span> <span class="m">44946590</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="m">44946660</span><span class="p">)</span>
<span class="c1">#show individual letters</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">alTrack</span><span class="p">,</span> <span class="n">sTrack</span><span class="p">),</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="s">&#34;chr21&#34;</span><span class="p">,</span>
           <span class="n">from</span> <span class="o">=</span> <span class="m">44946590</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="m">44946660</span><span class="p">,</span> <span class="n">cex</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> <span class="n">min.height</span> <span class="o">=</span> <span class="m">8</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p32.png" alt="p32"></p>
<h2 id="track-highlighting-and-overlays">Track highlighting and overlays</h2>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#highlight</span>
<span class="n">ht</span> <span class="o">&lt;-</span> <span class="nf">HighlightTrack</span><span class="p">(</span><span class="n">trackList</span> <span class="o">=</span> <span class="nf">list</span><span class="p">(</span><span class="n">atrack</span><span class="p">,</span> <span class="n">grtrack</span><span class="p">,</span><span class="n">dtrack</span><span class="p">),</span>
                     <span class="n">start</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">26705000</span><span class="p">,</span> <span class="m">26720000</span><span class="p">),</span> <span class="n">width</span> <span class="o">=</span> <span class="m">7000</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="m">7</span><span class="p">)</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span> <span class="n">gtrack</span><span class="p">,</span> <span class="n">ht</span><span class="p">),</span> <span class="n">from</span> <span class="o">=</span> <span class="n">lim[1]</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="n">lim[2]</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p33.png" alt="p33"></p>
<h3 id="overlays-multiple-tracks-on-the-same-area-of-the-plot">overlays multiple tracks on the same area of the plot</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#create data</span>
<span class="n">dat</span> <span class="o">&lt;-</span> <span class="nf">runif</span><span class="p">(</span><span class="m">100</span><span class="p">,</span> <span class="n">min</span> <span class="o">=</span> <span class="m">-2</span><span class="p">,</span> <span class="n">max</span> <span class="o">=</span> <span class="m">22</span><span class="p">)</span>
<span class="n">dtrack2</span> <span class="o">&lt;-</span> <span class="nf">DataTrack</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">dat</span><span class="p">,</span> <span class="n">start</span> <span class="o">=</span> <span class="n">coords[</span><span class="o">-</span><span class="nf">length</span><span class="p">(</span><span class="n">coords</span><span class="p">)</span><span class="n">]</span><span class="p">,</span>
                     <span class="n">end</span> <span class="o">=</span> <span class="n">coords[</span><span class="m">-1</span><span class="n">]</span><span class="p">,</span> <span class="n">chromosome</span> <span class="o">=</span> <span class="n">chr</span><span class="p">,</span> <span class="n">genome</span> <span class="o">=</span> <span class="n">gen</span><span class="p">,</span>
                     <span class="n">name</span> <span class="o">=</span> <span class="s">&#34;Uniform2&#34;</span><span class="p">,</span> <span class="n">groups</span> <span class="o">=</span> <span class="nf">factor</span><span class="p">(</span><span class="s">&#34;sample 2&#34;</span><span class="p">,</span><span class="n">levels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;sample 1&#34;</span><span class="p">,</span> <span class="s">&#34;sample 2&#34;</span><span class="p">)),</span>
                     <span class="n">legend</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="nf">displayPars</span><span class="p">(</span><span class="n">dtrack</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">list</span><span class="p">(</span><span class="n">groups</span> <span class="o">=</span> <span class="nf">factor</span><span class="p">(</span><span class="s">&#34;sample 1&#34;</span><span class="p">,</span><span class="n">levels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;sample 1&#34;</span><span class="p">,</span> <span class="s">&#34;sample 2&#34;</span><span class="p">)),</span> <span class="n">legend</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="n">ot</span> <span class="o">&lt;-</span> <span class="nf">OverlayTrack</span><span class="p">(</span><span class="n">trackList</span> <span class="o">=</span> <span class="nf">list</span><span class="p">(</span><span class="n">dtrack2</span><span class="p">,</span> <span class="n">dtrack</span><span class="p">))</span>
<span class="n">ylims</span> <span class="o">&lt;-</span> <span class="nf">extendrange</span><span class="p">(</span><span class="nf">range</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="nf">values</span><span class="p">(</span><span class="n">dtrack</span><span class="p">),</span> <span class="nf">values</span><span class="p">(</span><span class="n">dtrack2</span><span class="p">))))</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span> <span class="n">gtrack</span><span class="p">,</span> <span class="n">ot</span><span class="p">),</span> <span class="n">from</span> <span class="o">=</span> <span class="n">lim[1]</span><span class="p">,</span> <span class="n">to</span> <span class="o">=</span> <span class="n">lim[2]</span><span class="p">,</span> <span class="n">ylim</span> <span class="o">=</span> <span class="n">ylims</span><span class="p">,</span> <span class="n">type</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;smooth&#34;</span><span class="p">,</span> <span class="s">&#34;p&#34;</span><span class="p">))</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p34.png" alt="p34"></p>
<ul>
<li>alpha blending can be a useful tool to tease out even more information out of track overlays</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">displayPars</span><span class="p">(</span><span class="n">dtrack</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">list</span><span class="p">(</span><span class="n">alpha.title</span> <span class="o">=</span> <span class="m">1</span><span class="p">,</span> <span class="n">alpha</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">)</span>
<span class="nf">displayPars</span><span class="p">(</span><span class="n">dtrack2</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">list</span><span class="p">(</span><span class="n">alpha.title</span> <span class="o">=</span> <span class="m">1</span><span class="p">,</span> <span class="n">alpha</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">)</span>
<span class="n">ot</span> <span class="o">&lt;-</span> <span class="nf">OverlayTrack</span><span class="p">(</span><span class="n">trackList</span> <span class="o">=</span> <span class="nf">list</span><span class="p">(</span><span class="n">dtrack</span><span class="p">,</span> <span class="n">dtrack2</span><span class="p">))</span>
<span class="nf">plotTracks</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">itrack</span><span class="p">,</span> <span class="n">gtrack</span><span class="p">,</span> <span class="n">ot</span><span class="p">),</span> <span class="n">from</span> <span class="o">=</span> <span class="n">lim[1]</span><span class="p">,</span>
           <span class="n">to</span> <span class="o">=</span> <span class="n">lim[2]</span><span class="p">,</span> <span class="n">ylim</span> <span class="o">=</span> <span class="n">ylims</span><span class="p">,</span> <span class="n">type</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;hist&#34;</span><span class="p">),</span> <span class="n">window</span> <span class="o">=</span> <span class="m">30</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p>output
<img src="p35.png" alt="p35"></p>
<p>Attach is the <a href="Gviz2013.Rmd">.Rmd</a> file.</p>
    </div>

    
    


    
    

    <footer class="post-footer">
      <div class="post-tags">
          <a href="/tags/r/">R</a>
          <a href="/tags/bioinformatics/">Bioinformatics</a>
          <a href="/tags/ucsc/">UCSC</a>
          <a href="/tags/r-package/">R Package</a>
          
        </div>

      
      <nav class="post-nav">
        
          <a class="prev" href="/post/ucscgenebrowser2/">
            
            <i class="iconfont">
              <svg  class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="18" height="18">
  <path d="M691.908486 949.511495l75.369571-89.491197c10.963703-12.998035 10.285251-32.864502-1.499144-44.378743L479.499795 515.267417 757.434875 204.940602c11.338233-12.190647 11.035334-32.285311-0.638543-44.850487l-80.46666-86.564541c-11.680017-12.583596-30.356378-12.893658-41.662889-0.716314L257.233596 494.235404c-11.332093 12.183484-11.041474 32.266891 0.657986 44.844348l80.46666 86.564541c1.772366 1.910513 3.706415 3.533476 5.750981 4.877077l306.620399 321.703933C662.505829 963.726242 680.945807 962.528973 691.908486 949.511495z"></path>
</svg>

            </i>
            <span class="prev-text nav-default">UCSC gene browser with R package Gviz - part 2</span>
            <span class="prev-text nav-mobile">Prev</span>
          </a>
        
          <a class="next" href="/post/bioscirep/">
            <span class="next-text nav-default">Bioinformatics analysis of prognostic value of PITX1 gene in breast cancer</span>
            <span class="prev-text nav-mobile">Next</span>
            
            <i class="iconfont">
              <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="18" height="18">
  <path d="M332.091514 74.487481l-75.369571 89.491197c-10.963703 12.998035-10.285251 32.864502 1.499144 44.378743l286.278095 300.375162L266.565125 819.058374c-11.338233 12.190647-11.035334 32.285311 0.638543 44.850487l80.46666 86.564541c11.680017 12.583596 30.356378 12.893658 41.662889 0.716314l377.434212-421.426145c11.332093-12.183484 11.041474-32.266891-0.657986-44.844348l-80.46666-86.564541c-1.772366-1.910513-3.706415-3.533476-5.750981-4.877077L373.270379 71.774697C361.493148 60.273758 343.054193 61.470003 332.091514 74.487481z"></path>
</svg>

            </i>
          </a>
      </nav>
    </footer>
  </article>

  
  

  
  

  

  
  

  

  

  

    

  

        </div>
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="icon-links">
  
  
    <a href="mailto:xiaoni0601@gmail.com" rel="me noopener" class="iconfont"
      title="email" >
      <svg class="icon" viewBox="0 0 1451 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="36" height="36">
  <path d="M664.781909 681.472759 0 97.881301C0 3.997201 71.046997 0 71.046997 0L474.477909 0 961.649408 0 1361.641813 0C1361.641813 0 1432.688811 3.997201 1432.688811 97.881301L771.345323 681.472759C771.345323 681.472759 764.482731 685.154773 753.594283 688.65053L753.594283 688.664858C741.602731 693.493018 729.424896 695.068979 718.077952 694.839748 706.731093 695.068979 694.553173 693.493018 682.561621 688.664858L682.561621 688.65053C671.644501 685.140446 664.781909 681.472759 664.781909 681.472759L664.781909 681.472759ZM718.063616 811.603883C693.779541 811.016482 658.879232 802.205449 619.10784 767.734955 542.989056 701.759633 0 212.052267 0 212.052267L0 942.809523C0 942.809523 0 1024 83.726336 1024L682.532949 1024 753.579947 1024 1348.948139 1024C1432.688811 1024 1432.688811 942.809523 1432.688811 942.809523L1432.688811 212.052267C1432.688811 212.052267 893.138176 701.759633 817.019477 767.734955 777.248 802.205449 742.347691 811.03081 718.063616 811.603883L718.063616 811.603883Z"></path>
</svg>

    </a>
  
    <a href="https://github.com/xiaonilee" rel="me noopener" class="iconfont"
      title="github"  target="_blank"
      >
      <svg class="icon" style="" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="36" height="36">
  <path d="M512 12.672c-282.88 0-512 229.248-512 512 0 226.261333 146.688 418.133333 350.08 485.76 25.6 4.821333 34.986667-11.008 34.986667-24.618667 0-12.16-0.426667-44.373333-0.64-87.04-142.421333 30.890667-172.458667-68.693333-172.458667-68.693333C188.672 770.986667 155.008 755.2 155.008 755.2c-46.378667-31.744 3.584-31.104 3.584-31.104 51.413333 3.584 78.421333 52.736 78.421333 52.736 45.653333 78.293333 119.850667 55.68 149.12 42.581333 4.608-33.109333 17.792-55.68 32.426667-68.48-113.706667-12.8-233.216-56.832-233.216-253.013333 0-55.893333 19.84-101.546667 52.693333-137.386667-5.76-12.928-23.04-64.981333 4.48-135.509333 0 0 42.88-13.738667 140.8 52.48 40.96-11.392 84.48-17.024 128-17.28 43.52 0.256 87.04 5.888 128 17.28 97.28-66.218667 140.16-52.48 140.16-52.48 27.52 70.528 10.24 122.581333 5.12 135.509333 32.64 35.84 52.48 81.493333 52.48 137.386667 0 196.693333-119.68 240-233.6 252.586667 17.92 15.36 34.56 46.762667 34.56 94.72 0 68.522667-0.64 123.562667-0.64 140.202666 0 13.44 8.96 29.44 35.2 24.32C877.44 942.592 1024 750.592 1024 524.672c0-282.752-229.248-512-512-512"></path>
</svg>

    </a>


<a href="/index.xml" rel="noopener alternate" type="application/rss&#43;xml"
    class="iconfont" title="rss" target="_blank">
    <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="30" height="30">
  <path d="M819.157333 1024C819.157333 574.592 449.408 204.8 0 204.8V0c561.706667 0 1024 462.293333 1024 1024h-204.842667zM140.416 743.04a140.8 140.8 0 0 1 140.501333 140.586667A140.928 140.928 0 0 1 140.074667 1024C62.72 1024 0 961.109333 0 883.626667s62.933333-140.544 140.416-140.586667zM678.784 1024h-199.04c0-263.210667-216.533333-479.786667-479.744-479.786667V345.173333c372.352 0 678.784 306.517333 678.784 678.826667z"></path>
</svg>

  </a>
   
</div>

<div class="copyright">

  <span class="copyright-year">
    &copy;
    
      2020 -
    2021
    <span class="heart">
      
      <i class="iconfont">
        <svg class="icon" viewBox="0 0 1025 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="14" height="14">
  <path d="M1000.1 247.9c-15.5-37.3-37.6-70.6-65.7-98.9-54.4-54.8-125.8-85-201-85-85.7 0-166 39-221.4 107.4C456.6 103 376.3 64 290.6 64c-75.1 0-146.5 30.4-201.1 85.6-28.2 28.5-50.4 61.9-65.8 99.3-16 38.8-24 79.9-23.6 122.2 0.7 91.7 40.1 177.2 108.1 234.8 3.1 2.6 6 5.1 8.9 7.8 14.9 13.4 58 52.8 112.6 102.7 93.5 85.5 209.9 191.9 257.5 234.2 7 6.1 15.8 9.5 24.9 9.5 9.2 0 18.1-3.4 24.9-9.5 34.5-30.7 105.8-95.9 181.4-165 74.2-67.8 150.9-138 195.8-178.2 69.5-57.9 109.6-144.4 109.9-237.3 0.1-42.5-8-83.6-24-122.2z"
   fill="#8a8a8a"></path>
</svg>

      </i>
    </span><span class="author">
        xiaonilee
        
      </span></span>

  
  

  
</div>

    </footer>

    <div class="back-to-top" id="back-to-top">
      <i class="iconfont">
        
        <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="35" height="35">
  <path d="M510.866688 227.694839 95.449397 629.218702l235.761562 0-2.057869 328.796468 362.40389 0L691.55698 628.188232l241.942331-3.089361L510.866688 227.694839zM63.840492 63.962777l894.052392 0 0 131.813095L63.840492 195.775872 63.840492 63.962777 63.840492 63.962777zM63.840492 63.962777"></path>
</svg>

      </i>
    </div>
  </div>
  
<script type="text/javascript" src="/lib/jquery/jquery-3.2.1.min.js"></script>
  <script type="text/javascript" src="/lib/slideout/slideout-1.0.1.min.js"></script>




<script type="text/javascript" src="/js/main.638251f4230630f0335d8c6748e53a96f94b72670920b60c09a56fdc8bece214.js" integrity="sha256-Y4JR9CMGMPAzXYxnSOU6lvlLcmcJILYMCaVv3Ivs4hQ=" crossorigin="anonymous"></script>



  <script type="text/javascript">
    window.MathJax = {
      showProcessingMessages: false,
      messageStyle: 'none'
    };
  </script>
  <script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-MML-AM_CHTML' async></script>









  
    <script type="text/javascript" src="/js/load-photoswipe.js"></script>
    <script type="text/javascript" src="/lib/photoswipe/photoswipe.min.js"></script>
    <script type="text/javascript" src="/lib/photoswipe/photoswipe-ui-default.min.js"></script>
  















</body>
</html>
