<!DOCTYPE html>
<html lang="en" itemscope itemtype="http://schema.org/WebPage">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Analysis, visualization, and integration of spatial datasets with Seurat - xiaoni&#39;s blog</title>
  

<meta name="renderer" content="webkit" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>

<meta name="MobileOptimized" content="width"/>
<meta name="HandheldFriendly" content="true"/>


<meta name="applicable-device" content="pc,mobile">

<meta name="theme-color" content="#f8f5ec" />
<meta name="msapplication-navbutton-color" content="#f8f5ec">
<meta name="apple-mobile-web-app-capable" content="yes">
<meta name="apple-mobile-web-app-status-bar-style" content="#f8f5ec">

<meta name="mobile-web-app-capable" content="yes">

<meta name="author" content="Xiaoni" />
  <meta name="description" content="how to use Seurat to analyze spatially-resolved RNA-seq data?
Herein, the tutorial will cover these tasks:
  Normalization
  Dimensional reduction and clustering
  Detecting spatially-variable features
  Interactive visualization
  Integration with single-cell RNA-seq data
  Working with multiple slices
 " />

  <meta name="keywords" content="xiaonili, xiaonilee, xiaoni" />






<meta name="generator" content="Hugo 0.79.0" />


<link rel="canonical" href="/post/seurat3/" />





<link rel="icon" href="/favicon.ico" />











<link rel="stylesheet" href="/sass/jane.min.f1e506a781bf25d33ffc18aa6b4e972a965c58049d27d4f92b7db2e9bf28e4bf.css" integrity="sha256-8eUGp4G/JdM//Biqa06XKpZcWASdJ9T5K32y6b8o5L8=" media="screen" crossorigin="anonymous">



<link href="https://cdn.bootcdn.net/ajax/libs/font-awesome/5.15.1/css/all.css" rel="stylesheet">
<script src="https://cdn.bootcdn.net/ajax/libs/font-awesome/5.15.1/js/all.js"></script>




<meta property="og:title" content="Analysis, visualization, and integration of spatial datasets with Seurat" />
<meta property="og:description" content="how to use Seurat to analyze spatially-resolved RNA-seq data?
Herein, the tutorial will cover these tasks:


Normalization


Dimensional reduction and clustering


Detecting spatially-variable features


Interactive visualization


Integration with single-cell RNA-seq data


Working with multiple slices

" />
<meta property="og:type" content="article" />
<meta property="og:url" content="/post/seurat3/" />
<meta property="article:published_time" content="2021-02-28T00:00:00+00:00" />
<meta property="article:modified_time" content="2021-03-01T00:00:00+00:00" />
<meta itemprop="name" content="Analysis, visualization, and integration of spatial datasets with Seurat">
<meta itemprop="description" content="how to use Seurat to analyze spatially-resolved RNA-seq data?
Herein, the tutorial will cover these tasks:


Normalization


Dimensional reduction and clustering


Detecting spatially-variable features


Interactive visualization


Integration with single-cell RNA-seq data


Working with multiple slices

">
<meta itemprop="datePublished" content="2021-02-28T00:00:00+00:00" />
<meta itemprop="dateModified" content="2021-03-01T00:00:00+00:00" />
<meta itemprop="wordCount" content="1163">



<meta itemprop="keywords" content="Bioinformatics,R package,RNA Sequence,PCA,KNN," />
<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="Analysis, visualization, and integration of spatial datasets with Seurat"/>
<meta name="twitter:description" content="how to use Seurat to analyze spatially-resolved RNA-seq data?
Herein, the tutorial will cover these tasks:


Normalization


Dimensional reduction and clustering


Detecting spatially-variable features


Interactive visualization


Integration with single-cell RNA-seq data


Working with multiple slices

"/>

<!--[if lte IE 9]>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/classlist/1.1.20170427/classList.min.js"></script>
<![endif]-->

<!--[if lt IE 9]>
  <script src="https://cdn.jsdelivr.net/npm/html5shiv@3.7.3/dist/html5shiv.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/respond.js@1.4.2/dest/respond.min.js"></script>
<![endif]-->


<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'UA-177860608-1', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>



</head>
<body>
  <div id="mobile-navbar" class="mobile-navbar">
  <div class="mobile-header-logo">
    <a href="/" class="logo">Xiaoni's Blog</a>
  </div>
  <div class="mobile-navbar-icon">
    <span></span>
    <span></span>
    <span></span>
  </div>
</div>
<nav id="mobile-menu" class="mobile-menu slideout-menu">
  <ul class="mobile-menu-list">
    <li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/">Home</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/post/">Archives</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/categories/">Categories</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/tags/">Tags</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/about/">About Me</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          <div class="mobile-menu-parent">
            <span class="mobile-submenu-open"></span>
            <a href="/">
              docs
            </a>
          </div>
          <ul class="mobile-submenu-list">
            
              <li>
                <a href="/post/tensorflow2/"> Machine Learning: TensorFlow2 Notebook</a>
              </li>
            
              <li>
                <a href="/post/bioscirep/">Bioinformatics analysis of prognostic value of PITX1 gene in breast cancer</a>
              </li>
            
              <li>
                <a href="/post/rinactionmap/">Book: R in Action</a>
              </li>
            
              <li>
                <a href="/post/tumorimmune/">Comprehensive Bioinformatics Analysis of Tumor-Infiltrating Immune Cells</a>
              </li>
            
              <li>
                <a href="/post/aavcapsid/">Deep diversification of an AAV capsid protein by machine learning</a>
              </li>
            
              <li>
                <a href="/post/firebrowse/">FireBrowse Notebook</a>
              </li>
            
              <li>
                <a href="/post/gdc/">GDC Notebook</a>
              </li>
            
              <li>
                <a href="/post/git/">Git - Common commands</a>
              </li>
            
              <li>
                <a href="/post/gscalite/">GSCALite: Gene Set Cancer Analysis</a>
              </li>
            
              <li>
                <a href="/post/frontoncol/">Identification of Therapeutic Targets and Prognostic Biomarkers Among CXC Chemokines in the Renal Cell Carcinoma Microenvironment</a>
              </li>
            
              <li>
                <a href="/post/condacolab/">Installing Conda in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/oncolnc/">OncoLnc: A straightforward Tool</a>
              </li>
            
              <li>
                <a href="/post/webcrawler/">Python Spider</a>
              </li>
            
              <li>
                <a href="/post/rnasequenceanalysis/">RNA-seq Bioinformatics - RNA sequence analysis</a>
              </li>
            
              <li>
                <a href="/post/scrnaknn/">scRNA-sequencing analysis - PCA and KNN</a>
              </li>
            
              <li>
                <a href="/post/survival/">Survival Analysis with data from GSE database</a>
              </li>
            
              <li>
                <a href="/post/rincolab/">The way to use R in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/tissgdb/">TissGDB: tissue-specific gene database in cancer</a>
              </li>
            
              <li>
                <a href="/post/ucscmap/">UCSC Topics</a>
              </li>
            
              <li>
                <a href="/post/webfrontend/">web front end</a>
              </li>
            
              <li>
                <a href="/post/mathpreview/">Math Preview</a>
              </li>
            
          </ul>
        
      </li>
    

    
  </ul>
</nav>


  
    






  <link rel="stylesheet" href="/lib/photoswipe/photoswipe.min.css" />
  <link rel="stylesheet" href="/lib/photoswipe/default-skin/default-skin.min.css" />




<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

<div class="pswp__bg"></div>

<div class="pswp__scroll-wrap">
    
    <div class="pswp__container">
      <div class="pswp__item"></div>
      <div class="pswp__item"></div>
      <div class="pswp__item"></div>
    </div>
    
    <div class="pswp__ui pswp__ui--hidden">
    <div class="pswp__top-bar">
      
      <div class="pswp__counter"></div>
      <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>
      <button class="pswp__button pswp__button--share" title="Share"></button>
      <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>
      <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>
      
      
      <div class="pswp__preloader">
        <div class="pswp__preloader__icn">
          <div class="pswp__preloader__cut">
            <div class="pswp__preloader__donut"></div>
          </div>
        </div>
      </div>
    </div>
    <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
      <div class="pswp__share-tooltip"></div>
    </div>
    <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
    </button>
    <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
    </button>
    <div class="pswp__caption">
      <div class="pswp__caption__center"></div>
    </div>
    </div>
    </div>
</div>

  

  

  

  <header id="header" class="header container">
    <div class="logo-wrapper">
  <a href="/" class="logo">
    
      Xiaoni's Blog
    
  </a>
</div>

<nav class="site-navbar">
  <ul id="menu" class="menu">
    
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/">Home</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/post/">Archives</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/categories/">Categories</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/tags/">Tags</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/about/">About Me</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          <a class="menu-item-link menu-parent" href="/">docs</a>
          <ul class="submenu" >
            
              <li>
                <a href="/post/tensorflow2/"> Machine Learning: TensorFlow2 Notebook</a>
              </li>
            
              <li>
                <a href="/post/bioscirep/">Bioinformatics analysis of prognostic value of PITX1 gene in breast cancer</a>
              </li>
            
              <li>
                <a href="/post/rinactionmap/">Book: R in Action</a>
              </li>
            
              <li>
                <a href="/post/tumorimmune/">Comprehensive Bioinformatics Analysis of Tumor-Infiltrating Immune Cells</a>
              </li>
            
              <li>
                <a href="/post/aavcapsid/">Deep diversification of an AAV capsid protein by machine learning</a>
              </li>
            
              <li>
                <a href="/post/firebrowse/">FireBrowse Notebook</a>
              </li>
            
              <li>
                <a href="/post/gdc/">GDC Notebook</a>
              </li>
            
              <li>
                <a href="/post/git/">Git - Common commands</a>
              </li>
            
              <li>
                <a href="/post/gscalite/">GSCALite: Gene Set Cancer Analysis</a>
              </li>
            
              <li>
                <a href="/post/frontoncol/">Identification of Therapeutic Targets and Prognostic Biomarkers Among CXC Chemokines in the Renal Cell Carcinoma Microenvironment</a>
              </li>
            
              <li>
                <a href="/post/condacolab/">Installing Conda in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/oncolnc/">OncoLnc: A straightforward Tool</a>
              </li>
            
              <li>
                <a href="/post/webcrawler/">Python Spider</a>
              </li>
            
              <li>
                <a href="/post/rnasequenceanalysis/">RNA-seq Bioinformatics - RNA sequence analysis</a>
              </li>
            
              <li>
                <a href="/post/scrnaknn/">scRNA-sequencing analysis - PCA and KNN</a>
              </li>
            
              <li>
                <a href="/post/survival/">Survival Analysis with data from GSE database</a>
              </li>
            
              <li>
                <a href="/post/rincolab/">The way to use R in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/tissgdb/">TissGDB: tissue-specific gene database in cancer</a>
              </li>
            
              <li>
                <a href="/post/ucscmap/">UCSC Topics</a>
              </li>
            
              <li>
                <a href="/post/webfrontend/">web front end</a>
              </li>
            
              <li>
                <a href="/post/mathpreview/">Math Preview</a>
              </li>
            
          </ul>

        

      </li>
    

    
    

    
  </ul>
</nav>

  </header>

  <div id="mobile-panel">
    <main id="main" class="main bg-llight">
      <div class="content-wrapper">
        <div id="content" class="content container">
          <article class="post bg-white">
    
    <header class="post-header">
      <h1 class="post-title">Analysis, visualization, and integration of spatial datasets with Seurat</h1>
      
      <div class="post-meta">
        <time datetime="2021-02-28" class="post-time">
          2021-02-28
        </time>
        <div class="post-category">
            <a href="/categories/bioinformatics/"> Bioinformatics </a>
            <a href="/categories/r-package/"> R package </a>
            <a href="/categories/rna-sequence/"> RNA Sequence </a>
            <a href="/categories/pca/"> PCA </a>
            <a href="/categories/knn/"> KNN </a>
            
          </div>
        

        
        

        
        
      </div>
    </header>

    
    
<div class="post-toc" id="post-toc">
  <h2 class="post-toc-title">Table of Contents</h2>
  <div class="post-toc-content">
    <nav id="TableOfContents">
  <ul>
    <li>
      <ul>
        <li><a href="#material-and-methods">Material and Methods</a></li>
        <li><a href="#load-dataset">Load dataset</a></li>
        <li><a href="#data-preprocessing">Data preprocessing</a></li>
        <li><a href="#gene-expression-visualization">Gene expression visualization</a></li>
        <li><a href="#dimensionality-reduction-clustering-and-visualization">Dimensionality reduction, clustering, and visualization</a></li>
        <li><a href="#interactive-plotting">Interactive plotting</a></li>
        <li><a href="#identification-of-spatially-variable-features">Identification of Spatially Variable Features</a></li>
        <li><a href="#subset-out-anatomical-regions">Subset out anatomical regions</a></li>
        <li><a href="#integration-with-single-cell-data">Integration with single-cell data</a></li>
        <li><a href="#working-with-multiple-slices-in-seurat">Working with multiple slices in Seurat</a></li>
        <li><a href="#slide-seq">Slide-seq</a></li>
      </ul>
    </li>
  </ul>
</nav>
  </div>
</div>

    
    <div class="post-content">
      <p>how to use Seurat to analyze spatially-resolved RNA-seq data?</p>
<p>Herein, the tutorial will cover these tasks:</p>
<ul>
<li>
<p>Normalization</p>
</li>
<li>
<p>Dimensional reduction and clustering</p>
</li>
<li>
<p>Detecting spatially-variable features</p>
</li>
<li>
<p>Interactive visualization</p>
</li>
<li>
<p>Integration with single-cell RNA-seq data</p>
</li>
<li>
<p>Working with multiple slices</p>
</li>
</ul>
<h3 id="material-and-methods">Material and Methods</h3>
<ul>
<li><strong>Dataset</strong>: sagital mouse brain slices generated using the Visium v1 chemistry</li>
</ul>
<h3 id="load-dataset">Load dataset</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">library</span><span class="p">(</span><span class="n">Seurat</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">SeuratData</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">patchwork</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>

<span class="c1">#BiocManager::install(&#34;rhdf5&#34;)</span>
<span class="c1">#BiocManager::install(&#34;rhdf5r&#34;)</span>
<span class="nf">InstallData</span><span class="p">(</span><span class="s">&#34;stxBrain&#34;</span><span class="p">)</span>
<span class="n">brain</span> <span class="o">&lt;-</span> <span class="nf">LoadData</span><span class="p">(</span><span class="s">&#34;stxBrain&#34;</span><span class="p">,</span> <span class="n">type</span> <span class="o">=</span> <span class="s">&#34;anterior1&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><h3 id="data-preprocessing">Data preprocessing</h3>
<ul>
<li>
<p>Raw data performance</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">plot1</span> <span class="o">&lt;-</span> <span class="nf">VlnPlot</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="s">&#34;nCount_Spatial&#34;</span><span class="p">,</span> <span class="n">pt.size</span> <span class="o">=</span> <span class="m">0.1</span><span class="p">)</span> <span class="o">+</span> <span class="nf">NoLegend</span><span class="p">()</span>
<span class="n">plot2</span> <span class="o">&lt;-</span> <span class="nf">SpatialFeaturePlot</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="s">&#34;nCount_Spatial&#34;</span><span class="p">)</span> <span class="o">+</span> <span class="nf">theme</span><span class="p">(</span><span class="n">legend.position</span> <span class="o">=</span> <span class="s">&#34;right&#34;</span><span class="p">)</span>
<span class="nf">wrap_plots</span><span class="p">(</span><span class="n">plot1</span><span class="p">,</span> <span class="n">plot2</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig1.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fig1.png" alt="fig1"></p>
</li>
<li>
<p>The variance in molecular counts / spot can be substantial for spatial datasets, particularly if there are differences in cell density across the tissue.</p>
</li>
<li>
<p>Normalize the data in order to account for variance in sequencing depth across data points</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1"># Normalizes the data, detects high-variance features, and stores the data in the SCT assay.</span>
<span class="n">brain</span> <span class="o">&lt;-</span> <span class="nf">SCTransform</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">assay</span> <span class="o">=</span> <span class="s">&#34;Spatial&#34;</span><span class="p">,</span> <span class="n">verbose</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div></li>
</ul>
<h3 id="gene-expression-visualization">Gene expression visualization</h3>
<ul>
<li>
<p><code>SpatialFeaturePlot()</code> function can overlay molecular data on top of tissue histology.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">SpatialFeaturePlot</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;Hpca&#34;</span><span class="p">,</span> <span class="s">&#34;Ttr&#34;</span><span class="p">))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig2.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fig2.png" alt="fig2"></p>
</li>
<li>
<p>Adjust the size of the spots (and their transparency) to improve the visualization of the histology image.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">p1</span> <span class="o">&lt;-</span> <span class="nf">SpatialFeaturePlot</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="s">&#34;Ttr&#34;</span><span class="p">,</span> <span class="n">pt.size.factor</span> <span class="o">=</span> <span class="m">1</span><span class="p">)</span>

<span class="c1"># to downweight the transparency of points with lower expression with alpha</span>
<span class="n">p2</span> <span class="o">&lt;-</span> <span class="nf">SpatialFeaturePlot</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="s">&#34;Ttr&#34;</span><span class="p">,</span> <span class="n">alpha</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0.1</span><span class="p">,</span> <span class="m">1</span><span class="p">))</span>
<span class="n">p1</span> <span class="o">+</span> <span class="n">p2</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig3.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fig3.png" alt="fig3"></p>
</li>
</ul>
<h3 id="dimensionality-reduction-clustering-and-visualization">Dimensionality reduction, clustering, and visualization</h3>
<ul>
<li>With the same workflow to run PCA and clustering</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">brain</span> <span class="o">&lt;-</span> <span class="nf">RunPCA</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">assay</span> <span class="o">=</span> <span class="s">&#34;SCT&#34;</span><span class="p">,</span> <span class="n">verbose</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="n">brain</span> <span class="o">&lt;-</span> <span class="nf">FindNeighbors</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&#34;pca&#34;</span><span class="p">,</span> <span class="n">dims</span> <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">)</span>
<span class="n">brain</span> <span class="o">&lt;-</span> <span class="nf">FindClusters</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">verbose</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="n">brain</span> <span class="o">&lt;-</span> <span class="nf">RunUMAP</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&#34;pca&#34;</span><span class="p">,</span> <span class="n">dims</span> <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><ul>
<li>
<p>Visualize the results</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#in UMAP space  with DimPlot()</span>
<span class="n">p1</span> <span class="o">&lt;-</span> <span class="nf">DimPlot</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&#34;umap&#34;</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>

<span class="c1">#overlaid on the image with SpatialDimPlot()</span>
<span class="n">p2</span> <span class="o">&lt;-</span> <span class="nf">SpatialDimPlot</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">label.size</span> <span class="o">=</span> <span class="m">3</span><span class="p">)</span>
<span class="n">p1</span> <span class="o">+</span> <span class="n">p2</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fig4.png" alt="fig4"></p>
</li>
<li>
<p>Visualize special voxel belongs to which cluster</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">SpatialDimPlot</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">cells.highlight</span> <span class="o">=</span> <span class="nf">CellsByIdentities</span><span class="p">(</span><span class="n">object</span> <span class="o">=</span> <span class="n">brain</span><span class="p">,</span> <span class="n">idents</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span> <span class="m">1</span><span class="p">,</span> <span class="m">4</span><span class="p">,</span> <span class="m">3</span><span class="p">,</span> 
    <span class="m">5</span><span class="p">,</span> <span class="m">8</span><span class="p">)),</span> <span class="n">facet.highlight</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">3</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig5.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fig5.png" alt="fig5"></p>
</li>
</ul>
<h3 id="interactive-plotting">Interactive plotting</h3>
<ul>
<li>
<p>Both <code>SpatialDimPlot()</code> and <code>SpatialFeaturePlot()</code> now have an interactive parameter, that when set to <code>TRUE</code>, open up the Rstudio viewer pane with an interactive Shiny plot.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span><span class="lnt">9
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1">#with SpatialDimPlot()</span>
<span class="nf">SpatialDimPlot</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">interactive</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>

<span class="c1">#with SpatialFeaturePlot()</span>
<span class="nf">SpatialFeaturePlot</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="s">&#34;Ttr&#34;</span><span class="p">,</span> <span class="n">interactive</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>

<span class="c1">#with LinkedDimPlot()</span>
<span class="nf">LinkedDimPlot</span><span class="p">(</span><span class="n">brain</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig6.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fig6.png" alt="fig6"></p>
</li>
</ul>
<h3 id="identification-of-spatially-variable-features">Identification of Spatially Variable Features</h3>
<ul>
<li>
<p>Seurat offers two workflows to identify molecular features that correlate with spatial location within a tissue</p>
<ul>
<li>
<p>Based on pre-annotated anatomical regions within the tissue, which may be determined either from unsupervised clustering or prior knowledge.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">de_markers</span> <span class="o">&lt;-</span> <span class="nf">FindMarkers</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">ident.1</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span> <span class="n">ident.2</span> <span class="o">=</span> <span class="m">6</span><span class="p">)</span>
<span class="nf">SpatialFeaturePlot</span><span class="p">(</span><span class="n">object</span> <span class="o">=</span> <span class="n">brain</span><span class="p">,</span> 
                <span class="n">features</span> <span class="o">=</span> <span class="nf">rownames</span><span class="p">(</span><span class="n">de_markers</span><span class="p">)</span><span class="n">[1</span><span class="o">:</span><span class="m">3</span><span class="n">]</span><span class="p">,</span> 
                <span class="n">alpha</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0.1</span><span class="p">,</span> <span class="m">1</span><span class="p">),</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">3</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig7.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fig7.png" alt="fig7"></p>
</li>
<li>
<p>An alternative approach, implemented in <code>FindSpatiallyVariables()</code>.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">brain</span> <span class="o">&lt;-</span> <span class="nf">FindSpatiallyVariableFeatures</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">assay</span> <span class="o">=</span> <span class="s">&#34;SCT&#34;</span><span class="p">,</span> 
                                    <span class="n">features</span> <span class="o">=</span> <span class="nf">VariableFeatures</span><span class="p">(</span><span class="n">brain</span><span class="p">)</span><span class="n">[1</span><span class="o">:</span><span class="m">1000</span><span class="n">]</span><span class="p">,</span> 
                                    <span class="n">selection.method</span> <span class="o">=</span> <span class="s">&#34;markvariogram&#34;</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig8.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div></li>
</ul>
</li>
<li>
<p>Visualize the expression of the top 6 features identified by this measure.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">top.features</span> <span class="o">&lt;-</span> <span class="nf">head</span><span class="p">(</span><span class="nf">SpatiallyVariableFeatures</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">selection.method</span> <span class="o">=</span> <span class="s">&#34;markvariogram&#34;</span><span class="p">),</span> <span class="m">6</span><span class="p">)</span>
<span class="nf">SpatialFeaturePlot</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="n">top.features</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">3</span><span class="p">,</span> <span class="n">alpha</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0.1</span><span class="p">,</span> <span class="m">1</span><span class="p">))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig9.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div></li>
</ul>
<h3 id="subset-out-anatomical-regions">Subset out anatomical regions</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">cortex</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">idents</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span> <span class="m">2</span><span class="p">,</span> <span class="m">3</span><span class="p">,</span> <span class="m">4</span><span class="p">,</span> <span class="m">6</span><span class="p">,</span> <span class="m">7</span><span class="p">))</span>
<span class="c1"># now remove additional cells, use SpatialDimPlots to visualize what to remove</span>
<span class="c1"># SpatialDimPlot(cortex,cells.highlight = WhichCells(cortex, expression = image_imagerow &gt; 400 |</span>
<span class="c1"># image_imagecol &lt; 150))</span>
<span class="n">cortex</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">cortex</span><span class="p">,</span> <span class="n">anterior1_imagerow</span> <span class="o">&gt;</span> <span class="m">400</span> <span class="o">|</span> <span class="n">anterior1_imagecol</span> <span class="o">&lt;</span> <span class="m">150</span><span class="p">,</span> <span class="n">invert</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="n">cortex</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">cortex</span><span class="p">,</span> <span class="n">anterior1_imagerow</span> <span class="o">&gt;</span> <span class="m">275</span> <span class="o">&amp;</span> <span class="n">anterior1_imagecol</span> <span class="o">&gt;</span> <span class="m">370</span><span class="p">,</span> <span class="n">invert</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="n">cortex</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">cortex</span><span class="p">,</span> <span class="n">anterior1_imagerow</span> <span class="o">&gt;</span> <span class="m">250</span> <span class="o">&amp;</span> <span class="n">anterior1_imagecol</span> <span class="o">&gt;</span> <span class="m">440</span><span class="p">,</span> <span class="n">invert</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>

<span class="n">p1</span> <span class="o">&lt;-</span> <span class="nf">SpatialDimPlot</span><span class="p">(</span><span class="n">cortex</span><span class="p">,</span> <span class="n">crop</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="n">p2</span> <span class="o">&lt;-</span> <span class="nf">SpatialDimPlot</span><span class="p">(</span><span class="n">cortex</span><span class="p">,</span> <span class="n">crop</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">pt.size.factor</span> <span class="o">=</span> <span class="m">1</span><span class="p">,</span> <span class="n">label.size</span> <span class="o">=</span> <span class="m">3</span><span class="p">)</span>
<span class="n">p1</span> <span class="o">+</span> <span class="n">p2</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig10.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fig10.png" alt="fig10"></p>
<h3 id="integration-with-single-cell-data">Integration with single-cell data</h3>
<ul>
<li>
<p>Download <a href="https://www.dropbox.com/s/cuowvm4vrf65pvq/allen_cortex.rds?dl=1">data</a></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">allen_reference</span> <span class="o">&lt;-</span> <span class="nf">readRDS</span><span class="p">(</span><span class="s">&#34;/Users/xiaonili/Downloads/allen_cortex.rds&#34;</span><span class="p">)</span>
<span class="c1"># note that setting ncells=3000 normalizes the full dataset but learns noise models on 3k cells</span>
<span class="c1"># this speeds up SCTransform dramatically with no loss in performance</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="n">allen_reference</span> <span class="o">&lt;-</span> <span class="nf">SCTransform</span><span class="p">(</span><span class="n">allen_reference</span><span class="p">,</span> <span class="n">ncells</span> <span class="o">=</span> <span class="m">3000</span><span class="p">,</span> <span class="n">verbose</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">RunPCA</span><span class="p">(</span><span class="n">verbose</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span> <span class="o">%&gt;%</span> 
<span class="nf">RunUMAP</span><span class="p">(</span><span class="n">dims</span> <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">)</span>

<span class="c1"># After subsetting, we renormalize cortex</span>
<span class="n">cortex</span> <span class="o">&lt;-</span> <span class="nf">SCTransform</span><span class="p">(</span><span class="n">cortex</span><span class="p">,</span> <span class="n">assay</span> <span class="o">=</span> <span class="s">&#34;Spatial&#34;</span><span class="p">,</span> <span class="n">verbose</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">RunPCA</span><span class="p">(</span><span class="n">verbose</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="c1"># the annotation is stored in the &#39;subclass&#39; column of object metadata</span>
<span class="nf">DimPlot</span><span class="p">(</span><span class="n">allen_reference</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&#34;subclass&#34;</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig11.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fig11.png" alt="fig11"></p>
</li>
<li>
<p>Get prediction scores for each spot for each class.</p>
</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">DefaultAssay</span><span class="p">(</span><span class="n">cortex</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="s">&#34;predictions&#34;</span>
<span class="nf">SpatialFeaturePlot</span><span class="p">(</span><span class="n">cortex</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;L2/3 IT&#34;</span><span class="p">,</span> <span class="s">&#34;L4&#34;</span><span class="p">),</span> 
                   <span class="n">pt.size.factor</span> <span class="o">=</span> <span class="m">1.6</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">,</span> <span class="n">crop</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig12.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fig12.png" alt="fig12"></p>
<ul>
<li>
<p>Based on these prediction scores, to predict <code>cell types</code> whose location is spatially restricted.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">cortex</span> <span class="o">&lt;-</span> <span class="nf">FindSpatiallyVariableFeatures</span><span class="p">(</span><span class="n">cortex</span><span class="p">,</span> 
                                        <span class="n">assay</span> <span class="o">=</span> <span class="s">&#34;predictions&#34;</span><span class="p">,</span> 
                                        <span class="n">selection.method</span> <span class="o">=</span> <span class="s">&#34;markvariogram&#34;</span><span class="p">,</span> 
                                        <span class="n">features</span> <span class="o">=</span> <span class="nf">rownames</span><span class="p">(</span><span class="n">cortex</span><span class="p">),</span> 
                                        <span class="n">r.metric</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span> <span class="n">slot</span> <span class="o">=</span> <span class="s">&#34;data&#34;</span><span class="p">)</span>
<span class="n">top.clusters</span> <span class="o">&lt;-</span> <span class="nf">head</span><span class="p">(</span><span class="nf">SpatiallyVariableFeatures</span><span class="p">(</span><span class="n">cortex</span><span class="p">),</span> <span class="m">4</span><span class="p">)</span>
<span class="nf">SpatialPlot</span><span class="p">(</span><span class="n">object</span> <span class="o">=</span> <span class="n">cortex</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="n">top.clusters</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig13.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fig13.png" alt="fig13"></p>
</li>
<li>
<p>Finally, we show that our integrative procedure is capable of recovering the known spatial localization patterns of both neuronal and non-neuronal subsets, including laminar excitatory, layer-1 astrocytes, and the cortical grey matter.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">SpatialFeaturePlot</span><span class="p">(</span><span class="n">cortex</span><span class="p">,</span> 
                <span class="n">features</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;Astro&#34;</span><span class="p">,</span> <span class="s">&#34;L2/3 IT&#34;</span><span class="p">,</span> <span class="s">&#34;L4&#34;</span><span class="p">,</span> <span class="s">&#34;L5 PT&#34;</span><span class="p">,</span> <span class="s">&#34;L5 IT&#34;</span><span class="p">,</span> <span class="s">&#34;L6 CT&#34;</span><span class="p">,</span> <span class="s">&#34;L6 IT&#34;</span><span class="p">,</span> 
                                        <span class="s">&#34;L6b&#34;</span><span class="p">,</span> <span class="s">&#34;Oligo&#34;</span><span class="p">),</span> 
                <span class="n">pt.size.factor</span> <span class="o">=</span> <span class="m">1</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">,</span> <span class="n">crop</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">alpha</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0.1</span><span class="p">,</span> <span class="m">1</span><span class="p">))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig14.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fig14.png" alt="fig14"></p>
</li>
</ul>
<h3 id="working-with-multiple-slices-in-seurat">Working with multiple slices in Seurat</h3>
<ul>
<li>Load datasets</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">brain2</span> <span class="o">&lt;-</span> <span class="nf">LoadData</span><span class="p">(</span><span class="s">&#34;stxBrain&#34;</span><span class="p">,</span> <span class="n">type</span> <span class="o">=</span> <span class="s">&#34;posterior1&#34;</span><span class="p">)</span>
<span class="n">brain2</span> <span class="o">&lt;-</span> <span class="nf">SCTransform</span><span class="p">(</span><span class="n">brain2</span><span class="p">,</span> <span class="n">assay</span> <span class="o">=</span> <span class="s">&#34;Spatial&#34;</span><span class="p">,</span> <span class="n">verbose</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><ul>
<li>
<p>Merge and work multiple slices</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">brain.merge</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">brain</span><span class="p">,</span> <span class="n">brain2</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">DefaultAssay</span><span class="p">(</span><span class="n">brain.merge</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="s">&#34;SCT&#34;</span>
<span class="nf">VariableFeatures</span><span class="p">(</span><span class="n">brain.merge</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="nf">VariableFeatures</span><span class="p">(</span><span class="n">brain</span><span class="p">),</span> <span class="nf">VariableFeatures</span><span class="p">(</span><span class="n">brain2</span><span class="p">))</span>
<span class="n">brain.merge</span> <span class="o">&lt;-</span> <span class="nf">RunPCA</span><span class="p">(</span><span class="n">brain.merge</span><span class="p">,</span> <span class="n">verbose</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="n">brain.merge</span> <span class="o">&lt;-</span> <span class="nf">FindNeighbors</span><span class="p">(</span><span class="n">brain.merge</span><span class="p">,</span> <span class="n">dims</span> <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">)</span>
<span class="n">brain.merge</span> <span class="o">&lt;-</span> <span class="nf">FindClusters</span><span class="p">(</span><span class="n">brain.merge</span><span class="p">,</span> <span class="n">verbose</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="n">brain.merge</span> <span class="o">&lt;-</span> <span class="nf">RunUMAP</span><span class="p">(</span><span class="n">brain.merge</span><span class="p">,</span> <span class="n">dims</span> <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">DimPlot</span><span class="p">(</span><span class="n">brain.merge</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&#34;umap&#34;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;ident&#34;</span><span class="p">,</span> <span class="s">&#34;orig.ident&#34;</span><span class="p">))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig15.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fig15.png" alt="fig15"></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">SpatialDimPlot</span><span class="p">(</span><span class="n">brain.merge</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig16.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fig16.png" alt="fig16"></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">SpatialFeaturePlot</span><span class="p">(</span><span class="n">brain.merge</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;Hpca&#34;</span><span class="p">,</span> <span class="s">&#34;Plp1&#34;</span><span class="p">))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig17.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fig17.png" alt="fig17"></p>
</li>
</ul>
<h3 id="slide-seq">Slide-seq</h3>
<ul>
<li>
<p>Load datasets</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">InstallData</span><span class="p">(</span><span class="s">&#34;ssHippo&#34;</span><span class="p">)</span>
<span class="n">slide.seq</span> <span class="o">&lt;-</span> <span class="nf">LoadData</span><span class="p">(</span><span class="s">&#34;ssHippo&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>Data preprocessing</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">plot1</span> <span class="o">&lt;-</span> <span class="nf">VlnPlot</span><span class="p">(</span><span class="n">slide.seq</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="s">&#34;nCount_Spatial&#34;</span><span class="p">,</span> <span class="n">pt.size</span> <span class="o">=</span> <span class="m">0</span><span class="p">,</span> <span class="n">log</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span> <span class="o">+</span> <span class="nf">NoLegend</span><span class="p">()</span>
<span class="n">slide.seq</span><span class="o">$</span><span class="n">log_nCount_Spatial</span> <span class="o">&lt;-</span> <span class="nf">log</span><span class="p">(</span><span class="n">slide.seq</span><span class="o">$</span><span class="n">nCount_Spatial</span><span class="p">)</span>
<span class="n">plot2</span> <span class="o">&lt;-</span> <span class="nf">SpatialFeaturePlot</span><span class="p">(</span><span class="n">slide.seq</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="s">&#34;log_nCount_Spatial&#34;</span><span class="p">)</span> <span class="o">+</span> <span class="nf">theme</span><span class="p">(</span><span class="n">legend.position</span> <span class="o">=</span> <span class="s">&#34;right&#34;</span><span class="p">)</span>
<span class="nf">wrap_plots</span><span class="p">(</span><span class="n">plot1</span><span class="p">,</span> <span class="n">plot2</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;&#34;</span><span class="n">./fig18.png</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fig18.png" alt="fig18"></p>
</li>
<li>
<p>Normalize</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">slide.seq</span> <span class="o">&lt;-</span> <span class="nf">SCTransform</span><span class="p">(</span><span class="n">slide.seq</span><span class="p">,</span> <span class="n">assay</span> <span class="o">=</span> <span class="s">&#34;Spatial&#34;</span><span class="p">,</span> <span class="n">ncells</span> <span class="o">=</span> <span class="m">3000</span><span class="p">,</span> <span class="n">verbose</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="n">slide.seq</span> <span class="o">&lt;-</span> <span class="nf">RunPCA</span><span class="p">(</span><span class="n">slide.seq</span><span class="p">)</span>
<span class="n">slide.seq</span> <span class="o">&lt;-</span> <span class="nf">RunUMAP</span><span class="p">(</span><span class="n">slide.seq</span><span class="p">,</span> <span class="n">dims</span> <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">)</span>
<span class="n">slide.seq</span> <span class="o">&lt;-</span> <span class="nf">FindNeighbors</span><span class="p">(</span><span class="n">slide.seq</span><span class="p">,</span> <span class="n">dims</span> <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">)</span>
<span class="n">slide.seq</span> <span class="o">&lt;-</span> <span class="nf">FindClusters</span><span class="p">(</span><span class="n">slide.seq</span><span class="p">,</span> <span class="n">resolution</span> <span class="o">=</span> <span class="m">0.3</span><span class="p">,</span> <span class="n">verbose</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>Visualize</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">plot1</span> <span class="o">&lt;-</span> <span class="nf">DimPlot</span><span class="p">(</span><span class="n">slide.seq</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&#34;umap&#34;</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="n">plot2</span> <span class="o">&lt;-</span> <span class="nf">SpatialDimPlot</span><span class="p">(</span><span class="n">slide.seq</span><span class="p">,</span> <span class="n">stroke</span> <span class="o">=</span> <span class="m">0</span><span class="p">)</span>
<span class="n">plot1</span> <span class="o">+</span> <span class="n">plot2</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig19.png&#34;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>Alternately,</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">SpatialDimPlot</span><span class="p">(</span><span class="n">slide.seq</span><span class="p">,</span> 
            <span class="n">cells.highlight</span> <span class="o">=</span> <span class="nf">CellsByIdentities</span><span class="p">(</span><span class="n">object</span> <span class="o">=</span> <span class="n">slide.seq</span><span class="p">,</span> 
                                                <span class="n">idents</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span> <span class="m">6</span><span class="p">,</span> <span class="m">13</span><span class="p">)),</span> 
            <span class="n">facet.highlight</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#34;./fig20.png&#34;</span><span class="p">)</span>     
</code></pre></td></tr></table>
</div>
</div></li>
</ul>
    </div>

    
    


    
    

    <footer class="post-footer">
      <div class="post-tags">
          <a href="/tags/bioinformatics/">Bioinformatics</a>
          <a href="/tags/r-package/">R package</a>
          <a href="/tags/rna-sequence/">RNA Sequence</a>
          <a href="/tags/pca/">PCA</a>
          <a href="/tags/knn/">KNN</a>
          
        </div>

      
      <nav class="post-nav">
        
          <a class="prev" href="/post/git/">
            
            <i class="iconfont">
              <svg  class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="18" height="18">
  <path d="M691.908486 949.511495l75.369571-89.491197c10.963703-12.998035 10.285251-32.864502-1.499144-44.378743L479.499795 515.267417 757.434875 204.940602c11.338233-12.190647 11.035334-32.285311-0.638543-44.850487l-80.46666-86.564541c-11.680017-12.583596-30.356378-12.893658-41.662889-0.716314L257.233596 494.235404c-11.332093 12.183484-11.041474 32.266891 0.657986 44.844348l80.46666 86.564541c1.772366 1.910513 3.706415 3.533476 5.750981 4.877077l306.620399 321.703933C662.505829 963.726242 680.945807 962.528973 691.908486 949.511495z"></path>
</svg>

            </i>
            <span class="prev-text nav-default">Git - Common commands</span>
            <span class="prev-text nav-mobile">Prev</span>
          </a>
        
          <a class="next" href="/post/seurat2/">
            <span class="next-text nav-default">Using Seurat with multimodal data</span>
            <span class="prev-text nav-mobile">Next</span>
            
            <i class="iconfont">
              <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="18" height="18">
  <path d="M332.091514 74.487481l-75.369571 89.491197c-10.963703 12.998035-10.285251 32.864502 1.499144 44.378743l286.278095 300.375162L266.565125 819.058374c-11.338233 12.190647-11.035334 32.285311 0.638543 44.850487l80.46666 86.564541c11.680017 12.583596 30.356378 12.893658 41.662889 0.716314l377.434212-421.426145c11.332093-12.183484 11.041474-32.266891-0.657986-44.844348l-80.46666-86.564541c-1.772366-1.910513-3.706415-3.533476-5.750981-4.877077L373.270379 71.774697C361.493148 60.273758 343.054193 61.470003 332.091514 74.487481z"></path>
</svg>

            </i>
          </a>
      </nav>
    </footer>
  </article>

  
  

  
  

  

  
  

  

  

  

    

  

        </div>
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="icon-links">
  
  
    <a href="mailto:xiaoni0601@gmail.com" rel="me noopener" class="iconfont"
      title="email" >
      <svg class="icon" viewBox="0 0 1451 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="36" height="36">
  <path d="M664.781909 681.472759 0 97.881301C0 3.997201 71.046997 0 71.046997 0L474.477909 0 961.649408 0 1361.641813 0C1361.641813 0 1432.688811 3.997201 1432.688811 97.881301L771.345323 681.472759C771.345323 681.472759 764.482731 685.154773 753.594283 688.65053L753.594283 688.664858C741.602731 693.493018 729.424896 695.068979 718.077952 694.839748 706.731093 695.068979 694.553173 693.493018 682.561621 688.664858L682.561621 688.65053C671.644501 685.140446 664.781909 681.472759 664.781909 681.472759L664.781909 681.472759ZM718.063616 811.603883C693.779541 811.016482 658.879232 802.205449 619.10784 767.734955 542.989056 701.759633 0 212.052267 0 212.052267L0 942.809523C0 942.809523 0 1024 83.726336 1024L682.532949 1024 753.579947 1024 1348.948139 1024C1432.688811 1024 1432.688811 942.809523 1432.688811 942.809523L1432.688811 212.052267C1432.688811 212.052267 893.138176 701.759633 817.019477 767.734955 777.248 802.205449 742.347691 811.03081 718.063616 811.603883L718.063616 811.603883Z"></path>
</svg>

    </a>
  
    <a href="https://github.com/xiaonilee" rel="me noopener" class="iconfont"
      title="github"  target="_blank"
      >
      <svg class="icon" style="" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="36" height="36">
  <path d="M512 12.672c-282.88 0-512 229.248-512 512 0 226.261333 146.688 418.133333 350.08 485.76 25.6 4.821333 34.986667-11.008 34.986667-24.618667 0-12.16-0.426667-44.373333-0.64-87.04-142.421333 30.890667-172.458667-68.693333-172.458667-68.693333C188.672 770.986667 155.008 755.2 155.008 755.2c-46.378667-31.744 3.584-31.104 3.584-31.104 51.413333 3.584 78.421333 52.736 78.421333 52.736 45.653333 78.293333 119.850667 55.68 149.12 42.581333 4.608-33.109333 17.792-55.68 32.426667-68.48-113.706667-12.8-233.216-56.832-233.216-253.013333 0-55.893333 19.84-101.546667 52.693333-137.386667-5.76-12.928-23.04-64.981333 4.48-135.509333 0 0 42.88-13.738667 140.8 52.48 40.96-11.392 84.48-17.024 128-17.28 43.52 0.256 87.04 5.888 128 17.28 97.28-66.218667 140.16-52.48 140.16-52.48 27.52 70.528 10.24 122.581333 5.12 135.509333 32.64 35.84 52.48 81.493333 52.48 137.386667 0 196.693333-119.68 240-233.6 252.586667 17.92 15.36 34.56 46.762667 34.56 94.72 0 68.522667-0.64 123.562667-0.64 140.202666 0 13.44 8.96 29.44 35.2 24.32C877.44 942.592 1024 750.592 1024 524.672c0-282.752-229.248-512-512-512"></path>
</svg>

    </a>


<a href="/index.xml" rel="noopener alternate" type="application/rss&#43;xml"
    class="iconfont" title="rss" target="_blank">
    <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="30" height="30">
  <path d="M819.157333 1024C819.157333 574.592 449.408 204.8 0 204.8V0c561.706667 0 1024 462.293333 1024 1024h-204.842667zM140.416 743.04a140.8 140.8 0 0 1 140.501333 140.586667A140.928 140.928 0 0 1 140.074667 1024C62.72 1024 0 961.109333 0 883.626667s62.933333-140.544 140.416-140.586667zM678.784 1024h-199.04c0-263.210667-216.533333-479.786667-479.744-479.786667V345.173333c372.352 0 678.784 306.517333 678.784 678.826667z"></path>
</svg>

  </a>
   
</div>

<div class="copyright">

  <span class="copyright-year">
    &copy;
    
      2020 -
    2022
    <span class="heart">
      
      <i class="iconfont">
        <svg class="icon" viewBox="0 0 1025 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="14" height="14">
  <path d="M1000.1 247.9c-15.5-37.3-37.6-70.6-65.7-98.9-54.4-54.8-125.8-85-201-85-85.7 0-166 39-221.4 107.4C456.6 103 376.3 64 290.6 64c-75.1 0-146.5 30.4-201.1 85.6-28.2 28.5-50.4 61.9-65.8 99.3-16 38.8-24 79.9-23.6 122.2 0.7 91.7 40.1 177.2 108.1 234.8 3.1 2.6 6 5.1 8.9 7.8 14.9 13.4 58 52.8 112.6 102.7 93.5 85.5 209.9 191.9 257.5 234.2 7 6.1 15.8 9.5 24.9 9.5 9.2 0 18.1-3.4 24.9-9.5 34.5-30.7 105.8-95.9 181.4-165 74.2-67.8 150.9-138 195.8-178.2 69.5-57.9 109.6-144.4 109.9-237.3 0.1-42.5-8-83.6-24-122.2z"
   fill="#8a8a8a"></path>
</svg>

      </i>
    </span><span class="author">
        xiaonilee
        
      </span></span>

  
  

  
</div>

    </footer>

    <div class="back-to-top" id="back-to-top">
      <i class="iconfont">
        
        <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="35" height="35">
  <path d="M510.866688 227.694839 95.449397 629.218702l235.761562 0-2.057869 328.796468 362.40389 0L691.55698 628.188232l241.942331-3.089361L510.866688 227.694839zM63.840492 63.962777l894.052392 0 0 131.813095L63.840492 195.775872 63.840492 63.962777 63.840492 63.962777zM63.840492 63.962777"></path>
</svg>

      </i>
    </div>
  </div>
  
<script type="text/javascript" src="/lib/jquery/jquery-3.2.1.min.js"></script>
  <script type="text/javascript" src="/lib/slideout/slideout-1.0.1.min.js"></script>




<script type="text/javascript" src="/js/main.638251f4230630f0335d8c6748e53a96f94b72670920b60c09a56fdc8bece214.js" integrity="sha256-Y4JR9CMGMPAzXYxnSOU6lvlLcmcJILYMCaVv3Ivs4hQ=" crossorigin="anonymous"></script>



  <script type="text/javascript">
    window.MathJax = {
      showProcessingMessages: false,
      messageStyle: 'none'
    };
  </script>
  <script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-MML-AM_CHTML' async></script>









  
    <script type="text/javascript" src="/js/load-photoswipe.js"></script>
    <script type="text/javascript" src="/lib/photoswipe/photoswipe.min.js"></script>
    <script type="text/javascript" src="/lib/photoswipe/photoswipe-ui-default.min.js"></script>
  















</body>
</html>
