<!DOCTYPE html>
<html lang="en" itemscope itemtype="http://schema.org/WebPage">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>RNA-seq Bioinformatics - RNA sequence analysis - xiaoni&#39;s blog</title>
  

<meta name="renderer" content="webkit" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>

<meta name="MobileOptimized" content="width"/>
<meta name="HandheldFriendly" content="true"/>


<meta name="applicable-device" content="pc,mobile">

<meta name="theme-color" content="#f8f5ec" />
<meta name="msapplication-navbutton-color" content="#f8f5ec">
<meta name="apple-mobile-web-app-capable" content="yes">
<meta name="apple-mobile-web-app-status-bar-style" content="#f8f5ec">

<meta name="mobile-web-app-capable" content="yes">

<meta name="author" content="Xiaoni" />
  <meta name="description" content="Google Colab is a free service providing interactive computing resources via a user interface, runs on Google Cloud Platform (GCP), and provides free access to GPUs and TPUs.
RNA sequence is a technique that can examine the quantity and sequences of RNA in a sample using next generation sequencing (NGS). It analyzes the transcriptome of gene expression patterns encoded within the RNA.
Then, how to perform RNA sequence analysis in Google Colab?
" />

  <meta name="keywords" content="xiaonili, xiaonilee, xiaoni" />






<meta name="generator" content="Hugo 0.79.0" />


<link rel="canonical" href="/post/rnasequenceanalysis/" />





<link rel="icon" href="/favicon.ico" />











<link rel="stylesheet" href="/sass/jane.min.f1e506a781bf25d33ffc18aa6b4e972a965c58049d27d4f92b7db2e9bf28e4bf.css" integrity="sha256-8eUGp4G/JdM//Biqa06XKpZcWASdJ9T5K32y6b8o5L8=" media="screen" crossorigin="anonymous">



<link href="https://cdn.bootcdn.net/ajax/libs/font-awesome/5.15.1/css/all.css" rel="stylesheet">
<script src="https://cdn.bootcdn.net/ajax/libs/font-awesome/5.15.1/js/all.js"></script>




<meta property="og:title" content="RNA-seq Bioinformatics - RNA sequence analysis" />
<meta property="og:description" content="Google Colab is a free service providing interactive computing resources via a user interface, runs on Google Cloud Platform (GCP), and provides free access to GPUs and TPUs.
RNA sequence is a technique that can examine the quantity and sequences of RNA in a sample using next generation sequencing (NGS). It analyzes the transcriptome of gene expression patterns encoded within the RNA.
Then, how to perform RNA sequence analysis in Google Colab?" />
<meta property="og:type" content="article" />
<meta property="og:url" content="/post/rnasequenceanalysis/" />
<meta property="article:published_time" content="2021-02-17T00:00:00+00:00" />
<meta property="article:modified_time" content="2021-02-18T00:00:00+00:00" />
<meta itemprop="name" content="RNA-seq Bioinformatics - RNA sequence analysis">
<meta itemprop="description" content="Google Colab is a free service providing interactive computing resources via a user interface, runs on Google Cloud Platform (GCP), and provides free access to GPUs and TPUs.
RNA sequence is a technique that can examine the quantity and sequences of RNA in a sample using next generation sequencing (NGS). It analyzes the transcriptome of gene expression patterns encoded within the RNA.
Then, how to perform RNA sequence analysis in Google Colab?">
<meta itemprop="datePublished" content="2021-02-17T00:00:00+00:00" />
<meta itemprop="dateModified" content="2021-02-18T00:00:00+00:00" />
<meta itemprop="wordCount" content="942">



<meta itemprop="keywords" content="Python,Bioinformatics,Colab,Linux,RNA Sequence," />
<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="RNA-seq Bioinformatics - RNA sequence analysis"/>
<meta name="twitter:description" content="Google Colab is a free service providing interactive computing resources via a user interface, runs on Google Cloud Platform (GCP), and provides free access to GPUs and TPUs.
RNA sequence is a technique that can examine the quantity and sequences of RNA in a sample using next generation sequencing (NGS). It analyzes the transcriptome of gene expression patterns encoded within the RNA.
Then, how to perform RNA sequence analysis in Google Colab?"/>

<!--[if lte IE 9]>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/classlist/1.1.20170427/classList.min.js"></script>
<![endif]-->

<!--[if lt IE 9]>
  <script src="https://cdn.jsdelivr.net/npm/html5shiv@3.7.3/dist/html5shiv.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/respond.js@1.4.2/dest/respond.min.js"></script>
<![endif]-->


<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'UA-177860608-1', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>



</head>
<body>
  <div id="mobile-navbar" class="mobile-navbar">
  <div class="mobile-header-logo">
    <a href="/" class="logo">Xiaoni's Blog</a>
  </div>
  <div class="mobile-navbar-icon">
    <span></span>
    <span></span>
    <span></span>
  </div>
</div>
<nav id="mobile-menu" class="mobile-menu slideout-menu">
  <ul class="mobile-menu-list">
    <li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/">Home</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/post/">Archives</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/categories/">Categories</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/tags/">Tags</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/about/">About Me</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          <div class="mobile-menu-parent">
            <span class="mobile-submenu-open"></span>
            <a href="/">
              docs
            </a>
          </div>
          <ul class="mobile-submenu-list">
            
              <li>
                <a href="/post/tensorflow2/"> Machine Learning: TensorFlow2 Notebook</a>
              </li>
            
              <li>
                <a href="/post/bioscirep/">Bioinformatics analysis of prognostic value of PITX1 gene in breast cancer</a>
              </li>
            
              <li>
                <a href="/post/rinactionmap/">Book: R in Action</a>
              </li>
            
              <li>
                <a href="/post/tumorimmune/">Comprehensive Bioinformatics Analysis of Tumor-Infiltrating Immune Cells</a>
              </li>
            
              <li>
                <a href="/post/aavcapsid/">Deep diversification of an AAV capsid protein by machine learning</a>
              </li>
            
              <li>
                <a href="/post/firebrowse/">FireBrowse Notebook</a>
              </li>
            
              <li>
                <a href="/post/gdc/">GDC Notebook</a>
              </li>
            
              <li>
                <a href="/post/git/">Git - Common commands</a>
              </li>
            
              <li>
                <a href="/post/gscalite/">GSCALite: Gene Set Cancer Analysis</a>
              </li>
            
              <li>
                <a href="/post/frontoncol/">Identification of Therapeutic Targets and Prognostic Biomarkers Among CXC Chemokines in the Renal Cell Carcinoma Microenvironment</a>
              </li>
            
              <li>
                <a href="/post/condacolab/">Installing Conda in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/oncolnc/">OncoLnc: A straightforward Tool</a>
              </li>
            
              <li>
                <a href="/post/webcrawler/">Python Spider</a>
              </li>
            
              <li>
                <a href="/post/rnasequenceanalysis/">RNA-seq Bioinformatics - RNA sequence analysis</a>
              </li>
            
              <li>
                <a href="/post/scrnaknn/">scRNA-sequencing analysis - PCA and KNN</a>
              </li>
            
              <li>
                <a href="/post/survival/">Survival Analysis with data from GSE database</a>
              </li>
            
              <li>
                <a href="/post/rincolab/">The way to use R in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/tissgdb/">TissGDB: tissue-specific gene database in cancer</a>
              </li>
            
              <li>
                <a href="/post/ucscmap/">UCSC Topics</a>
              </li>
            
              <li>
                <a href="/post/webfrontend/">web front end</a>
              </li>
            
          </ul>
        
      </li>
    

    
  </ul>
</nav>


  
    






  <link rel="stylesheet" href="/lib/photoswipe/photoswipe.min.css" />
  <link rel="stylesheet" href="/lib/photoswipe/default-skin/default-skin.min.css" />




<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

<div class="pswp__bg"></div>

<div class="pswp__scroll-wrap">
    
    <div class="pswp__container">
      <div class="pswp__item"></div>
      <div class="pswp__item"></div>
      <div class="pswp__item"></div>
    </div>
    
    <div class="pswp__ui pswp__ui--hidden">
    <div class="pswp__top-bar">
      
      <div class="pswp__counter"></div>
      <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>
      <button class="pswp__button pswp__button--share" title="Share"></button>
      <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>
      <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>
      
      
      <div class="pswp__preloader">
        <div class="pswp__preloader__icn">
          <div class="pswp__preloader__cut">
            <div class="pswp__preloader__donut"></div>
          </div>
        </div>
      </div>
    </div>
    <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
      <div class="pswp__share-tooltip"></div>
    </div>
    <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
    </button>
    <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
    </button>
    <div class="pswp__caption">
      <div class="pswp__caption__center"></div>
    </div>
    </div>
    </div>
</div>

  

  

  

  <header id="header" class="header container">
    <div class="logo-wrapper">
  <a href="/" class="logo">
    
      Xiaoni's Blog
    
  </a>
</div>

<nav class="site-navbar">
  <ul id="menu" class="menu">
    
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/">Home</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/post/">Archives</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/categories/">Categories</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/tags/">Tags</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/about/">About Me</a>
          

        

      </li>
    
        <li class="menu-item active">
        
          
          <a class="menu-item-link menu-parent" href="/">docs</a>
          <ul class="submenu" >
            
              <li>
                <a href="/post/tensorflow2/"> Machine Learning: TensorFlow2 Notebook</a>
              </li>
            
              <li>
                <a href="/post/bioscirep/">Bioinformatics analysis of prognostic value of PITX1 gene in breast cancer</a>
              </li>
            
              <li>
                <a href="/post/rinactionmap/">Book: R in Action</a>
              </li>
            
              <li>
                <a href="/post/tumorimmune/">Comprehensive Bioinformatics Analysis of Tumor-Infiltrating Immune Cells</a>
              </li>
            
              <li>
                <a href="/post/aavcapsid/">Deep diversification of an AAV capsid protein by machine learning</a>
              </li>
            
              <li>
                <a href="/post/firebrowse/">FireBrowse Notebook</a>
              </li>
            
              <li>
                <a href="/post/gdc/">GDC Notebook</a>
              </li>
            
              <li>
                <a href="/post/git/">Git - Common commands</a>
              </li>
            
              <li>
                <a href="/post/gscalite/">GSCALite: Gene Set Cancer Analysis</a>
              </li>
            
              <li>
                <a href="/post/frontoncol/">Identification of Therapeutic Targets and Prognostic Biomarkers Among CXC Chemokines in the Renal Cell Carcinoma Microenvironment</a>
              </li>
            
              <li>
                <a href="/post/condacolab/">Installing Conda in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/oncolnc/">OncoLnc: A straightforward Tool</a>
              </li>
            
              <li>
                <a href="/post/webcrawler/">Python Spider</a>
              </li>
            
              <li>
                <a href="/post/rnasequenceanalysis/">RNA-seq Bioinformatics - RNA sequence analysis</a>
              </li>
            
              <li>
                <a href="/post/scrnaknn/">scRNA-sequencing analysis - PCA and KNN</a>
              </li>
            
              <li>
                <a href="/post/survival/">Survival Analysis with data from GSE database</a>
              </li>
            
              <li>
                <a href="/post/rincolab/">The way to use R in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/tissgdb/">TissGDB: tissue-specific gene database in cancer</a>
              </li>
            
              <li>
                <a href="/post/ucscmap/">UCSC Topics</a>
              </li>
            
              <li>
                <a href="/post/webfrontend/">web front end</a>
              </li>
            
          </ul>

        

      </li>
    

    
    

    
  </ul>
</nav>

  </header>

  <div id="mobile-panel">
    <main id="main" class="main bg-llight">
      <div class="content-wrapper">
        <div id="content" class="content container">
          <article class="post bg-white">
    
    <header class="post-header">
      <h1 class="post-title">RNA-seq Bioinformatics - RNA sequence analysis</h1>
      
      <div class="post-meta">
        <time datetime="2021-02-17" class="post-time">
          2021-02-17
        </time>
        <div class="post-category">
            <a href="/categories/python/"> Python </a>
            <a href="/categories/bioinformatics/"> Bioinformatics </a>
            <a href="/categories/colab/"> Colab </a>
            <a href="/categories/linux/"> Linux </a>
            <a href="/categories/rna-sequence/"> RNA Sequence </a>
            
          </div>
        

        
        

        
        
      </div>
    </header>

    
    
<div class="post-toc" id="post-toc">
  <h2 class="post-toc-title">Table of Contents</h2>
  <div class="post-toc-content">
    <nav id="TableOfContents">
  <ul>
    <li>
      <ul>
        <li><a href="#prerequisites">Prerequisites</a></li>
        <li><a href="#download-fastq-files-with-aspera">Download FastQ files with aspera</a></li>
        <li><a href="#cut-adapter-with-trim_galore">Cut adapter with Trim_galore</a></li>
        <li><a href="#reads-alignment-with-hisat2">Reads alignment with HISAT2</a></li>
        <li><a href="#build-indexes">Build Indexes</a></li>
        <li><a href="#quanity-and-counts">Quanity and counts</a></li>
      </ul>
    </li>
  </ul>
</nav>
  </div>
</div>

    
    <div class="post-content">
      <p><strong>Google Colab</strong> is a <em><strong>free</strong></em> service providing interactive computing resources via a user interface, runs on Google Cloud Platform (GCP), and provides free access to GPUs and TPUs.</p>
<p><strong>RNA sequence</strong> is a technique that can examine the quantity and sequences of RNA in a sample using next generation sequencing (NGS). It analyzes the <em><strong>transcriptome</strong></em> of gene expression patterns encoded within the RNA.</p>
<p>Then, how to perform <em><strong>RNA sequence analysis</strong></em> in Google Colab?</p>
<h3 id="prerequisites">Prerequisites</h3>
<ul>
<li>
<p>Install condas, follow my article <a href="/post/CondaColab">here</a>.</p>
</li>
<li>
<p>Continued, setup the channels for conda.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-python" data-lang="python"><span class="err">!</span><span class="n">conda</span> <span class="n">config</span> <span class="o">--</span><span class="n">add</span> <span class="n">channels</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">mirrors</span><span class="o">.</span><span class="n">tuna</span><span class="o">.</span><span class="n">tsinghua</span><span class="o">.</span><span class="n">edu</span><span class="o">.</span><span class="n">cn</span><span class="o">/</span><span class="n">anaconda</span><span class="o">/</span><span class="n">pkgs</span><span class="o">/</span><span class="n">free</span>
<span class="err">!</span><span class="n">conda</span> <span class="n">config</span> <span class="o">--</span><span class="n">add</span> <span class="n">channels</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">mirrors</span><span class="o">.</span><span class="n">tuna</span><span class="o">.</span><span class="n">tsinghua</span><span class="o">.</span><span class="n">edu</span><span class="o">.</span><span class="n">cn</span><span class="o">/</span><span class="n">anaconda</span><span class="o">/</span><span class="n">cloud</span><span class="o">/</span><span class="n">conda</span><span class="o">-</span><span class="n">forge</span>
<span class="err">!</span><span class="n">conda</span> <span class="n">config</span> <span class="o">--</span><span class="n">add</span> <span class="n">channels</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">mirrors</span><span class="o">.</span><span class="n">tuna</span><span class="o">.</span><span class="n">tsinghua</span><span class="o">.</span><span class="n">edu</span><span class="o">.</span><span class="n">cn</span><span class="o">/</span><span class="n">anaconda</span><span class="o">/</span><span class="n">cloud</span><span class="o">/</span><span class="n">bioconda</span>
<span class="err">!</span><span class="n">conda</span> <span class="n">config</span> <span class="o">--</span><span class="nb">set</span> <span class="n">show_channel_urls</span> <span class="n">yes</span> 
</code></pre></td></tr></table>
</div>
</div></li>
</ul>
<h3 id="download-fastq-files-with-aspera">Download FastQ files with aspera</h3>
<ul>
<li>
<p>Conda install aspera</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-python" data-lang="python"><span class="err">!</span><span class="n">conda</span> <span class="n">install</span> <span class="o">-</span><span class="n">y</span> <span class="o">-</span><span class="n">c</span> <span class="n">hcc</span> <span class="n">aspera</span><span class="o">-</span><span class="n">cli</span>

<span class="c1">#checkup</span>
<span class="err">!</span><span class="n">which</span> <span class="n">ascp</span>
</code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>Check and cat the path for <code>asperaweb_id.dsa.openssh</code></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-python" data-lang="python"><span class="err">!</span><span class="n">cat</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">etc</span><span class="o">/</span><span class="n">asperaweb_id_dsa</span><span class="o">.</span><span class="n">openssh</span>
</code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>To get the SRA address for GSE130437</p>
<ul>
<li><a href="https://sra-explorer.info">SRA</a>.</li>
<li>Search for <code>GSE130437</code>, herein, it&rsquo;s the number <code>PRJNA540259</code>, which got from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE130437">ncbi</a>.</li>
<li>Click on <code>Title</code> and choose all results.</li>
<li>Add to collection.</li>
<li>Click on <code>#n</code> saved datasets(Herein, I just choose 2 of them: for <code>SRR8984523.fastq.gz</code> and <code>SRR8984529.fastq.gz</code>).</li>
<li>Pull down to click on <code>FastQ Downloads</code>.</li>
<li>Choose <code>Bash script for downloading FastQ files</code>.</li>
<li>Creat Newfile <code>dl.sh</code> in the path <code>/content/rawdata</code>.</li>
<li>Click <code>Copy</code> and paste into <code>dl.sh</code> to save it.</li>
</ul>
</li>
<li>
<p>Show me the code <i class="far fa-hand-point-down"></i></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-shell" data-lang="shell"><span class="cp">#!/usr/bin/env bash
</span><span class="cp"></span>curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR898/003/SRR8984523/SRR8984523.fastq.gz -o SRR8984523_GSM3738651_MCF7_palbociclib_resistant_Replicate_1_Homo_sapiens_RNA-Seq.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR898/009/SRR8984529/SRR8984529.fastq.gz -o SRR8984529_GSM3738657_MDAMB231_palbociclib_resistant_Replicate_1_Homo_sapiens_RNA-Seq.fastq.gz
</code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>Start to download</p>
<ul>
<li>
<p><code>chmod a+x</code> and run script to download files from links of <code>dl.sh</code>.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-python" data-lang="python"><span class="err">!</span><span class="n">mkdir</span> <span class="n">rawdata</span><span class="o">/</span>
<span class="err">!</span><span class="n">cd</span> <span class="o">/</span><span class="n">content</span><span class="o">/</span><span class="n">rawdata</span><span class="o">/</span>
<span class="err">!</span><span class="n">chmod</span> <span class="n">a</span><span class="o">+</span><span class="n">x</span> <span class="o">/</span><span class="n">content</span><span class="o">/</span><span class="n">rawdata</span><span class="o">/</span><span class="n">dl</span><span class="o">.</span><span class="n">sh</span>
<span class="err">!</span><span class="n">bash</span> <span class="o">/</span><span class="n">content</span><span class="o">/</span><span class="n">rawdata</span><span class="o">/</span><span class="n">dl</span><span class="o">.</span><span class="n">sh</span>
</code></pre></td></tr></table>
</div>
</div></li>
</ul>
</li>
</ul>
<h3 id="cut-adapter-with-trim_galore">Cut adapter with Trim_galore</h3>
<ul>
<li>
<p><a href="https://github.com/FelixKrueger/TrimGalore">Trim Galore</a> is a wrapper around Cutadapt and FastQC to consistently apply adapter and quality trimming to FastQ files, with extra functionality for RRBS data.</p>
</li>
<li>
<p>Trim Galore is a a Perl wrapper around two tools: Cutadapt and FastQC.</p>
<ul>
<li>
<p>To use, ensure that these two pieces of software are available and copy the trim_galore script to a location available on the PATH.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-python" data-lang="python"><span class="err">!</span><span class="n">mkdir</span> <span class="n">clean</span>
<span class="err">!</span><span class="n">cd</span> <span class="o">/</span><span class="n">content</span><span class="o">/</span><span class="n">clean</span><span class="o">/</span>
<span class="c1"># Install trim galore(perl)</span>
<span class="err">!</span><span class="n">wget</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">raw</span><span class="o">.</span><span class="n">githubusercontent</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">FelixKrueger</span><span class="o">/</span><span class="n">TrimGalore</span><span class="o">/</span><span class="n">master</span><span class="o">/</span><span class="n">trim_galore</span>
<span class="err">!</span><span class="n">chmod</span> <span class="n">a</span><span class="o">+</span><span class="n">x</span> <span class="o">./</span><span class="n">trim_galore</span>
<span class="err">!</span><span class="o">./</span><span class="n">trim_galore</span>

<span class="c1"># Install cutadapt(python)</span>
<span class="err">!</span><span class="n">pip3</span> <span class="n">install</span> <span class="n">cutadapt</span>

<span class="err">!</span><span class="n">cd</span> <span class="o">/</span><span class="n">content</span><span class="o">/</span><span class="n">clean</span><span class="o">/</span>
<span class="c1"># write into the config.txt</span>
<span class="err">!</span><span class="n">ls</span> <span class="o">/</span><span class="n">content</span><span class="o">/</span><span class="n">rawdata</span><span class="o">/*.</span><span class="n">gz</span> <span class="o">&gt;</span> <span class="o">/</span><span class="n">content</span><span class="o">/</span><span class="n">clean</span><span class="o">/</span><span class="n">config</span><span class="o">.</span><span class="n">txt</span>
<span class="err">!</span><span class="n">cat</span> <span class="n">config</span><span class="o">.</span><span class="n">txt</span>

<span class="c1"># touch cl.sh for single pass</span>
<span class="err">!</span><span class="n">chmod</span> <span class="n">a</span><span class="o">+</span><span class="n">x</span> <span class="o">/</span><span class="n">content</span><span class="o">/</span><span class="n">clean</span><span class="o">/</span><span class="n">cl</span><span class="o">.</span><span class="n">sh</span>
</code></pre></td></tr></table>
</div>
</div></li>
</ul>
</li>
<li>
<p>Show me the code <i class="far fa-hand-point-down"></i></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span><span class="lnt">9
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-shell" data-lang="shell"><span class="nv">trim_galore_cmd</span><span class="o">=</span>/content/clean/trim_galore
<span class="nv">dir</span><span class="o">=</span><span class="s1">&#39;/content/clean&#39;</span>
<span class="nv">file</span><span class="o">=</span><span class="s2">&#34;config.txt&#34;</span>
<span class="nb">cd</span> <span class="nv">$dir</span>
cat <span class="nv">$file</span> <span class="p">|</span><span class="k">while</span> <span class="nb">read</span> idpath
<span class="k">do</span>
        <span class="nv">rawdatapath</span><span class="o">=</span><span class="sb">`</span><span class="nb">echo</span> <span class="nv">$idpath</span><span class="sb">`</span>
        <span class="nv">$trim_galore_cmd</span> -q <span class="m">25</span> --phred33 --length <span class="m">36</span> --stringency <span class="m">3</span> -o <span class="nv">$dir</span> <span class="nv">$rawdatapath</span>
<span class="k">done</span>
</code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>Do adapter trimming with script <code>cl.sh</code></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-python" data-lang="python"><span class="err">!</span><span class="o">/</span><span class="n">content</span><span class="o">/</span><span class="n">clean</span><span class="o">/</span><span class="n">cl</span><span class="o">.</span><span class="n">sh</span>
</code></pre></td></tr></table>
</div>
</div></li>
</ul>
<h3 id="reads-alignment-with-hisat2">Reads alignment with HISAT2</h3>
<ul>
<li>
<p><a href="http://daehwankimlab.github.io/hisat2/">HISAT2</a> is a fast and sensitive alignment program for mapping next-generation sequencing reads (both DNA and RNA) to a population of human genomes as well as to a single reference genome.</p>
</li>
<li>
<p>HISAT2 uses a large set of small GFM indexes that collectively cover the whole genome.</p>
</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-python" data-lang="python"><span class="n">cd</span> <span class="o">/</span><span class="n">content</span><span class="o">/</span><span class="n">clean</span><span class="o">/</span>

<span class="c1"># rename to simple</span>
<span class="err">!</span><span class="n">mv</span> <span class="n">SRR8984523_GSM3738651_MCF7_palbociclib_resistant_Replicate_1_Homo_sapiens_RNA</span><span class="o">-</span><span class="n">Seq_trimmed</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span> <span class="n">SRR8984523_trimmed</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span>

<span class="c1"># rename to simple</span>
<span class="err">!</span><span class="n">mv</span> <span class="n">SRR8984529_GSM3738651_MCF7_palbociclib_resistant_Replicate_1_Homo_sapiens_RNA</span><span class="o">-</span><span class="n">Seq_trimmed</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span> <span class="n">SRR8984529_trimmed</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span>

<span class="c1"># make file</span>
<span class="err">!</span><span class="n">touch</span> <span class="o">/</span><span class="n">content</span><span class="o">/</span><span class="n">align</span><span class="o">/</span><span class="n">SRR_Acc_List</span><span class="o">.</span><span class="n">txt</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="4.png" alt="4"></p>
<h3 id="build-indexes">Build Indexes</h3>
<ul>
<li>Install HISAT2 and Samtools</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-python" data-lang="python"><span class="err">!</span><span class="n">cat</span> <span class="o">/</span><span class="n">etc</span><span class="o">/</span><span class="n">os</span><span class="o">-</span><span class="n">release</span>
<span class="err">!</span><span class="n">sudo</span> <span class="n">apt</span> <span class="n">install</span> <span class="n">hisat2</span> <span class="n">samtools</span>
</code></pre></td></tr></table>
</div>
</div><ul>
<li>Prepare data</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-python" data-lang="python"><span class="c1"># Download reference</span>
<span class="err">!</span><span class="n">wget</span> <span class="n">ftp</span><span class="p">:</span><span class="o">//</span><span class="n">ftp</span><span class="o">.</span><span class="n">ensembl</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">pub</span><span class="o">/</span><span class="n">release</span><span class="o">-</span><span class="mi">84</span><span class="o">/</span><span class="n">fasta</span><span class="o">/</span><span class="n">homo_sapiens</span><span class="o">/</span><span class="n">dna</span><span class="o">/</span><span class="n">Homo_sapiens</span><span class="o">.</span><span class="n">GRCh38</span><span class="o">.</span><span class="n">dna</span><span class="o">.</span><span class="n">primary_assembly</span><span class="o">.</span><span class="n">fa</span><span class="o">.</span><span class="n">gz</span>
<span class="err">!</span><span class="n">gzip</span> <span class="o">-</span><span class="n">d</span> <span class="n">Homo_sapiens</span><span class="o">.</span><span class="n">GRCh38</span><span class="o">.</span><span class="n">dna</span><span class="o">.</span><span class="n">primary_assembly</span><span class="o">.</span><span class="n">fa</span><span class="o">.</span><span class="n">gz</span>
<span class="err">!</span><span class="n">mv</span> <span class="n">Homo_sapiens</span><span class="o">.</span><span class="n">GRCh38</span><span class="o">.</span><span class="n">dna</span><span class="o">.</span><span class="n">primary_assembly</span><span class="o">.</span><span class="n">fa</span> <span class="n">genome</span><span class="o">.</span><span class="n">fa</span>
</code></pre></td></tr></table>
</div>
</div><ul>
<li>Build HFM index</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-python" data-lang="python"><span class="c1"># Build HFM index</span>
<span class="c1">#!hisat2-build --help</span>
<span class="c1">#!cat /proc/cpuinfo</span>
<span class="c1"># Consider the systerm is limited, -p 16 is change to -p 2;</span>
<span class="err">!</span><span class="n">hisat2</span><span class="o">-</span><span class="n">build</span> <span class="o">-</span><span class="n">p</span> <span class="mi">2</span> <span class="n">genome</span><span class="o">.</span><span class="n">fa</span> <span class="n">genome</span>

<span class="err">!</span><span class="n">mkdir</span> <span class="o">/</span><span class="n">content</span><span class="o">/</span><span class="n">genome</span>
<span class="c1"># Move `genome.1.ht2 genome.2.ht2 genome.3.ht2 genome.4.ht2 genome.5.ht2 genome.6.ht2 genome.7.ht2 genome.8.ht2` to /content/genome</span>
</code></pre></td></tr></table>
</div>
</div><ul>
<li>
<p>Do alignment with script <code>sn.sh</code></p>
</li>
<li>
<p>Touch sn.sh</p>
<ul>
<li>
<p>Show me the code <i class="far fa-hand-point-down"></i></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-python" data-lang="python"><span class="n">hisat2_indexes_dir</span><span class="o">=/</span><span class="n">content</span><span class="o">/</span><span class="n">genome</span>
<span class="n">index</span><span class="o">=</span><span class="n">genome</span>
<span class="n">inputdir</span><span class="o">=/</span><span class="n">content</span><span class="o">/</span><span class="n">clean</span>
<span class="n">outdir</span><span class="o">=/</span><span class="n">content</span><span class="o">/</span><span class="n">align</span>
<span class="n">echo</span> <span class="s2">&#34;export HISAT2_INDEXES=$hisat2_indexes_dir&#34;</span> <span class="o">&gt;</span> <span class="n">Hisat</span><span class="o">.</span><span class="n">sh</span>

<span class="n">cat</span> <span class="o">/</span><span class="n">content</span><span class="o">/</span><span class="n">align</span><span class="o">/</span><span class="n">SRR_Acc_List</span><span class="o">.</span><span class="n">txt</span><span class="o">|</span><span class="k">while</span> <span class="n">read</span> <span class="nb">id</span>
<span class="n">do</span>
<span class="c1"># echo &#34;nohup hisat2 -p 2 -x $index -U $inputdir/${id}_trimmed.fq.gz|samtools sort -@ 5 -o $outdir/${id}_aln.sorted.bam  &amp;&amp; samtools index $outdir/${id}_aln.sorted.bam $outdir/${id}_aln.sorted.bam.bai &amp;&amp; samtools flagstat $outdir/${id}_aln.sorted.bam &gt;$id.txt &amp;&#34;</span>
<span class="n">echo</span> <span class="s2">&#34;hisat2 -p 2 -x $index -U $inputdir/${id}_trimmed.fq.gz|samtools sort -@ 5 -o $outdir/${id}_aln.sorted.bam  &amp;&amp; samtools index $outdir/${id}_aln.sorted.bam $outdir/${id}_aln.sorted.bam.bai &amp;&amp; samtools flagstat $outdir/${id}_aln.sorted.bam &gt;$id.txt&#34;</span>
<span class="n">done</span> <span class="o">&gt;&gt;</span><span class="n">Hisat</span><span class="o">.</span><span class="n">sh</span>
<span class="n">echo</span> <span class="s2">&#34;&#34;</span> <span class="o">&gt;&gt;</span> <span class="n">Hisat</span><span class="o">.</span><span class="n">sh</span>
</code></pre></td></tr></table>
</div>
</div></li>
</ul>
</li>
<li>
<p>Run script</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span><span class="lnt">9
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-shell" data-lang="shell">!chmod a+x sn.sh
!cat /content/align/SRR_Acc_List.txt
!./sn.sh

!cat Hisat.sh
!chmod a+x Hisat.sh
<span class="c1">#!hisat2 --help</span>

!./Hisat.sh
</code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>Have a look at the files
<img src="2.png" alt="2"></p>
</li>
<li>
<p>Check Results</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-python" data-lang="python"><span class="err">!</span><span class="n">cat</span> <span class="o">/</span><span class="n">content</span><span class="o">/</span><span class="n">result</span><span class="o">/</span><span class="n">SRR8984523</span><span class="o">.</span><span class="n">txt</span>

<span class="c1">#%cd /content/align</span>
<span class="err">!</span><span class="n">chmod</span> <span class="n">a</span><span class="o">+</span><span class="n">x</span> <span class="n">res</span><span class="o">.</span><span class="n">sh</span>
<span class="err">!</span><span class="o">./</span><span class="n">res</span><span class="o">.</span><span class="n">sh</span>
</code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>Have a look at the txt file
<img src="3.png" alt="3"></p>
</li>
<li>
<p>Show me the code <i class="far fa-hand-point-down"></i></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-shell" data-lang="shell">
<span class="nv">dir</span><span class="o">=</span><span class="s1">&#39;/content/align&#39;</span>
<span class="nv">outfile</span><span class="o">=</span><span class="s1">&#39;mapinfo.txt&#39;</span>
<span class="nb">cd</span> <span class="nv">$dir</span>

<span class="k">if</span>  <span class="o">[</span> -f  <span class="nv">$outfile</span>  <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
rm  -rf  <span class="nv">$outfile</span>
<span class="k">fi</span>

ls *txt<span class="p">|</span><span class="k">while</span> <span class="nb">read</span> id<span class="p">;</span><span class="k">do</span><span class="o">(</span><span class="nb">echo</span> <span class="nv">$id</span><span class="p">|</span>awk -F <span class="s2">&#34;.&#34;</span> <span class="s1">&#39;{print $1}&#39;</span>&gt;&gt;sample<span class="o">)</span><span class="p">;</span><span class="k">done</span>

ls *txt<span class="p">|</span><span class="k">while</span> <span class="nb">read</span> id<span class="p">;</span><span class="k">do</span><span class="o">(</span>cat <span class="nv">$id</span><span class="p">|</span>cut -d <span class="s2">&#34; &#34;</span> -f 1<span class="p">|</span>sed -n <span class="s2">&#34;1p&#34;</span><span class="p">|</span>awk <span class="s1">&#39;{printf &#34;%18.0f\n&#34;,$0}&#39;</span> &gt;&gt;totalreads<span class="o">)</span><span class="p">;</span><span class="k">done</span>
ls *txt<span class="p">|</span><span class="k">while</span> <span class="nb">read</span> id<span class="p">;</span><span class="k">do</span><span class="o">(</span>cat <span class="nv">$id</span><span class="p">|</span>cut -d <span class="s2">&#34; &#34;</span> -f 1<span class="p">|</span>sed -n <span class="s2">&#34;2p&#34;</span><span class="p">|</span>awk <span class="s1">&#39;{printf &#34;%18.0f\n&#34;,$0}&#39;</span>&gt;&gt;secondary<span class="o">)</span><span class="p">;</span><span class="k">done</span>
ls *txt<span class="p">|</span><span class="k">while</span> <span class="nb">read</span> id<span class="p">;</span><span class="k">do</span><span class="o">(</span>cat <span class="nv">$id</span><span class="p">|</span>cut -d <span class="s2">&#34; &#34;</span> -f 4,5<span class="p">|</span>sed -n <span class="s1">&#39;5p&#39;</span><span class="p">|</span>awk -F <span class="s2">&#34;(&#34;</span> <span class="s1">&#39;{print $2}&#39;</span> &gt;&gt;mapratio<span class="o">)</span><span class="p">;</span><span class="k">done</span>
<span class="nb">echo</span> <span class="s2">&#34;sample    totalreads      secondary       mapratio&#34;</span> &gt;<span class="nv">$outfile</span>
paste -d <span class="s2">&#34;\t&#34;</span> sample totalreads secondary mapratio &gt;&gt;<span class="nv">$outfile</span>

rm sample totalreads secondary mapratio
</code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>Have a look at the mapratio.txt file
<img src="1.png" alt="1"></p>
</li>
</ul>
<h3 id="quanity-and-counts">Quanity and counts</h3>
<ul>
<li>
<p>Install TOOLS</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-python" data-lang="python">  <span class="c1"># Install FeatureCounts</span>
  <span class="err">!</span><span class="n">apt</span><span class="o">-</span><span class="n">get</span> <span class="n">install</span> <span class="n">subread</span>
  <span class="err">!</span><span class="n">featureCounts</span> <span class="o">--</span><span class="n">help</span>
    
  <span class="c1">#Install multiqc</span>
  <span class="err">!</span><span class="n">pip3</span> <span class="n">install</span> <span class="n">multiqc</span>
  <span class="err">!</span><span class="n">multiqc</span> <span class="o">--</span><span class="n">help</span>
</code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>Download annotation</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-python" data-lang="python">  <span class="c1">#%cd /content/counts/</span>
  <span class="c1">#!wget ftp://ftp.ensembl.org/pub/release-90/gtf/homo_sapiens/Homo_sapiens.GRCh38.90.gtf.gz</span>
  <span class="err">!</span><span class="n">wget</span> <span class="n">ftp</span><span class="p">:</span><span class="o">//</span><span class="n">ftp</span><span class="o">.</span><span class="n">ensembl</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">pub</span><span class="o">/</span><span class="n">release</span><span class="o">-</span><span class="mi">86</span><span class="o">/</span><span class="n">gtf</span><span class="o">/</span><span class="n">homo_sapiens</span><span class="o">/</span><span class="n">Homo_sapiens</span><span class="o">.</span><span class="n">GRCh38</span><span class="o">.</span><span class="mf">86.</span><span class="n">gtf</span><span class="o">.</span><span class="n">gz</span>
</code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>Touch count.sh</p>
</li>
<li>
<p>Show me the code <i class="far fa-hand-point-down"></i></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-shell" data-lang="shell">    
  <span class="nv">gtf</span><span class="o">=</span><span class="s2">&#34;/content/counts/Homo_sapiens.GRCh38.86.gtf.gz&#34;</span>
  <span class="nv">inputdir</span><span class="o">=</span><span class="s2">&#34;/content/align&#34;</span>
    
  featureCounts -T <span class="m">10</span> -g gene_id -a <span class="nv">$gtf</span> -o SRR8984523.txt <span class="nv">$inputdir</span>/*.sorted.bam

    
  multiqc -f SRR8984523.txt.summary

  cat SRR8984523.txt <span class="p">|</span> cut -f1,7- &gt; counts.txt
</code></pre></td></tr></table>
</div>
</div></li>
<li>
<p>Run script</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-python" data-lang="python"><span class="err">!</span><span class="n">chmod</span> <span class="n">a</span><span class="o">+</span><span class="n">x</span> <span class="o">/</span><span class="n">content</span><span class="o">/</span><span class="n">counts</span><span class="o">/</span><span class="n">count</span><span class="o">.</span><span class="n">sh</span>
<span class="err">!</span><span class="o">./</span><span class="n">count</span><span class="o">.</span><span class="n">sh</span>
</code></pre></td></tr></table>
</div>
</div></li>
</ul>
    </div>

    
    


    
    

    <footer class="post-footer">
      <div class="post-tags">
          <a href="/tags/python/">Python</a>
          <a href="/tags/bioinformatics/">Bioinformatics</a>
          <a href="/tags/colab/">Colab</a>
          <a href="/tags/linux/">Linux</a>
          <a href="/tags/rna-sequence/">RNA Sequence</a>
          
        </div>

      
      <nav class="post-nav">
        
          <a class="prev" href="/post/ucscmap/">
            
            <i class="iconfont">
              <svg  class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="18" height="18">
  <path d="M691.908486 949.511495l75.369571-89.491197c10.963703-12.998035 10.285251-32.864502-1.499144-44.378743L479.499795 515.267417 757.434875 204.940602c11.338233-12.190647 11.035334-32.285311-0.638543-44.850487l-80.46666-86.564541c-11.680017-12.583596-30.356378-12.893658-41.662889-0.716314L257.233596 494.235404c-11.332093 12.183484-11.041474 32.266891 0.657986 44.844348l80.46666 86.564541c1.772366 1.910513 3.706415 3.533476 5.750981 4.877077l306.620399 321.703933C662.505829 963.726242 680.945807 962.528973 691.908486 949.511495z"></path>
</svg>

            </i>
            <span class="prev-text nav-default">UCSC Topics</span>
            <span class="prev-text nav-mobile">Prev</span>
          </a>
        
          <a class="next" href="/post/condacolab/">
            <span class="next-text nav-default">Installing Conda in Google Colab</span>
            <span class="prev-text nav-mobile">Next</span>
            
            <i class="iconfont">
              <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="18" height="18">
  <path d="M332.091514 74.487481l-75.369571 89.491197c-10.963703 12.998035-10.285251 32.864502 1.499144 44.378743l286.278095 300.375162L266.565125 819.058374c-11.338233 12.190647-11.035334 32.285311 0.638543 44.850487l80.46666 86.564541c11.680017 12.583596 30.356378 12.893658 41.662889 0.716314l377.434212-421.426145c11.332093-12.183484 11.041474-32.266891-0.657986-44.844348l-80.46666-86.564541c-1.772366-1.910513-3.706415-3.533476-5.750981-4.877077L373.270379 71.774697C361.493148 60.273758 343.054193 61.470003 332.091514 74.487481z"></path>
</svg>

            </i>
          </a>
      </nav>
    </footer>
  </article>

  
  

  
  

  

  
  

  

  

  

    

  

        </div>
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="icon-links">
  
  
    <a href="mailto:xiaoni0601@gmail.com" rel="me noopener" class="iconfont"
      title="email" >
      <svg class="icon" viewBox="0 0 1451 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="36" height="36">
  <path d="M664.781909 681.472759 0 97.881301C0 3.997201 71.046997 0 71.046997 0L474.477909 0 961.649408 0 1361.641813 0C1361.641813 0 1432.688811 3.997201 1432.688811 97.881301L771.345323 681.472759C771.345323 681.472759 764.482731 685.154773 753.594283 688.65053L753.594283 688.664858C741.602731 693.493018 729.424896 695.068979 718.077952 694.839748 706.731093 695.068979 694.553173 693.493018 682.561621 688.664858L682.561621 688.65053C671.644501 685.140446 664.781909 681.472759 664.781909 681.472759L664.781909 681.472759ZM718.063616 811.603883C693.779541 811.016482 658.879232 802.205449 619.10784 767.734955 542.989056 701.759633 0 212.052267 0 212.052267L0 942.809523C0 942.809523 0 1024 83.726336 1024L682.532949 1024 753.579947 1024 1348.948139 1024C1432.688811 1024 1432.688811 942.809523 1432.688811 942.809523L1432.688811 212.052267C1432.688811 212.052267 893.138176 701.759633 817.019477 767.734955 777.248 802.205449 742.347691 811.03081 718.063616 811.603883L718.063616 811.603883Z"></path>
</svg>

    </a>
  
    <a href="https://github.com/xiaonilee" rel="me noopener" class="iconfont"
      title="github"  target="_blank"
      >
      <svg class="icon" style="" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="36" height="36">
  <path d="M512 12.672c-282.88 0-512 229.248-512 512 0 226.261333 146.688 418.133333 350.08 485.76 25.6 4.821333 34.986667-11.008 34.986667-24.618667 0-12.16-0.426667-44.373333-0.64-87.04-142.421333 30.890667-172.458667-68.693333-172.458667-68.693333C188.672 770.986667 155.008 755.2 155.008 755.2c-46.378667-31.744 3.584-31.104 3.584-31.104 51.413333 3.584 78.421333 52.736 78.421333 52.736 45.653333 78.293333 119.850667 55.68 149.12 42.581333 4.608-33.109333 17.792-55.68 32.426667-68.48-113.706667-12.8-233.216-56.832-233.216-253.013333 0-55.893333 19.84-101.546667 52.693333-137.386667-5.76-12.928-23.04-64.981333 4.48-135.509333 0 0 42.88-13.738667 140.8 52.48 40.96-11.392 84.48-17.024 128-17.28 43.52 0.256 87.04 5.888 128 17.28 97.28-66.218667 140.16-52.48 140.16-52.48 27.52 70.528 10.24 122.581333 5.12 135.509333 32.64 35.84 52.48 81.493333 52.48 137.386667 0 196.693333-119.68 240-233.6 252.586667 17.92 15.36 34.56 46.762667 34.56 94.72 0 68.522667-0.64 123.562667-0.64 140.202666 0 13.44 8.96 29.44 35.2 24.32C877.44 942.592 1024 750.592 1024 524.672c0-282.752-229.248-512-512-512"></path>
</svg>

    </a>


<a href="/index.xml" rel="noopener alternate" type="application/rss&#43;xml"
    class="iconfont" title="rss" target="_blank">
    <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="30" height="30">
  <path d="M819.157333 1024C819.157333 574.592 449.408 204.8 0 204.8V0c561.706667 0 1024 462.293333 1024 1024h-204.842667zM140.416 743.04a140.8 140.8 0 0 1 140.501333 140.586667A140.928 140.928 0 0 1 140.074667 1024C62.72 1024 0 961.109333 0 883.626667s62.933333-140.544 140.416-140.586667zM678.784 1024h-199.04c0-263.210667-216.533333-479.786667-479.744-479.786667V345.173333c372.352 0 678.784 306.517333 678.784 678.826667z"></path>
</svg>

  </a>
   
</div>

<div class="copyright">

  <span class="copyright-year">
    &copy;
    
      2020 -
    2021
    <span class="heart">
      
      <i class="iconfont">
        <svg class="icon" viewBox="0 0 1025 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="14" height="14">
  <path d="M1000.1 247.9c-15.5-37.3-37.6-70.6-65.7-98.9-54.4-54.8-125.8-85-201-85-85.7 0-166 39-221.4 107.4C456.6 103 376.3 64 290.6 64c-75.1 0-146.5 30.4-201.1 85.6-28.2 28.5-50.4 61.9-65.8 99.3-16 38.8-24 79.9-23.6 122.2 0.7 91.7 40.1 177.2 108.1 234.8 3.1 2.6 6 5.1 8.9 7.8 14.9 13.4 58 52.8 112.6 102.7 93.5 85.5 209.9 191.9 257.5 234.2 7 6.1 15.8 9.5 24.9 9.5 9.2 0 18.1-3.4 24.9-9.5 34.5-30.7 105.8-95.9 181.4-165 74.2-67.8 150.9-138 195.8-178.2 69.5-57.9 109.6-144.4 109.9-237.3 0.1-42.5-8-83.6-24-122.2z"
   fill="#8a8a8a"></path>
</svg>

      </i>
    </span><span class="author">
        xiaonilee
        
      </span></span>

  
  

  
</div>

    </footer>

    <div class="back-to-top" id="back-to-top">
      <i class="iconfont">
        
        <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="35" height="35">
  <path d="M510.866688 227.694839 95.449397 629.218702l235.761562 0-2.057869 328.796468 362.40389 0L691.55698 628.188232l241.942331-3.089361L510.866688 227.694839zM63.840492 63.962777l894.052392 0 0 131.813095L63.840492 195.775872 63.840492 63.962777 63.840492 63.962777zM63.840492 63.962777"></path>
</svg>

      </i>
    </div>
  </div>
  
<script type="text/javascript" src="/lib/jquery/jquery-3.2.1.min.js"></script>
  <script type="text/javascript" src="/lib/slideout/slideout-1.0.1.min.js"></script>




<script type="text/javascript" src="/js/main.638251f4230630f0335d8c6748e53a96f94b72670920b60c09a56fdc8bece214.js" integrity="sha256-Y4JR9CMGMPAzXYxnSOU6lvlLcmcJILYMCaVv3Ivs4hQ=" crossorigin="anonymous"></script>



  <script type="text/javascript">
    window.MathJax = {
      showProcessingMessages: false,
      messageStyle: 'none'
    };
  </script>
  <script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-MML-AM_CHTML' async></script>









  
    <script type="text/javascript" src="/js/load-photoswipe.js"></script>
    <script type="text/javascript" src="/lib/photoswipe/photoswipe.min.js"></script>
    <script type="text/javascript" src="/lib/photoswipe/photoswipe-ui-default.min.js"></script>
  















</body>
</html>
