<!DOCTYPE html>
<html lang="en" itemscope itemtype="http://schema.org/WebPage">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Chapter 17. Classification - xiaoni&#39;s blog</title>
  

<meta name="renderer" content="webkit" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>

<meta name="MobileOptimized" content="width"/>
<meta name="HandheldFriendly" content="true"/>


<meta name="applicable-device" content="pc,mobile">

<meta name="theme-color" content="#f8f5ec" />
<meta name="msapplication-navbutton-color" content="#f8f5ec">
<meta name="apple-mobile-web-app-capable" content="yes">
<meta name="apple-mobile-web-app-status-bar-style" content="#f8f5ec">

<meta name="mobile-web-app-capable" content="yes">

<meta name="author" content="Xiaoni" />
  <meta name="description" content="Notebook of Reading Books: R in Action_Chapter 17.
" />

  <meta name="keywords" content="xiaonili, xiaonilee, xiaoni" />






<meta name="generator" content="Hugo 0.79.0" />


<link rel="canonical" href="/post/rinaction17/" />





<link rel="icon" href="/favicon.ico" />











<link rel="stylesheet" href="/sass/jane.min.f1e506a781bf25d33ffc18aa6b4e972a965c58049d27d4f92b7db2e9bf28e4bf.css" integrity="sha256-8eUGp4G/JdM//Biqa06XKpZcWASdJ9T5K32y6b8o5L8=" media="screen" crossorigin="anonymous">



<link href="https://cdn.bootcdn.net/ajax/libs/font-awesome/5.15.1/css/all.css" rel="stylesheet">
<script src="https://cdn.bootcdn.net/ajax/libs/font-awesome/5.15.1/js/all.js"></script>




<meta property="og:title" content="Chapter 17. Classification" />
<meta property="og:description" content="Notebook of Reading Books: R in Action_Chapter 17." />
<meta property="og:type" content="article" />
<meta property="og:url" content="/post/rinaction17/" />
<meta property="article:published_time" content="2020-11-13T00:00:00+00:00" />
<meta property="article:modified_time" content="2020-11-13T00:00:00+00:00" />
<meta itemprop="name" content="Chapter 17. Classification">
<meta itemprop="description" content="Notebook of Reading Books: R in Action_Chapter 17.">
<meta itemprop="datePublished" content="2020-11-13T00:00:00+00:00" />
<meta itemprop="dateModified" content="2020-11-13T00:00:00+00:00" />
<meta itemprop="wordCount" content="2768">



<meta itemprop="keywords" content="R,R in Action,Bioinformatics,Book," />
<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="Chapter 17. Classification"/>
<meta name="twitter:description" content="Notebook of Reading Books: R in Action_Chapter 17."/>

<!--[if lte IE 9]>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/classlist/1.1.20170427/classList.min.js"></script>
<![endif]-->

<!--[if lt IE 9]>
  <script src="https://cdn.jsdelivr.net/npm/html5shiv@3.7.3/dist/html5shiv.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/respond.js@1.4.2/dest/respond.min.js"></script>
<![endif]-->


<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'UA-177860608-1', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>



</head>
<body>
  <div id="mobile-navbar" class="mobile-navbar">
  <div class="mobile-header-logo">
    <a href="/" class="logo">Xiaoni's Blog</a>
  </div>
  <div class="mobile-navbar-icon">
    <span></span>
    <span></span>
    <span></span>
  </div>
</div>
<nav id="mobile-menu" class="mobile-menu slideout-menu">
  <ul class="mobile-menu-list">
    <li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/">Home</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/post/">Archives</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/categories/">Categories</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/tags/">Tags</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="/about/">About Me</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          <div class="mobile-menu-parent">
            <span class="mobile-submenu-open"></span>
            <a href="/">
              docs
            </a>
          </div>
          <ul class="mobile-submenu-list">
            
              <li>
                <a href="/post/tensorflow2/"> ML Notebook: TensorFlow2</a>
              </li>
            
              <li>
                <a href="/post/bioscirep/">Bioinformatics analysis of prognostic value of PITX1 gene in breast cancer</a>
              </li>
            
              <li>
                <a href="/post/rinactionmap/">Book: R in Action</a>
              </li>
            
              <li>
                <a href="/post/tumorimmune/">Comprehensive Bioinformatics Analysis of Tumor-Infiltrating Immune Cells</a>
              </li>
            
              <li>
                <a href="/post/firebrowse/">FireBrowse Notebook</a>
              </li>
            
              <li>
                <a href="/post/gdc/">GDC Notebook</a>
              </li>
            
              <li>
                <a href="/post/gscalite/">GSCALite: Gene Set Cancer Analysis</a>
              </li>
            
              <li>
                <a href="/post/frontoncol/">Identification of Therapeutic Targets and Prognostic Biomarkers Among CXC Chemokines in the Renal Cell Carcinoma Microenvironment</a>
              </li>
            
              <li>
                <a href="/post/condacolab/">Installing Conda in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/localxenadatahub/">Local UCSC Xena Data Hub</a>
              </li>
            
              <li>
                <a href="/post/oncolnc/">OncoLnc: A straightforward Tool</a>
              </li>
            
              <li>
                <a href="/post/webcrawler/">Python Spider</a>
              </li>
            
              <li>
                <a href="/post/hmwk10/">R Notebook : Application Exercises</a>
              </li>
            
              <li>
                <a href="/post/ggplot2/">R Notebook : ggplot2</a>
              </li>
            
              <li>
                <a href="/post/tricktcga/">R re-analysis in TCGA</a>
              </li>
            
              <li>
                <a href="/post/rnasequenceanalysis/">RNA-seq Bioinformatics - bioinformatics for RNA sequence analysis</a>
              </li>
            
              <li>
                <a href="/post/survival/">Survival Analysis with data from GSE database</a>
              </li>
            
              <li>
                <a href="/post/rincolab/">The way to use R in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/tissgdb/">TissGDB: tissue-specific gene database in cancer</a>
              </li>
            
              <li>
                <a href="/post/ucscgenebrowser/">UCSC gene browser with R package Gviz - part 1</a>
              </li>
            
              <li>
                <a href="/post/ucscgenebrowser2/">UCSC gene browser with R package Gviz - part 2</a>
              </li>
            
              <li>
                <a href="/post/genomicsignatures/">UCSC Xena Features: Genomic Signatures</a>
              </li>
            
              <li>
                <a href="/post/loaddata/">UCSC Xena: Load Data From Public Database</a>
              </li>
            
              <li>
                <a href="/post/privatepublictogether/">UCSC Xena: Private and Public Data Put Together</a>
              </li>
            
              <li>
                <a href="/post/webfrontend/">web front end</a>
              </li>
            
          </ul>
        
      </li>
    

    
  </ul>
</nav>


  
    






  <link rel="stylesheet" href="/lib/photoswipe/photoswipe.min.css" />
  <link rel="stylesheet" href="/lib/photoswipe/default-skin/default-skin.min.css" />




<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

<div class="pswp__bg"></div>

<div class="pswp__scroll-wrap">
    
    <div class="pswp__container">
      <div class="pswp__item"></div>
      <div class="pswp__item"></div>
      <div class="pswp__item"></div>
    </div>
    
    <div class="pswp__ui pswp__ui--hidden">
    <div class="pswp__top-bar">
      
      <div class="pswp__counter"></div>
      <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>
      <button class="pswp__button pswp__button--share" title="Share"></button>
      <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>
      <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>
      
      
      <div class="pswp__preloader">
        <div class="pswp__preloader__icn">
          <div class="pswp__preloader__cut">
            <div class="pswp__preloader__donut"></div>
          </div>
        </div>
      </div>
    </div>
    <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
      <div class="pswp__share-tooltip"></div>
    </div>
    <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
    </button>
    <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
    </button>
    <div class="pswp__caption">
      <div class="pswp__caption__center"></div>
    </div>
    </div>
    </div>
</div>

  

  

  

  <header id="header" class="header container">
    <div class="logo-wrapper">
  <a href="/" class="logo">
    
      Xiaoni's Blog
    
  </a>
</div>

<nav class="site-navbar">
  <ul id="menu" class="menu">
    
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/">Home</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/post/">Archives</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/categories/">Categories</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/tags/">Tags</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="/about/">About Me</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          <a class="menu-item-link menu-parent" href="/">docs</a>
          <ul class="submenu" >
            
              <li>
                <a href="/post/tensorflow2/"> ML Notebook: TensorFlow2</a>
              </li>
            
              <li>
                <a href="/post/bioscirep/">Bioinformatics analysis of prognostic value of PITX1 gene in breast cancer</a>
              </li>
            
              <li>
                <a href="/post/rinactionmap/">Book: R in Action</a>
              </li>
            
              <li>
                <a href="/post/tumorimmune/">Comprehensive Bioinformatics Analysis of Tumor-Infiltrating Immune Cells</a>
              </li>
            
              <li>
                <a href="/post/firebrowse/">FireBrowse Notebook</a>
              </li>
            
              <li>
                <a href="/post/gdc/">GDC Notebook</a>
              </li>
            
              <li>
                <a href="/post/gscalite/">GSCALite: Gene Set Cancer Analysis</a>
              </li>
            
              <li>
                <a href="/post/frontoncol/">Identification of Therapeutic Targets and Prognostic Biomarkers Among CXC Chemokines in the Renal Cell Carcinoma Microenvironment</a>
              </li>
            
              <li>
                <a href="/post/condacolab/">Installing Conda in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/localxenadatahub/">Local UCSC Xena Data Hub</a>
              </li>
            
              <li>
                <a href="/post/oncolnc/">OncoLnc: A straightforward Tool</a>
              </li>
            
              <li>
                <a href="/post/webcrawler/">Python Spider</a>
              </li>
            
              <li>
                <a href="/post/hmwk10/">R Notebook : Application Exercises</a>
              </li>
            
              <li>
                <a href="/post/ggplot2/">R Notebook : ggplot2</a>
              </li>
            
              <li>
                <a href="/post/tricktcga/">R re-analysis in TCGA</a>
              </li>
            
              <li>
                <a href="/post/rnasequenceanalysis/">RNA-seq Bioinformatics - bioinformatics for RNA sequence analysis</a>
              </li>
            
              <li>
                <a href="/post/survival/">Survival Analysis with data from GSE database</a>
              </li>
            
              <li>
                <a href="/post/rincolab/">The way to use R in Google Colab</a>
              </li>
            
              <li>
                <a href="/post/tissgdb/">TissGDB: tissue-specific gene database in cancer</a>
              </li>
            
              <li>
                <a href="/post/ucscgenebrowser/">UCSC gene browser with R package Gviz - part 1</a>
              </li>
            
              <li>
                <a href="/post/ucscgenebrowser2/">UCSC gene browser with R package Gviz - part 2</a>
              </li>
            
              <li>
                <a href="/post/genomicsignatures/">UCSC Xena Features: Genomic Signatures</a>
              </li>
            
              <li>
                <a href="/post/loaddata/">UCSC Xena: Load Data From Public Database</a>
              </li>
            
              <li>
                <a href="/post/privatepublictogether/">UCSC Xena: Private and Public Data Put Together</a>
              </li>
            
              <li>
                <a href="/post/webfrontend/">web front end</a>
              </li>
            
          </ul>

        

      </li>
    

    
    

    
  </ul>
</nav>

  </header>

  <div id="mobile-panel">
    <main id="main" class="main bg-llight">
      <div class="content-wrapper">
        <div id="content" class="content container">
          <article class="post bg-white">
    
    <header class="post-header">
      <h1 class="post-title">Chapter 17. Classification</h1>
      
      <div class="post-meta">
        <time datetime="2020-11-13" class="post-time">
          2020-11-13
        </time>
        <div class="post-category">
            <a href="/categories/r/"> R </a>
            <a href="/categories/bioinformatics/"> Bioinformatics </a>
            
          </div>
        

        
        

        
        
      </div>
    </header>

    
    
<div class="post-toc" id="post-toc">
  <h2 class="post-toc-title">Table of Contents</h2>
  <div class="post-toc-content">
    <nav id="TableOfContents">
  <ul>
    <li><a href="#this-chapter-covers">This chapter covers</a>
      <ul>
        <li><a href="#171-preparing-the-data">17.1 Preparing the data</a></li>
        <li><a href="#172-logistic-regression">17.2 Logistic regression</a></li>
        <li><a href="#173-decision-trees">17.3. Decision trees</a></li>
        <li><a href="#174-random-forests">17.4 Random forests</a></li>
        <li><a href="#175-support-vector-machines">17.5. Support vector machines</a></li>
        <li><a href="#177-using-the-rattle-package-for-data-mining">17.7 Using the rattle package for data mining</a></li>
      </ul>
    </li>
  </ul>
</nav>
  </div>
</div>

    
    <div class="post-content">
      <p>Notebook of Reading Books: R in Action_Chapter 17.</p>
<h2 id="this-chapter-covers">This chapter covers</h2>
<ul>
<li>
<p>Classifying with decision trees</p>
</li>
<li>
<p>Ensemble classification with random forests</p>
</li>
<li>
<p>Creating a support vector machine</p>
</li>
<li>
<p>Evaluating classification accuracy</p>
</li>
</ul>
<h3 id="171-preparing-the-data">17.1 Preparing the data</h3>
<ul>
<li>Data from the UCI Machine Learning Server
<ul>
<li><a href="http://archive.ics.uci.edu/ml/index.php">http://archive.ics.uci.edu/ml/index.php</a></li>
<li>Breast Cancer Wisconsin (Original) Data Set
<ul>
<li><a href="http://archive.ics.uci.edu/ml/datasets/Breast+Cancer+Wisconsin+%28Original%29">http://archive.ics.uci.edu/ml/datasets/Breast+Cancer+Wisconsin+%28Original%29</a></li>
</ul>
</li>
</ul>
</li>
</ul>
<h3 id="172-logistic-regression">17.2 Logistic regression</h3>
<ul>
<li><code>glm()</code> &amp; <code>predict()</code></li>
</ul>
<h3 id="173-decision-trees">17.3. Decision trees</h3>
<h4 id="1731-classical-decision-trees">17.3.1. Classical decision trees</h4>
<ul>
<li>
<p>Figure 17.1. Complexity parameter vs. cross-validated error.</p>
<ul>
<li>The dotted line is the upper limit of the one standard deviation rule <code>(0.12 + 1 * 0.026 = .15)</code></li>
<li>The plot suggests selecting the tree with the leftmost cp value below the line.</li>
</ul>
<p><img src="fig171.png" alt="fig171"></p>
</li>
<li>
<p>Figure 17.2. Traditional (pruned) decision tree for predicting cancer status.</p>
<ul>
<li>Start at the top of the tree, moving left if a condition is true or right otherwise.</li>
<li>When an observation hits a terminal node, it’s classified. Each node contains the probability of the classes in that node, along with the percentage of the sample.</li>
</ul>
<p><img src="fig172.png" alt="fig172"></p>
</li>
</ul>
<h4 id="1732-conditional-inference-trees">17.3.2 Conditional inference trees</h4>
<ul>
<li>
<p>Figure 17.3. Conditional inference tree for the breast cancer data.</p>
<p><img src="fig173.png" alt="fig173"></p>
</li>
</ul>
<h3 id="174-random-forests">17.4 Random forests</h3>
<h3 id="175-support-vector-machines">17.5. Support vector machines</h3>
<h4 id="176-choosing-a-best-predictive-solution">17.6. Choosing a best predictive solution</h4>
<p><img src="tab171.png" alt="tab171"></p>
<ul>
<li>define performance() function</li>
</ul>
<h3 id="177-using-the-rattle-package-for-data-mining">17.7 Using the rattle package for data mining</h3>
<ul>
<li>Installing rattle on MacOS 10.11 (or above) with <a href="install_rattle.R">R Script</a>.</li>
</ul>
<p>Show me the code <i class="far fa-hand-pointer"></i></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1"># Installing rattle on MacOS 10.11 (or above)</span>

<span class="c1"># install gtk+ 2.24.32_3</span>
<span class="nf">system</span><span class="p">(</span><span class="s">&#39;brew install gtk+&#39;</span><span class="p">)</span>


<span class="c1"># macOS version:catalina </span>
<span class="nf">local</span><span class="p">({</span>
  <span class="nf">if </span><span class="p">(</span><span class="nf">Sys.info</span><span class="p">()</span><span class="n">[[</span><span class="s">&#39;sysname&#39;</span><span class="n">]]</span> <span class="o">!=</span> <span class="s">&#39;Darwin&#39;</span><span class="p">)</span> <span class="nf">return</span><span class="p">()</span>
  
  <span class="n">.Platform</span><span class="o">$</span><span class="n">pkgType</span> <span class="o">=</span> <span class="s">&#39;mac.binary.catalina&#39;</span>
  <span class="nf">unlockBinding</span><span class="p">(</span><span class="s">&#39;.Platform&#39;</span><span class="p">,</span> <span class="nf">baseenv</span><span class="p">())</span>
  <span class="nf">assign</span><span class="p">(</span><span class="s">&#39;.Platform&#39;</span><span class="p">,</span> <span class="n">.Platform</span><span class="p">,</span> <span class="s">&#39;package:base&#39;</span><span class="p">)</span>
  <span class="nf">lockBinding</span><span class="p">(</span><span class="s">&#39;.Platform&#39;</span><span class="p">,</span> <span class="nf">baseenv</span><span class="p">())</span>
  
  <span class="nf">options</span><span class="p">(</span>
    <span class="n">pkgType</span> <span class="o">=</span> <span class="s">&#39;both&#39;</span><span class="p">,</span> <span class="n">install.packages.compile.from.source</span> <span class="o">=</span> <span class="s">&#39;always&#39;</span><span class="p">,</span>
    <span class="n">repos</span> <span class="o">=</span> <span class="s">&#39;https://macos.rbind.io&#39;</span>
  <span class="p">)</span>
<span class="p">})</span>

<span class="nf">install.packages</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;RGtk2&#39;</span><span class="p">,</span> <span class="s">&#39;cairoDevice&#39;</span><span class="p">,</span> <span class="s">&#39;rattle&#39;</span><span class="p">))</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="rattleUI.png" alt="rattleUI"></p>
<ul>
<li>
<p>Load and setup data:</p>
<p><img src="rattleUI20.png" alt="rattleUI2"></p>
</li>
<li>
<p>Execute</p>
<p><img src="rattleUI2.png" alt="rattleUI22"></p>
</li>
<li>
<p>model(up) and draw(bottom)</p>
<p><img src="model4.png" alt="model">
<img src="fig179.png" alt="fig179"></p>
</li>
<li>
<p>Evaluate</p>
<p><img src="evaluate5.png" alt="evaluate5"></p>
</li>
</ul>
<p>Attach is the <a href="chapter17.R">Script</a> of chapter17.</p>
<p>Show me the code <i class="far fa-hand-pointer"></i></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">  1
</span><span class="lnt">  2
</span><span class="lnt">  3
</span><span class="lnt">  4
</span><span class="lnt">  5
</span><span class="lnt">  6
</span><span class="lnt">  7
</span><span class="lnt">  8
</span><span class="lnt">  9
</span><span class="lnt"> 10
</span><span class="lnt"> 11
</span><span class="lnt"> 12
</span><span class="lnt"> 13
</span><span class="lnt"> 14
</span><span class="lnt"> 15
</span><span class="lnt"> 16
</span><span class="lnt"> 17
</span><span class="lnt"> 18
</span><span class="lnt"> 19
</span><span class="lnt"> 20
</span><span class="lnt"> 21
</span><span class="lnt"> 22
</span><span class="lnt"> 23
</span><span class="lnt"> 24
</span><span class="lnt"> 25
</span><span class="lnt"> 26
</span><span class="lnt"> 27
</span><span class="lnt"> 28
</span><span class="lnt"> 29
</span><span class="lnt"> 30
</span><span class="lnt"> 31
</span><span class="lnt"> 32
</span><span class="lnt"> 33
</span><span class="lnt"> 34
</span><span class="lnt"> 35
</span><span class="lnt"> 36
</span><span class="lnt"> 37
</span><span class="lnt"> 38
</span><span class="lnt"> 39
</span><span class="lnt"> 40
</span><span class="lnt"> 41
</span><span class="lnt"> 42
</span><span class="lnt"> 43
</span><span class="lnt"> 44
</span><span class="lnt"> 45
</span><span class="lnt"> 46
</span><span class="lnt"> 47
</span><span class="lnt"> 48
</span><span class="lnt"> 49
</span><span class="lnt"> 50
</span><span class="lnt"> 51
</span><span class="lnt"> 52
</span><span class="lnt"> 53
</span><span class="lnt"> 54
</span><span class="lnt"> 55
</span><span class="lnt"> 56
</span><span class="lnt"> 57
</span><span class="lnt"> 58
</span><span class="lnt"> 59
</span><span class="lnt"> 60
</span><span class="lnt"> 61
</span><span class="lnt"> 62
</span><span class="lnt"> 63
</span><span class="lnt"> 64
</span><span class="lnt"> 65
</span><span class="lnt"> 66
</span><span class="lnt"> 67
</span><span class="lnt"> 68
</span><span class="lnt"> 69
</span><span class="lnt"> 70
</span><span class="lnt"> 71
</span><span class="lnt"> 72
</span><span class="lnt"> 73
</span><span class="lnt"> 74
</span><span class="lnt"> 75
</span><span class="lnt"> 76
</span><span class="lnt"> 77
</span><span class="lnt"> 78
</span><span class="lnt"> 79
</span><span class="lnt"> 80
</span><span class="lnt"> 81
</span><span class="lnt"> 82
</span><span class="lnt"> 83
</span><span class="lnt"> 84
</span><span class="lnt"> 85
</span><span class="lnt"> 86
</span><span class="lnt"> 87
</span><span class="lnt"> 88
</span><span class="lnt"> 89
</span><span class="lnt"> 90
</span><span class="lnt"> 91
</span><span class="lnt"> 92
</span><span class="lnt"> 93
</span><span class="lnt"> 94
</span><span class="lnt"> 95
</span><span class="lnt"> 96
</span><span class="lnt"> 97
</span><span class="lnt"> 98
</span><span class="lnt"> 99
</span><span class="lnt">100
</span><span class="lnt">101
</span><span class="lnt">102
</span><span class="lnt">103
</span><span class="lnt">104
</span><span class="lnt">105
</span><span class="lnt">106
</span><span class="lnt">107
</span><span class="lnt">108
</span><span class="lnt">109
</span><span class="lnt">110
</span><span class="lnt">111
</span><span class="lnt">112
</span><span class="lnt">113
</span><span class="lnt">114
</span><span class="lnt">115
</span><span class="lnt">116
</span><span class="lnt">117
</span><span class="lnt">118
</span><span class="lnt">119
</span><span class="lnt">120
</span><span class="lnt">121
</span><span class="lnt">122
</span><span class="lnt">123
</span><span class="lnt">124
</span><span class="lnt">125
</span><span class="lnt">126
</span><span class="lnt">127
</span><span class="lnt">128
</span><span class="lnt">129
</span><span class="lnt">130
</span><span class="lnt">131
</span><span class="lnt">132
</span><span class="lnt">133
</span><span class="lnt">134
</span><span class="lnt">135
</span><span class="lnt">136
</span><span class="lnt">137
</span><span class="lnt">138
</span><span class="lnt">139
</span><span class="lnt">140
</span><span class="lnt">141
</span><span class="lnt">142
</span><span class="lnt">143
</span><span class="lnt">144
</span><span class="lnt">145
</span><span class="lnt">146
</span><span class="lnt">147
</span><span class="lnt">148
</span><span class="lnt">149
</span><span class="lnt">150
</span><span class="lnt">151
</span><span class="lnt">152
</span><span class="lnt">153
</span><span class="lnt">154
</span><span class="lnt">155
</span><span class="lnt">156
</span><span class="lnt">157
</span><span class="lnt">158
</span><span class="lnt">159
</span><span class="lnt">160
</span><span class="lnt">161
</span><span class="lnt">162
</span><span class="lnt">163
</span><span class="lnt">164
</span><span class="lnt">165
</span><span class="lnt">166
</span><span class="lnt">167
</span><span class="lnt">168
</span><span class="lnt">169
</span><span class="lnt">170
</span><span class="lnt">171
</span><span class="lnt">172
</span><span class="lnt">173
</span><span class="lnt">174
</span><span class="lnt">175
</span><span class="lnt">176
</span><span class="lnt">177
</span><span class="lnt">178
</span><span class="lnt">179
</span><span class="lnt">180
</span><span class="lnt">181
</span><span class="lnt">182
</span><span class="lnt">183
</span><span class="lnt">184
</span><span class="lnt">185
</span><span class="lnt">186
</span><span class="lnt">187
</span><span class="lnt">188
</span><span class="lnt">189
</span><span class="lnt">190
</span><span class="lnt">191
</span><span class="lnt">192
</span><span class="lnt">193
</span><span class="lnt">194
</span><span class="lnt">195
</span><span class="lnt">196
</span><span class="lnt">197
</span><span class="lnt">198
</span><span class="lnt">199
</span><span class="lnt">200
</span><span class="lnt">201
</span><span class="lnt">202
</span><span class="lnt">203
</span><span class="lnt">204
</span><span class="lnt">205
</span><span class="lnt">206
</span><span class="lnt">207
</span><span class="lnt">208
</span><span class="lnt">209
</span><span class="lnt">210
</span><span class="lnt">211
</span><span class="lnt">212
</span><span class="lnt">213
</span><span class="lnt">214
</span><span class="lnt">215
</span><span class="lnt">216
</span><span class="lnt">217
</span><span class="lnt">218
</span><span class="lnt">219
</span><span class="lnt">220
</span><span class="lnt">221
</span><span class="lnt">222
</span><span class="lnt">223
</span><span class="lnt">224
</span><span class="lnt">225
</span><span class="lnt">226
</span><span class="lnt">227
</span><span class="lnt">228
</span><span class="lnt">229
</span><span class="lnt">230
</span><span class="lnt">231
</span><span class="lnt">232
</span><span class="lnt">233
</span><span class="lnt">234
</span><span class="lnt">235
</span><span class="lnt">236
</span><span class="lnt">237
</span><span class="lnt">238
</span><span class="lnt">239
</span><span class="lnt">240
</span><span class="lnt">241
</span><span class="lnt">242
</span><span class="lnt">243
</span><span class="lnt">244
</span><span class="lnt">245
</span><span class="lnt">246
</span><span class="lnt">247
</span><span class="lnt">248
</span><span class="lnt">249
</span><span class="lnt">250
</span><span class="lnt">251
</span><span class="lnt">252
</span><span class="lnt">253
</span><span class="lnt">254
</span><span class="lnt">255
</span><span class="lnt">256
</span><span class="lnt">257
</span><span class="lnt">258
</span><span class="lnt">259
</span><span class="lnt">260
</span><span class="lnt">261
</span><span class="lnt">262
</span><span class="lnt">263
</span><span class="lnt">264
</span><span class="lnt">265
</span><span class="lnt">266
</span><span class="lnt">267
</span><span class="lnt">268
</span><span class="lnt">269
</span><span class="lnt">270
</span><span class="lnt">271
</span><span class="lnt">272
</span><span class="lnt">273
</span><span class="lnt">274
</span><span class="lnt">275
</span><span class="lnt">276
</span><span class="lnt">277
</span><span class="lnt">278
</span><span class="lnt">279
</span><span class="lnt">280
</span><span class="lnt">281
</span><span class="lnt">282
</span><span class="lnt">283
</span><span class="lnt">284
</span><span class="lnt">285
</span><span class="lnt">286
</span><span class="lnt">287
</span><span class="lnt">288
</span><span class="lnt">289
</span><span class="lnt">290
</span><span class="lnt">291
</span><span class="lnt">292
</span><span class="lnt">293
</span><span class="lnt">294
</span><span class="lnt">295
</span><span class="lnt">296
</span><span class="lnt">297
</span><span class="lnt">298
</span><span class="lnt">299
</span><span class="lnt">300
</span><span class="lnt">301
</span><span class="lnt">302
</span><span class="lnt">303
</span><span class="lnt">304
</span><span class="lnt">305
</span><span class="lnt">306
</span><span class="lnt">307
</span><span class="lnt">308
</span><span class="lnt">309
</span><span class="lnt">310
</span><span class="lnt">311
</span><span class="lnt">312
</span><span class="lnt">313
</span><span class="lnt">314
</span><span class="lnt">315
</span><span class="lnt">316
</span><span class="lnt">317
</span><span class="lnt">318
</span><span class="lnt">319
</span><span class="lnt">320
</span><span class="lnt">321
</span><span class="lnt">322
</span><span class="lnt">323
</span><span class="lnt">324
</span><span class="lnt">325
</span><span class="lnt">326
</span><span class="lnt">327
</span><span class="lnt">328
</span><span class="lnt">329
</span><span class="lnt">330
</span><span class="lnt">331
</span><span class="lnt">332
</span><span class="lnt">333
</span><span class="lnt">334
</span><span class="lnt">335
</span><span class="lnt">336
</span><span class="lnt">337
</span><span class="lnt">338
</span><span class="lnt">339
</span><span class="lnt">340
</span><span class="lnt">341
</span><span class="lnt">342
</span><span class="lnt">343
</span><span class="lnt">344
</span><span class="lnt">345
</span><span class="lnt">346
</span><span class="lnt">347
</span><span class="lnt">348
</span><span class="lnt">349
</span><span class="lnt">350
</span><span class="lnt">351
</span><span class="lnt">352
</span><span class="lnt">353
</span><span class="lnt">354
</span><span class="lnt">355
</span><span class="lnt">356
</span><span class="lnt">357
</span><span class="lnt">358
</span><span class="lnt">359
</span><span class="lnt">360
</span><span class="lnt">361
</span><span class="lnt">362
</span><span class="lnt">363
</span><span class="lnt">364
</span><span class="lnt">365
</span><span class="lnt">366
</span><span class="lnt">367
</span><span class="lnt">368
</span><span class="lnt">369
</span><span class="lnt">370
</span><span class="lnt">371
</span><span class="lnt">372
</span><span class="lnt">373
</span><span class="lnt">374
</span><span class="lnt">375
</span><span class="lnt">376
</span><span class="lnt">377
</span><span class="lnt">378
</span><span class="lnt">379
</span><span class="lnt">380
</span><span class="lnt">381
</span><span class="lnt">382
</span><span class="lnt">383
</span><span class="lnt">384
</span><span class="lnt">385
</span><span class="lnt">386
</span><span class="lnt">387
</span><span class="lnt">388
</span><span class="lnt">389
</span><span class="lnt">390
</span><span class="lnt">391
</span><span class="lnt">392
</span><span class="lnt">393
</span><span class="lnt">394
</span><span class="lnt">395
</span><span class="lnt">396
</span><span class="lnt">397
</span><span class="lnt">398
</span><span class="lnt">399
</span><span class="lnt">400
</span><span class="lnt">401
</span><span class="lnt">402
</span><span class="lnt">403
</span><span class="lnt">404
</span><span class="lnt">405
</span><span class="lnt">406
</span><span class="lnt">407
</span><span class="lnt">408
</span><span class="lnt">409
</span><span class="lnt">410
</span><span class="lnt">411
</span><span class="lnt">412
</span><span class="lnt">413
</span><span class="lnt">414
</span><span class="lnt">415
</span><span class="lnt">416
</span><span class="lnt">417
</span><span class="lnt">418
</span><span class="lnt">419
</span><span class="lnt">420
</span><span class="lnt">421
</span><span class="lnt">422
</span><span class="lnt">423
</span><span class="lnt">424
</span><span class="lnt">425
</span><span class="lnt">426
</span><span class="lnt">427
</span><span class="lnt">428
</span><span class="lnt">429
</span><span class="lnt">430
</span><span class="lnt">431
</span><span class="lnt">432
</span><span class="lnt">433
</span><span class="lnt">434
</span><span class="lnt">435
</span><span class="lnt">436
</span><span class="lnt">437
</span><span class="lnt">438
</span><span class="lnt">439
</span><span class="lnt">440
</span><span class="lnt">441
</span><span class="lnt">442
</span><span class="lnt">443
</span><span class="lnt">444
</span><span class="lnt">445
</span><span class="lnt">446
</span><span class="lnt">447
</span><span class="lnt">448
</span><span class="lnt">449
</span><span class="lnt">450
</span><span class="lnt">451
</span><span class="lnt">452
</span><span class="lnt">453
</span><span class="lnt">454
</span><span class="lnt">455
</span><span class="lnt">456
</span><span class="lnt">457
</span><span class="lnt">458
</span><span class="lnt">459
</span><span class="lnt">460
</span><span class="lnt">461
</span><span class="lnt">462
</span><span class="lnt">463
</span><span class="lnt">464
</span><span class="lnt">465
</span><span class="lnt">466
</span><span class="lnt">467
</span><span class="lnt">468
</span><span class="lnt">469
</span><span class="lnt">470
</span><span class="lnt">471
</span><span class="lnt">472
</span><span class="lnt">473
</span><span class="lnt">474
</span><span class="lnt">475
</span><span class="lnt">476
</span><span class="lnt">477
</span><span class="lnt">478
</span><span class="lnt">479
</span><span class="lnt">480
</span><span class="lnt">481
</span><span class="lnt">482
</span><span class="lnt">483
</span><span class="lnt">484
</span><span class="lnt">485
</span><span class="lnt">486
</span><span class="lnt">487
</span><span class="lnt">488
</span><span class="lnt">489
</span><span class="lnt">490
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1"># Remove most objects from the working environment</span>
<span class="nf">rm</span><span class="p">(</span><span class="n">list</span> <span class="o">=</span> <span class="nf">ls</span><span class="p">())</span>
<span class="nf">options</span><span class="p">(</span><span class="n">stringsAsFactors</span> <span class="o">=</span> <span class="bp">F</span><span class="p">)</span>

<span class="c1"># Prerequisites</span>
<span class="n">pkgs</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;rpart&#34;</span><span class="p">,</span> <span class="s">&#34;rpart.plot&#34;</span><span class="p">,</span> <span class="s">&#34;party&#34;</span><span class="p">,</span>
          <span class="s">&#34;randomForest&#34;</span><span class="p">,</span> <span class="s">&#34;e1071&#34;</span><span class="p">)</span>
<span class="nf">install.packages</span><span class="p">(</span><span class="n">pkgs</span><span class="p">,</span> <span class="n">depend</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>


<span class="c1"># 17.1 Preparing the data</span>
<span class="c1"># code listing 17.1. Preparing the breast cancer data</span>
<span class="n">loc</span> <span class="o">&lt;-</span> <span class="s">&#34;http://archive.ics.uci.edu/ml/machine-learning-databases/&#34;</span>
<span class="n">ds</span> <span class="o">&lt;-</span> <span class="s">&#34;breast-cancer-wisconsin/breast-cancer-wisconsin.data&#34;</span>
<span class="n">url</span> <span class="o">&lt;-</span> <span class="nf">paste</span><span class="p">(</span><span class="n">loc</span><span class="p">,</span> <span class="n">ds</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&#34;&#34;</span><span class="p">)</span>
<span class="n">url</span>


<span class="n">breast</span> <span class="o">&lt;-</span> <span class="nf">read.table</span><span class="p">(</span><span class="n">url</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&#34;,&#34;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">na.strings</span> <span class="o">=</span> <span class="s">&#34;?&#34;</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">breast</span><span class="p">)</span>
<span class="c1">#        V1 V2 V3 V4 V5 V6 V7 V8 V9 V10 V11</span>
<span class="c1"># 1 1000025  5  1  1  1  2  1  3  1   1   2</span>
<span class="c1"># 2 1002945  5  4  4  5  7 10  3  2   1   2</span>
<span class="c1"># 3 1015425  3  1  1  1  2  2  3  1   1   2</span>
<span class="c1"># 4 1016277  6  8  8  1  3  4  3  7   1   2</span>
<span class="c1"># 5 1017023  4  1  1  3  2  1  3  1   1   2</span>
<span class="c1"># 6 1017122  8 10 10  8  7 10  9  7   1   4</span>


<span class="nf">names</span><span class="p">(</span><span class="n">breast</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;ID&#34;</span><span class="p">,</span> <span class="s">&#34;clumpThickness&#34;</span><span class="p">,</span> <span class="s">&#34;sizeUniformity&#34;</span><span class="p">,</span>
                   <span class="s">&#34;shapeUniformity&#34;</span><span class="p">,</span> <span class="s">&#34;maginalAdhesion&#34;</span><span class="p">,</span>
                   <span class="s">&#34;singleEpithelialCellSize&#34;</span><span class="p">,</span> <span class="s">&#34;bareNuclei&#34;</span><span class="p">,</span>
                   <span class="s">&#34;blandChromatin&#34;</span><span class="p">,</span> <span class="s">&#34;normalNucleoli&#34;</span><span class="p">,</span> <span class="s">&#34;mitosis&#34;</span><span class="p">,</span> <span class="s">&#34;class&#34;</span><span class="p">)</span>

<span class="nf">table</span><span class="p">(</span><span class="n">breast</span><span class="o">$</span><span class="n">class</span><span class="p">)</span>
<span class="c1">#   2   4 </span>
<span class="c1"># 458 241</span>

<span class="n">df</span> <span class="o">&lt;-</span> <span class="n">breast[</span><span class="m">-1</span><span class="n">]</span>

<span class="nf">nrow</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="c1"># [1] 699</span>

<span class="n">df</span><span class="o">$</span><span class="n">class</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">class</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span> <span class="m">4</span><span class="p">),</span>
                   <span class="n">labels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;benign&#34;</span><span class="p">,</span> <span class="s">&#34;malignant&#34;</span><span class="p">))</span>
<span class="nf">set.seed</span><span class="p">(</span><span class="m">1234</span><span class="p">)</span>
<span class="n">train</span> <span class="o">&lt;-</span> <span class="nf">sample</span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">df</span><span class="p">),</span> <span class="m">0.7</span><span class="o">*</span><span class="nf">nrow</span><span class="p">(</span><span class="n">df</span><span class="p">))</span>
<span class="n">df.train</span> <span class="o">&lt;-</span> <span class="n">df[train</span><span class="p">,</span><span class="n">]</span>
<span class="n">df.validate</span> <span class="o">&lt;-</span> <span class="n">df[</span><span class="o">-</span><span class="n">train</span><span class="p">,</span><span class="n">]</span>
<span class="nf">table</span><span class="p">(</span><span class="n">df.train</span><span class="o">$</span><span class="n">class</span><span class="p">)</span>
<span class="c1"># benign malignant </span>
<span class="c1">#   319       170</span>

<span class="nf">table</span><span class="p">(</span><span class="n">df.validate</span><span class="o">$</span><span class="n">class</span><span class="p">)</span>
<span class="c1"># benign malignant </span>
<span class="c1">#   139        71</span>


<span class="c1">#=========================================================</span>
<span class="c1"># 17.2 Logistic regression</span>
<span class="c1"># code listing 17.2. Logistic regression with glm()</span>
<span class="c1">## Fits the logistic regression with df.train dataset</span>
<span class="n">fit.logit</span> <span class="o">&lt;-</span> <span class="nf">glm</span><span class="p">(</span><span class="n">class</span> <span class="o">~</span> <span class="n">.,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">df.train</span><span class="p">,</span> <span class="n">family</span> <span class="o">=</span> <span class="nf">binomial</span><span class="p">())</span>

<span class="c1">##Examines the model</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">fit.logit</span><span class="p">)</span>
<span class="c1"># Call:</span>
<span class="c1"># glm(formula = class ~ ., family = binomial(), data = df.train)</span>
<span class="c1"># </span>
<span class="c1"># Deviance Residuals: </span>
<span class="c1">#      Min        1Q    Median        3Q       Max  </span>
<span class="c1"># -2.24605  -0.08012  -0.03110   0.00266   2.11576  </span>
<span class="c1"># </span>
<span class="c1"># Coefficients:</span>
<span class="c1">#                          Estimate Std. Error z value Pr(&gt;|z|)    </span>
<span class="c1"># (Intercept)              -12.4412     2.0547  -6.055  1.4e-09 ***</span>
<span class="c1"># clumpThickness             0.7407     0.2262   3.275  0.00106 ** </span>
<span class="c1"># sizeUniformity            -0.0320     0.3399  -0.094  0.92500    </span>
<span class="c1"># shapeUniformity            0.2073     0.3715   0.558  0.57680    </span>
<span class="c1"># maginalAdhesion            0.5194     0.1708   3.041  0.00236 ** </span>
<span class="c1"># singleEpithelialCellSize  -0.3217     0.2613  -1.231  0.21831    </span>
<span class="c1"># bareNuclei                 0.5851     0.1881   3.111  0.00187 ** </span>
<span class="c1"># blandChromatin             0.8599     0.2923   2.942  0.00326 ** </span>
<span class="c1"># normalNucleoli             0.4036     0.1828   2.208  0.02725 *  </span>
<span class="c1"># mitosis                    0.8923     0.3552   2.512  0.01200 *  </span>
<span class="c1"># ---</span>
<span class="c1"># Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1</span>
<span class="c1"># </span>
<span class="c1"># (Dispersion parameter for binomial family taken to be 1)</span>
<span class="c1"># </span>
<span class="c1">#     Null deviance: 621.04  on 477  degrees of freedom</span>
<span class="c1"># Residual deviance:  52.39  on 468  degrees of freedom</span>
<span class="c1">#   (11 observations deleted due to missingness)</span>
<span class="c1"># AIC: 72.39</span>
<span class="c1"># </span>
<span class="c1"># Number of Fisher Scoring iterations: 9</span>

<span class="c1">## Classifies new cases with df.validate dataset</span>
<span class="n">prob</span> <span class="o">&lt;-</span> <span class="nf">predict</span><span class="p">(</span><span class="n">fit.logit</span><span class="p">,</span> <span class="n">df.validate</span><span class="p">,</span> <span class="n">type</span> <span class="o">=</span> <span class="s">&#34;response&#34;</span><span class="p">)</span>
<span class="n">logit.pred</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">prob</span> <span class="o">&gt;</span> <span class="m">.5</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="kc">FALSE</span><span class="p">,</span> <span class="kc">TRUE</span><span class="p">),</span>
                     <span class="n">labels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;benign&#34;</span><span class="p">,</span> <span class="s">&#34;malignant&#34;</span><span class="p">))</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">logit.pred</span><span class="p">)</span>
<span class="c1">#    benign malignant      NA&#39;s </span>
<span class="c1">#       130        75         5 </span>


<span class="c1">## Evaluates the predictive accuracy</span>
<span class="n">logit.perf</span> <span class="o">&lt;-</span> <span class="nf">table</span><span class="p">(</span><span class="n">df.validate</span><span class="o">$</span><span class="n">class</span><span class="p">,</span> <span class="n">logit.pred</span><span class="p">,</span>
                    <span class="n">dnn</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;Actual&#34;</span><span class="p">,</span> <span class="s">&#34;Predicted&#34;</span><span class="p">))</span>
<span class="n">logit.perf</span>
<span class="c1">#            Predicted</span>
<span class="c1"># Actual      benign malignant</span>
<span class="c1">#   benign       129         6</span>
<span class="c1">#   malignant      1        69</span>

<span class="nf">summary</span><span class="p">(</span><span class="n">logit.perf</span><span class="p">)</span>
<span class="c1"># Number of cases in table: 205 </span>
<span class="c1"># Number of factors: 2 </span>
<span class="c1"># Test for independence of all factors:</span>
<span class="c1">#   Chisq = 176.04, df = 1, p-value = 3.55e-40</span>
<span class="p">(</span><span class="m">129+69</span><span class="p">)</span><span class="o">/</span><span class="p">(</span><span class="m">129+6+1+69</span><span class="p">)</span><span class="o">*</span><span class="m">100</span>
<span class="c1"># [1] 96.58537</span>



<span class="n">logit.fit.reduced</span> <span class="o">&lt;-</span> <span class="nf">step</span><span class="p">(</span><span class="n">fit.logit</span><span class="p">)</span>

<span class="c1"># Start:  AIC=72.39</span>
<span class="c1"># class ~ clumpThickness + sizeUniformity + shapeUniformity + maginalAdhesion + </span>
<span class="c1">#   singleEpithelialCellSize + bareNuclei + blandChromatin + </span>
<span class="c1">#   normalNucleoli + mitosis</span>
<span class="c1"># </span>
<span class="c1"># Df Deviance    AIC</span>
<span class="c1"># - sizeUniformity            1   52.399 70.399</span>
<span class="c1"># - shapeUniformity           1   52.686 70.686</span>
<span class="c1"># - singleEpithelialCellSize  1   53.910 71.910</span>
<span class="c1"># &lt;none&gt;                          52.390 72.390</span>
<span class="c1"># - normalNucleoli            1   57.465 75.465</span>
<span class="c1"># - mitosis                   1   57.966 75.966</span>
<span class="c1"># - blandChromatin            1   62.856 80.856</span>
<span class="c1"># - maginalAdhesion           1   63.044 81.044</span>
<span class="c1"># - bareNuclei                1   64.982 82.982</span>
<span class="c1"># - clumpThickness            1   68.323 86.323</span>
<span class="c1"># </span>
<span class="c1"># Step:  AIC=70.4</span>
<span class="c1"># class ~ clumpThickness + shapeUniformity + maginalAdhesion + </span>
<span class="c1">#   singleEpithelialCellSize + bareNuclei + blandChromatin + </span>
<span class="c1">#   normalNucleoli + mitosis</span>
<span class="c1"># </span>
<span class="c1"># Df Deviance    AIC</span>
<span class="c1"># - shapeUniformity           1   52.876 68.876</span>
<span class="c1"># - singleEpithelialCellSize  1   53.918 69.918</span>
<span class="c1"># &lt;none&gt;                          52.399 70.399</span>
<span class="c1"># - normalNucleoli            1   57.916 73.916</span>
<span class="c1"># - mitosis                   1   58.024 74.024</span>
<span class="c1"># - blandChromatin            1   63.272 79.272</span>
<span class="c1"># - maginalAdhesion           1   63.735 79.735</span>
<span class="c1"># - bareNuclei                1   64.985 80.985</span>
<span class="c1"># - clumpThickness            1   68.728 84.728</span>
<span class="c1"># </span>
<span class="c1"># Step:  AIC=68.88</span>
<span class="c1"># class ~ clumpThickness + maginalAdhesion + singleEpithelialCellSize + </span>
<span class="c1">#   bareNuclei + blandChromatin + normalNucleoli + mitosis</span>
<span class="c1"># </span>
<span class="c1"># Df Deviance    AIC</span>
<span class="c1"># - singleEpithelialCellSize  1   54.154 68.154</span>
<span class="c1"># &lt;none&gt;                          52.876 68.876</span>
<span class="c1"># - mitosis                   1   59.402 73.402</span>
<span class="c1"># - normalNucleoli            1   60.929 74.929</span>
<span class="c1"># - blandChromatin            1   64.053 78.053</span>
<span class="c1"># - maginalAdhesion           1   64.995 78.995</span>
<span class="c1"># - bareNuclei                1   75.634 89.634</span>
<span class="c1"># - clumpThickness            1   76.942 90.942</span>
<span class="c1"># </span>
<span class="c1"># Step:  AIC=68.15</span>
<span class="c1"># class ~ clumpThickness + maginalAdhesion + bareNuclei + blandChromatin + </span>
<span class="c1">#   normalNucleoli + mitosis</span>
<span class="c1"># </span>
<span class="c1"># Df Deviance    AIC</span>
<span class="c1"># &lt;none&gt;                 54.154 68.154</span>
<span class="c1"># - mitosis          1   60.177 72.177</span>
<span class="c1"># - normalNucleoli   1   60.937 72.937</span>
<span class="c1"># - blandChromatin   1   64.056 76.056</span>
<span class="c1"># - maginalAdhesion  1   65.022 77.022</span>
<span class="c1"># - bareNuclei       1   76.417 88.417</span>
<span class="c1"># - clumpThickness   1   77.027 89.027</span>


<span class="c1">#========================================================================</span>
<span class="c1"># 17.3.1. Classical decision trees</span>
<span class="c1"># code listing Creating a classical decision tree with rpart()</span>
<span class="nf">library</span><span class="p">(</span><span class="n">rpart</span><span class="p">)</span>
<span class="nf">set.seed</span><span class="p">(</span><span class="m">1234</span><span class="p">)</span>
<span class="n">dtree</span> <span class="o">&lt;-</span> <span class="nf">rpart</span><span class="p">(</span><span class="n">class</span> <span class="o">~</span> <span class="n">.,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">df.train</span><span class="p">,</span> <span class="n">method</span> <span class="o">=</span> <span class="s">&#34;class&#34;</span><span class="p">,</span>
               <span class="n">parms</span> <span class="o">=</span> <span class="nf">list</span><span class="p">(</span><span class="n">split</span><span class="o">=</span><span class="s">&#34;information&#34;</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="n">dtree</span><span class="p">)</span>
<span class="c1"># n= 489 </span>
<span class="c1"># </span>
<span class="c1"># node), split, n, loss, yval, (yprob)</span>
<span class="c1"># * denotes terminal node</span>
<span class="c1"># </span>
<span class="c1"># 1) root 489 170 benign (0.65235174 0.34764826)  </span>
<span class="c1"># 2) sizeUniformity&lt; 2.5 304   8 benign (0.97368421 0.02631579)  </span>
<span class="c1"># 4) clumpThickness&lt; 6.5 297   3 benign (0.98989899 0.01010101) *</span>
<span class="c1">#   5) clumpThickness&gt;=6.5 7   2 malignant (0.28571429 0.71428571) *</span>
<span class="c1">#   3) sizeUniformity&gt;=2.5 185  23 malignant (0.12432432 0.87567568)  </span>
<span class="c1"># 6) bareNuclei&lt; 2.5 28  13 benign (0.53571429 0.46428571)  </span>
<span class="c1"># 12) sizeUniformity&lt; 3.5 16   1 benign (0.93750000 0.06250000) *</span>
<span class="c1">#   13) sizeUniformity&gt;=3.5 12   0 malignant (0.00000000 1.00000000) *</span>
<span class="c1">#   7) bareNuclei&gt;=2.5 157   8 malignant (0.05095541 0.94904459) *</span>


<span class="n">dtree</span><span class="o">$</span><span class="n">cptable</span>
<span class="c1">#           CP nsplit  rel error    xerror       xstd</span>
<span class="c1"># 1 0.81764706      0 1.00000000 1.0000000 0.06194645</span>
<span class="c1"># 2 0.04117647      1 0.18235294 0.1823529 0.03169642</span>
<span class="c1"># 3 0.01764706      3 0.10000000 0.1588235 0.02970979</span>
<span class="c1"># 4 0.01000000      4 0.08235294 0.1235294 0.02637116</span>

<span class="nf">plotcp</span><span class="p">(</span><span class="n">dtree</span><span class="p">)</span> <span class="c1"># figure 17.1</span>

<span class="n">dtree.pruned</span> <span class="o">&lt;-</span> <span class="nf">prune</span><span class="p">(</span><span class="n">dtree</span><span class="p">,</span> <span class="n">cp</span><span class="o">=</span><span class="m">0.01764706</span><span class="p">)</span>

<span class="nf">library</span><span class="p">(</span><span class="n">rpart.plot</span><span class="p">)</span>
<span class="nf">prp</span><span class="p">(</span><span class="n">dtree.pruned</span><span class="p">,</span> <span class="n">type</span> <span class="o">=</span> <span class="m">2</span><span class="p">,</span> <span class="n">extra</span> <span class="o">=</span> <span class="m">104</span><span class="p">,</span>
    <span class="n">fallen.leaves</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">main</span><span class="o">=</span><span class="s">&#34;Decision Tree&#34;</span><span class="p">)</span> <span class="c1"># figure 17.2</span>

<span class="n">dtree.pred</span> <span class="o">&lt;-</span> <span class="nf">predict</span><span class="p">(</span><span class="n">dtree.pruned</span><span class="p">,</span> <span class="n">df.validate</span><span class="p">,</span> <span class="n">type</span> <span class="o">=</span> <span class="s">&#34;class&#34;</span><span class="p">)</span>
<span class="n">dtree.perf</span> <span class="o">&lt;-</span> <span class="nf">table</span><span class="p">(</span><span class="n">df.validate</span><span class="o">$</span><span class="n">class</span><span class="p">,</span> <span class="n">dtree.pred</span><span class="p">,</span>
                    <span class="n">dnn</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;Actual&#34;</span><span class="p">,</span> <span class="s">&#34;Predicted&#34;</span><span class="p">))</span>
<span class="n">dtree.perf</span>
<span class="c1">#            Predicted</span>
<span class="c1"># Actual      benign malignant</span>
<span class="c1"># benign        129        10</span>
<span class="c1"># malignant       4        67</span>

<span class="nf">summary</span><span class="p">(</span><span class="n">dtree.perf</span><span class="p">)</span>
<span class="c1"># Number of cases in table: 210 </span>
<span class="c1"># Number of factors: 2 </span>
<span class="c1"># Test for independence of all factors:</span>
<span class="c1">#   Chisq = 153.78, df = 1, p-value = 2.585e-35</span>

<span class="p">(</span><span class="m">129+67</span><span class="p">)</span><span class="o">/</span><span class="p">(</span><span class="m">129+10+4+67</span><span class="p">)</span><span class="o">*</span><span class="m">100</span>
<span class="c1"># [1] 93.33333</span>


<span class="c1"># 17.3.2 Conditional inference trees</span>
<span class="nf">library</span><span class="p">(</span><span class="n">party</span><span class="p">)</span>
<span class="n">fit.ctree</span> <span class="o">&lt;-</span> <span class="nf">ctree</span><span class="p">(</span><span class="n">class</span> <span class="o">~</span> <span class="n">.,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">df.train</span><span class="p">)</span>
<span class="nf">plot</span><span class="p">(</span><span class="n">fit.ctree</span><span class="p">,</span> <span class="n">main</span><span class="o">=</span><span class="s">&#34;Conditional Inference Tree&#34;</span><span class="p">)</span>

<span class="n">ctree.pred</span> <span class="o">&lt;-</span> <span class="nf">predict</span><span class="p">(</span><span class="n">fit.ctree</span><span class="p">,</span> <span class="n">df.validate</span><span class="p">,</span> <span class="n">type</span><span class="o">=</span><span class="s">&#34;response&#34;</span><span class="p">)</span>
<span class="n">ctree.perf</span> <span class="o">&lt;-</span> <span class="nf">table</span><span class="p">(</span><span class="n">df.validate</span><span class="o">$</span><span class="n">class</span><span class="p">,</span> <span class="n">ctree.pred</span><span class="p">,</span>
                    <span class="n">dnn</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&#34;Actual&#34;</span><span class="p">,</span> <span class="s">&#34;Predicted&#34;</span><span class="p">))</span>
<span class="n">ctree.perf</span>
<span class="c1">#            Predicted</span>
<span class="c1"># Actual      benign malignant</span>
<span class="c1"># benign         131         8</span>
<span class="c1"># malignant        4        67</span>

<span class="nf">summary</span><span class="p">(</span><span class="n">ctree.perf</span><span class="p">)</span>
<span class="c1"># Number of cases in table: 210 </span>
<span class="c1"># Number of factors: 2 </span>
<span class="c1"># Test for independence of all factors:</span>
<span class="c1">#   Chisq = 160.72, df = 1, p-value = 7.875e-37</span>

<span class="p">(</span><span class="m">131+67</span><span class="p">)</span><span class="o">/</span><span class="p">(</span><span class="m">131+4+67+8</span><span class="p">)</span><span class="o">*</span><span class="m">100</span>
<span class="c1"># [1] 94.28571</span>

<span class="c1"># 17.4 Random forests</span>
<span class="c1"># code listing 17.5. Random forest</span>
<span class="nf">library</span><span class="p">(</span><span class="n">randomForest</span><span class="p">)</span>
<span class="nf">set.seed</span><span class="p">(</span><span class="m">1234</span><span class="p">)</span>
<span class="n">fit.forest</span> <span class="o">&lt;-</span> <span class="nf">randomForest</span><span class="p">(</span><span class="n">class</span> <span class="o">~</span> <span class="n">.,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">df.train</span><span class="p">,</span>
                           <span class="n">na.action</span> <span class="o">=</span> <span class="n">na.roughfix</span><span class="p">,</span>
                           <span class="n">importance</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="n">fit.forest</span>
<span class="c1"># Call:</span>
<span class="c1"># randomForest(formula = class ~ ., data = df.train, importance = TRUE,      na.action = na.roughfix) </span>
<span class="c1">#                Type of random forest: classification</span>
<span class="c1">#                      Number of trees: 500</span>
<span class="c1"># No. of variables tried at each split: 3</span>
<span class="c1"># </span>
<span class="c1">#       OOB estimate of  error rate: 2.66%</span>
<span class="c1"># Confusion matrix:</span>
<span class="c1">#           benign malignant class.error</span>
<span class="c1"># benign       313         6  0.01880878</span>
<span class="c1"># malignant      7       163  0.04117647</span>

<span class="nf">importance</span><span class="p">(</span><span class="n">fit.forest</span><span class="p">,</span> <span class="n">type</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span>
<span class="c1">#                          MeanDecreaseGini</span>
<span class="c1"># clumpThickness                  11.382432</span>
<span class="c1"># sizeUniformity                  63.037519</span>
<span class="c1"># shapeUniformity                 49.027649</span>
<span class="c1"># maginalAdhesion                  4.275345</span>
<span class="c1"># singleEpithelialCellSize        14.504981</span>
<span class="c1"># bareNuclei                      42.736364</span>
<span class="c1"># blandChromatin                  22.484755</span>
<span class="c1"># normalNucleoli                  11.375285</span>
<span class="c1"># mitosis                          1.755382</span>


<span class="n">forest.pred</span> <span class="o">&lt;-</span> <span class="nf">predict</span><span class="p">(</span><span class="n">fit.forest</span><span class="p">,</span> <span class="n">df.validate</span><span class="p">)</span>
<span class="n">forest.perf</span> <span class="o">&lt;-</span> <span class="nf">table</span><span class="p">(</span><span class="n">df.validate</span><span class="o">$</span><span class="n">class</span><span class="p">,</span> <span class="n">forest.pred</span><span class="p">,</span>
                     <span class="n">dnn</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;Actual&#34;</span><span class="p">,</span> <span class="s">&#34;Predicted&#34;</span><span class="p">))</span>
<span class="n">forest.perf</span>
<span class="c1">#            Predicted</span>
<span class="c1"># Actual      benign malignant</span>
<span class="c1">#   benign      128         7</span>
<span class="c1">#   malignant     2        68</span>

<span class="nf">summary</span><span class="p">(</span><span class="n">forest.perf</span><span class="p">)</span>
<span class="c1"># Number of cases in table: 205 </span>
<span class="c1"># Number of factors: 2 </span>
<span class="c1"># Test for independence of all factors:</span>
<span class="c1">#   Chisq = 168.02, df = 1, p-value = 2.004e-38</span>

<span class="p">(</span><span class="m">128+68</span><span class="p">)</span><span class="o">/</span><span class="p">(</span><span class="m">128+7+2+68</span><span class="p">)</span><span class="o">*</span><span class="m">100</span>
<span class="c1"># [1] 95.60976</span>

<span class="c1"># 17.5. Support vector machines</span>
<span class="c1"># code listing 17.6. A support vector machine</span>
<span class="nf">library</span><span class="p">(</span><span class="n">e1071</span><span class="p">)</span>
<span class="nf">set.seed</span><span class="p">(</span><span class="m">1234</span><span class="p">)</span>
<span class="n">fit.svm</span> <span class="o">&lt;-</span> <span class="nf">svm</span><span class="p">(</span><span class="n">class</span> <span class="o">~</span> <span class="n">.,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">df.train</span><span class="p">)</span>
<span class="n">fit.svm</span>
<span class="c1"># Call:</span>
<span class="c1"># svm(formula = class ~ ., data = df.train)</span>
<span class="c1"># </span>
<span class="c1"># </span>
<span class="c1"># Parameters:</span>
<span class="c1">#   SVM-Type:  C-classification </span>
<span class="c1"># SVM-Kernel:  radial </span>
<span class="c1">#       cost:  1 </span>
<span class="c1"># </span>
<span class="c1"># Number of Support Vectors:  74</span>


<span class="n">svm.pred</span> <span class="o">&lt;-</span> <span class="nf">predict</span><span class="p">(</span><span class="n">fit.svm</span><span class="p">,</span> <span class="nf">na.omit</span><span class="p">(</span><span class="n">df.validate</span><span class="p">))</span>
<span class="n">svm.perf</span> <span class="o">&lt;-</span> <span class="nf">table</span><span class="p">(</span><span class="nf">na.omit</span><span class="p">(</span><span class="n">df.validate</span><span class="p">)</span><span class="o">$</span><span class="n">class</span><span class="p">,</span>
                  <span class="n">svm.pred</span><span class="p">,</span> <span class="n">dnn</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;Actual&#34;</span><span class="p">,</span> <span class="s">&#34;Predicted&#34;</span><span class="p">))</span>
<span class="n">svm.perf</span>
<span class="c1">#            Predicted</span>
<span class="c1"># Actual      benign malignant</span>
<span class="c1">#   benign       126         9</span>
<span class="c1">#   malignant      0        70</span>


<span class="c1"># code listing Tuning an RBF support vector machine</span>
<span class="nf">set.seed</span><span class="p">(</span><span class="m">1234</span><span class="p">)</span>
<span class="n">tuned</span> <span class="o">&lt;-</span> <span class="nf">tune.svm</span><span class="p">(</span><span class="n">class</span> <span class="o">~</span> <span class="n">.,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">df.train</span><span class="p">,</span>
                  <span class="n">gamma</span> <span class="o">=</span> <span class="m">10</span><span class="nf">^</span><span class="p">(</span><span class="m">-6</span><span class="o">:</span><span class="m">1</span><span class="p">),</span>
                  <span class="n">cost</span> <span class="o">=</span> <span class="m">10</span><span class="nf">^</span><span class="p">(</span><span class="m">-10</span><span class="o">:</span><span class="m">10</span><span class="p">))</span>
<span class="n">tuned</span>
<span class="c1"># Parameter tuning of ‘svm’:</span>
<span class="c1">#   </span>
<span class="c1"># - sampling method: 10-fold cross validation </span>
<span class="c1"># </span>
<span class="c1"># - best parameters:</span>
<span class="c1">#   gamma cost</span>
<span class="c1">#    0.01    1</span>
<span class="c1"># </span>
<span class="c1"># - best performance: 0.02740302</span>


<span class="n">fit.svm</span> <span class="o">&lt;-</span> <span class="nf">svm</span><span class="p">(</span><span class="n">class</span> <span class="o">~</span> <span class="n">.,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">df.train</span><span class="p">,</span> <span class="n">gamma</span> <span class="o">=</span> <span class="m">.01</span><span class="p">,</span> <span class="n">cost</span> <span class="o">=</span> <span class="m">1</span><span class="p">)</span>
<span class="n">svm.pred</span> <span class="o">&lt;-</span> <span class="nf">predict</span><span class="p">(</span><span class="n">fit.svm</span><span class="p">,</span> <span class="nf">na.omit</span><span class="p">(</span><span class="n">df.validate</span><span class="p">))</span>
<span class="n">svm.perf</span> <span class="o">&lt;-</span> <span class="nf">table</span><span class="p">(</span><span class="nf">na.omit</span><span class="p">(</span><span class="n">df.validate</span><span class="p">)</span><span class="o">$</span><span class="n">class</span><span class="p">,</span>
                  <span class="n">svm.pred</span><span class="p">,</span> <span class="n">dnn</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;Actual&#34;</span><span class="p">,</span> <span class="s">&#34;Predicted&#34;</span><span class="p">))</span>
<span class="n">svm.perf</span>
<span class="c1">#            Predicted</span>
<span class="c1"># Actual      benign malignant</span>
<span class="c1">#   benign       128         7</span>
<span class="c1">#   malignant      0        70</span>

<span class="nf">summary</span><span class="p">(</span><span class="n">svm.perf</span><span class="p">)</span>
<span class="c1"># Number of cases in table: 205 </span>
<span class="c1"># Number of factors: 2 </span>
<span class="c1"># Test for independence of all factors:</span>
<span class="c1">#   Chisq = 176.7, df = 1, p-value = 2.546e-40</span>

<span class="p">(</span><span class="m">128+70</span><span class="p">)</span><span class="o">/</span><span class="p">(</span><span class="m">128+7+70</span><span class="p">)</span><span class="o">*</span><span class="m">100</span>
<span class="c1"># [1] 96.58537</span>

<span class="c1"># 17.6. Choosing a best predictive solution</span>
<span class="c1"># code listing 17.8. Function for assessing binary classification accuracy</span>
<span class="n">performance</span> <span class="o">&lt;-</span> <span class="nf">function</span><span class="p">(</span><span class="n">table</span><span class="p">,</span> <span class="n">n</span><span class="o">=</span><span class="m">2</span><span class="p">)</span> <span class="p">{</span>
  <span class="nf">if</span><span class="p">(</span><span class="o">!</span><span class="nf">all</span><span class="p">(</span><span class="nf">dim</span><span class="p">(</span><span class="n">table</span><span class="p">)</span> <span class="o">==</span> <span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="m">2</span><span class="p">)))</span>
    <span class="nf">stop</span><span class="p">(</span><span class="s">&#34;Must be a 2 x 2 table&#34;</span><span class="p">)</span>
  <span class="n">tn</span> <span class="o">=</span> <span class="n">table[1</span><span class="p">,</span><span class="m">1</span><span class="n">]</span>
  <span class="n">fp</span> <span class="o">=</span> <span class="n">table[1</span><span class="p">,</span><span class="m">2</span><span class="n">]</span>
  <span class="n">fn</span> <span class="o">=</span> <span class="n">table[2</span><span class="p">,</span><span class="m">1</span><span class="n">]</span>
  <span class="n">tp</span> <span class="o">=</span> <span class="n">table[2</span><span class="p">,</span><span class="m">2</span><span class="n">]</span>
  <span class="n">sensitivity</span> <span class="o">=</span> <span class="n">tp</span><span class="o">/</span><span class="p">(</span><span class="n">tp</span><span class="o">+</span><span class="n">fn</span><span class="p">)</span>
  <span class="n">specificity</span> <span class="o">=</span> <span class="n">tn</span><span class="o">/</span><span class="p">(</span><span class="n">tn</span><span class="o">+</span><span class="n">fp</span><span class="p">)</span>
  <span class="n">ppp</span> <span class="o">=</span> <span class="n">tp</span><span class="o">/</span><span class="p">(</span><span class="n">tp</span><span class="o">+</span><span class="n">fp</span><span class="p">)</span>
  <span class="n">npp</span> <span class="o">=</span> <span class="n">tn</span><span class="o">/</span><span class="p">(</span><span class="n">tn</span><span class="o">+</span><span class="n">fn</span><span class="p">)</span>
  <span class="n">hitrate</span> <span class="o">=</span> <span class="p">(</span><span class="n">tp</span><span class="o">+</span><span class="n">tn</span><span class="p">)</span><span class="o">/</span><span class="p">(</span><span class="n">tp</span><span class="o">+</span><span class="n">tn</span><span class="o">+</span><span class="n">fp</span><span class="o">+</span><span class="n">fn</span><span class="p">)</span>
  <span class="n">result</span> <span class="o">&lt;-</span> <span class="nf">paste</span><span class="p">(</span><span class="s">&#34;Sensitivity = &#34;</span><span class="p">,</span> <span class="nf">round</span><span class="p">(</span><span class="n">sensitivity</span><span class="p">,</span> <span class="n">n</span><span class="p">)</span> <span class="p">,</span>
                  <span class="s">&#34;\nSpecificity = &#34;</span><span class="p">,</span> <span class="nf">round</span><span class="p">(</span><span class="n">specificity</span><span class="p">,</span> <span class="n">n</span><span class="p">),</span>
                  <span class="s">&#34;\nPositive Predictive Value = &#34;</span><span class="p">,</span> <span class="nf">round</span><span class="p">(</span><span class="n">ppp</span><span class="p">,</span> <span class="n">n</span><span class="p">),</span>
                  <span class="s">&#34;\nNegative Predictive Value = &#34;</span><span class="p">,</span> <span class="nf">round</span><span class="p">(</span><span class="n">npp</span><span class="p">,</span> <span class="n">n</span><span class="p">),</span>
                  <span class="s">&#34;\nAccuracy = &#34;</span><span class="p">,</span> <span class="nf">round</span><span class="p">(</span><span class="n">hitrate</span><span class="p">,</span> <span class="n">n</span><span class="p">),</span> <span class="s">&#34;\n&#34;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&#34;&#34;</span><span class="p">)</span>
  <span class="nf">cat</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
  <span class="p">}</span>


<span class="c1"># code listing 17.9. Performance of breast cancer data classifiers</span>
<span class="nf">performance</span><span class="p">(</span><span class="n">logit.perf</span><span class="p">)</span>
<span class="c1"># Sensitivity = 0.99</span>
<span class="c1"># Specificity = 0.96</span>
<span class="c1"># Positive Predictive Value = 0.92</span>
<span class="c1"># Negative Predictive Value = 0.99</span>
<span class="c1"># Accuracy = 0.97</span>

<span class="nf">performance</span><span class="p">(</span><span class="n">dtree.perf</span><span class="p">)</span>
<span class="c1"># Sensitivity = 0.94</span>
<span class="c1"># Specificity = 0.93</span>
<span class="c1"># Positive Predictive Value = 0.87</span>
<span class="c1"># Negative Predictive Value = 0.97</span>
<span class="c1"># Accuracy = 0.93</span>

<span class="nf">performance</span><span class="p">(</span><span class="n">ctree.perf</span><span class="p">)</span>
<span class="c1"># Sensitivity = 0.94</span>
<span class="c1"># Specificity = 0.94</span>
<span class="c1"># Positive Predictive Value = 0.89</span>
<span class="c1"># Negative Predictive Value = 0.97</span>
<span class="c1"># Accuracy = 0.94</span>

<span class="nf">performance</span><span class="p">(</span><span class="n">forest.perf</span><span class="p">)</span>
<span class="c1"># Sensitivity = 0.97</span>
<span class="c1"># Specificity = 0.95</span>
<span class="c1"># Positive Predictive Value = 0.91</span>
<span class="c1"># Negative Predictive Value = 0.98</span>
<span class="c1"># Accuracy = 0.96</span>

<span class="nf">performance</span><span class="p">(</span><span class="n">svm.perf</span><span class="p">)</span>
<span class="c1"># Sensitivity = 1</span>
<span class="c1"># Specificity = 0.95</span>
<span class="c1"># Positive Predictive Value = 0.91</span>
<span class="c1"># Negative Predictive Value = 1</span>
<span class="c1"># Accuracy = 0.97</span>


<span class="c1">#==============================================================</span>
<span class="c1"># 17.7 Using the rattle package for data mining</span>
<span class="c1"># install.packages(&#34;rattle&#34;), detail in the &#34;install_rattle.R&#34;</span>


<span class="c1"># The Pima Indians Diabetes dataset is no longer available due to permission restrictions</span>
<span class="c1"># loc &lt;- &#34;http://archive.ics.uci.edu/ml/machine-learning-databases/&#34;</span>
<span class="c1"># ds &lt;- &#34;pima-indians-diabetes/pima-indians-diabetes.data&#34;</span>

<span class="c1"># url &lt;- paste(loc, ds, sep=&#34;&#34;)</span>
<span class="c1"># </span>
<span class="c1"># diabetes &lt;- read.table(url, sep=&#34;,&#34;, header=FALSE)</span>
<span class="c1"># names(diabetes) &lt;- c(&#34;npregant&#34;, &#34;plasma&#34;, &#34;bp&#34;, &#34;triceps&#34;,</span>
<span class="c1">#                      &#34;insulin&#34;, &#34;bmi&#34;, &#34;pedigree&#34;, &#34;age&#34;, &#34;class&#34;)</span>
<span class="c1"># diabetes$class &lt;- factor(diabetes$class, levels=c(0,1),</span>
<span class="c1">#                          labels=c(&#34;normal&#34;, &#34;diabetic&#34;))</span>


<span class="nf">library</span><span class="p">(</span><span class="n">rattle</span><span class="p">)</span>
<span class="c1"># Loading required package: tibble</span>
<span class="c1"># Loading required package: bitops</span>
<span class="c1"># Rattle: A free graphical interface for data science with R.</span>
<span class="c1"># Version 5.4.0 Copyright (c) 2006-2020 Togaware Pty Ltd.</span>
<span class="c1"># Type &#39;rattle()&#39; to shake, rattle, and roll your data.</span>
<span class="c1"># </span>
<span class="c1"># Attaching package: ‘rattle’</span>
<span class="c1"># </span>
<span class="c1"># The following object is masked from ‘package:randomForest’:</span>
<span class="c1">#   </span>
<span class="c1">#   importance</span>


<span class="nf">rattle</span><span class="p">()</span>
<span class="c1"># Loading required package: RGtk2</span>
<span class="c1"># 2020-11-14 11:03:35.908 rsession[43141:820420] *** WARNING: Method userSpaceScaleFactor </span>
<span class="c1"># in class NSView is deprecated on 10.7 and later. It should not be used in new </span>
<span class="c1"># applications. Use convertRectToBacking: instead.</span>
<span class="c1">#</span>
<span class="c1"># Attaching package: ‘zoo’</span>

<span class="c1"># The following objects are masked from ‘package:base’:</span>
  
<span class="c1">#   as.Date, as.Date.numeric</span>

<span class="c1"># export a Decision Tree model with pmml package.</span>
<span class="nf">install.packages</span><span class="p">(</span><span class="s">&#39;pmml&#39;</span><span class="p">)</span> 
</code></pre></td></tr></table>
</div>
</div>
    </div>

    
    


    
    

    <footer class="post-footer">
      <div class="post-tags">
          <a href="/tags/r/">R</a>
          <a href="/tags/r-in-action/">R in Action</a>
          <a href="/tags/bioinformatics/">Bioinformatics</a>
          <a href="/tags/book/">Book</a>
          
        </div>

      
      <nav class="post-nav">
        
          <a class="prev" href="/post/rinaction18/">
            
            <i class="iconfont">
              <svg  class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="18" height="18">
  <path d="M691.908486 949.511495l75.369571-89.491197c10.963703-12.998035 10.285251-32.864502-1.499144-44.378743L479.499795 515.267417 757.434875 204.940602c11.338233-12.190647 11.035334-32.285311-0.638543-44.850487l-80.46666-86.564541c-11.680017-12.583596-30.356378-12.893658-41.662889-0.716314L257.233596 494.235404c-11.332093 12.183484-11.041474 32.266891 0.657986 44.844348l80.46666 86.564541c1.772366 1.910513 3.706415 3.533476 5.750981 4.877077l306.620399 321.703933C662.505829 963.726242 680.945807 962.528973 691.908486 949.511495z"></path>
</svg>

            </i>
            <span class="prev-text nav-default">Chapter 18. Advanced methods for missing data</span>
            <span class="prev-text nav-mobile">Prev</span>
          </a>
        
          <a class="next" href="/post/rinaction16/">
            <span class="next-text nav-default">Chapter 16. Cluster analysis</span>
            <span class="prev-text nav-mobile">Next</span>
            
            <i class="iconfont">
              <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="18" height="18">
  <path d="M332.091514 74.487481l-75.369571 89.491197c-10.963703 12.998035-10.285251 32.864502 1.499144 44.378743l286.278095 300.375162L266.565125 819.058374c-11.338233 12.190647-11.035334 32.285311 0.638543 44.850487l80.46666 86.564541c11.680017 12.583596 30.356378 12.893658 41.662889 0.716314l377.434212-421.426145c11.332093-12.183484 11.041474-32.266891-0.657986-44.844348l-80.46666-86.564541c-1.772366-1.910513-3.706415-3.533476-5.750981-4.877077L373.270379 71.774697C361.493148 60.273758 343.054193 61.470003 332.091514 74.487481z"></path>
</svg>

            </i>
          </a>
      </nav>
    </footer>
  </article>

  
  

  
  

  

  
  

  

  

  

    

  

        </div>
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="icon-links">
  
  
    <a href="mailto:xiaoni0601@gmail.com" rel="me noopener" class="iconfont"
      title="email" >
      <svg class="icon" viewBox="0 0 1451 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="36" height="36">
  <path d="M664.781909 681.472759 0 97.881301C0 3.997201 71.046997 0 71.046997 0L474.477909 0 961.649408 0 1361.641813 0C1361.641813 0 1432.688811 3.997201 1432.688811 97.881301L771.345323 681.472759C771.345323 681.472759 764.482731 685.154773 753.594283 688.65053L753.594283 688.664858C741.602731 693.493018 729.424896 695.068979 718.077952 694.839748 706.731093 695.068979 694.553173 693.493018 682.561621 688.664858L682.561621 688.65053C671.644501 685.140446 664.781909 681.472759 664.781909 681.472759L664.781909 681.472759ZM718.063616 811.603883C693.779541 811.016482 658.879232 802.205449 619.10784 767.734955 542.989056 701.759633 0 212.052267 0 212.052267L0 942.809523C0 942.809523 0 1024 83.726336 1024L682.532949 1024 753.579947 1024 1348.948139 1024C1432.688811 1024 1432.688811 942.809523 1432.688811 942.809523L1432.688811 212.052267C1432.688811 212.052267 893.138176 701.759633 817.019477 767.734955 777.248 802.205449 742.347691 811.03081 718.063616 811.603883L718.063616 811.603883Z"></path>
</svg>

    </a>
  
    <a href="https://github.com/xiaonilee" rel="me noopener" class="iconfont"
      title="github"  target="_blank"
      >
      <svg class="icon" style="" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="36" height="36">
  <path d="M512 12.672c-282.88 0-512 229.248-512 512 0 226.261333 146.688 418.133333 350.08 485.76 25.6 4.821333 34.986667-11.008 34.986667-24.618667 0-12.16-0.426667-44.373333-0.64-87.04-142.421333 30.890667-172.458667-68.693333-172.458667-68.693333C188.672 770.986667 155.008 755.2 155.008 755.2c-46.378667-31.744 3.584-31.104 3.584-31.104 51.413333 3.584 78.421333 52.736 78.421333 52.736 45.653333 78.293333 119.850667 55.68 149.12 42.581333 4.608-33.109333 17.792-55.68 32.426667-68.48-113.706667-12.8-233.216-56.832-233.216-253.013333 0-55.893333 19.84-101.546667 52.693333-137.386667-5.76-12.928-23.04-64.981333 4.48-135.509333 0 0 42.88-13.738667 140.8 52.48 40.96-11.392 84.48-17.024 128-17.28 43.52 0.256 87.04 5.888 128 17.28 97.28-66.218667 140.16-52.48 140.16-52.48 27.52 70.528 10.24 122.581333 5.12 135.509333 32.64 35.84 52.48 81.493333 52.48 137.386667 0 196.693333-119.68 240-233.6 252.586667 17.92 15.36 34.56 46.762667 34.56 94.72 0 68.522667-0.64 123.562667-0.64 140.202666 0 13.44 8.96 29.44 35.2 24.32C877.44 942.592 1024 750.592 1024 524.672c0-282.752-229.248-512-512-512"></path>
</svg>

    </a>


<a href="/index.xml" rel="noopener alternate" type="application/rss&#43;xml"
    class="iconfont" title="rss" target="_blank">
    <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="30" height="30">
  <path d="M819.157333 1024C819.157333 574.592 449.408 204.8 0 204.8V0c561.706667 0 1024 462.293333 1024 1024h-204.842667zM140.416 743.04a140.8 140.8 0 0 1 140.501333 140.586667A140.928 140.928 0 0 1 140.074667 1024C62.72 1024 0 961.109333 0 883.626667s62.933333-140.544 140.416-140.586667zM678.784 1024h-199.04c0-263.210667-216.533333-479.786667-479.744-479.786667V345.173333c372.352 0 678.784 306.517333 678.784 678.826667z"></path>
</svg>

  </a>
   
</div>

<div class="copyright">

  <span class="copyright-year">
    &copy;
    
      2020 -
    2021
    <span class="heart">
      
      <i class="iconfont">
        <svg class="icon" viewBox="0 0 1025 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="14" height="14">
  <path d="M1000.1 247.9c-15.5-37.3-37.6-70.6-65.7-98.9-54.4-54.8-125.8-85-201-85-85.7 0-166 39-221.4 107.4C456.6 103 376.3 64 290.6 64c-75.1 0-146.5 30.4-201.1 85.6-28.2 28.5-50.4 61.9-65.8 99.3-16 38.8-24 79.9-23.6 122.2 0.7 91.7 40.1 177.2 108.1 234.8 3.1 2.6 6 5.1 8.9 7.8 14.9 13.4 58 52.8 112.6 102.7 93.5 85.5 209.9 191.9 257.5 234.2 7 6.1 15.8 9.5 24.9 9.5 9.2 0 18.1-3.4 24.9-9.5 34.5-30.7 105.8-95.9 181.4-165 74.2-67.8 150.9-138 195.8-178.2 69.5-57.9 109.6-144.4 109.9-237.3 0.1-42.5-8-83.6-24-122.2z"
   fill="#8a8a8a"></path>
</svg>

      </i>
    </span><span class="author">
        xiaonilee
        
      </span></span>

  
  

  
</div>

    </footer>

    <div class="back-to-top" id="back-to-top">
      <i class="iconfont">
        
        <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="35" height="35">
  <path d="M510.866688 227.694839 95.449397 629.218702l235.761562 0-2.057869 328.796468 362.40389 0L691.55698 628.188232l241.942331-3.089361L510.866688 227.694839zM63.840492 63.962777l894.052392 0 0 131.813095L63.840492 195.775872 63.840492 63.962777 63.840492 63.962777zM63.840492 63.962777"></path>
</svg>

      </i>
    </div>
  </div>
  
<script type="text/javascript" src="/lib/jquery/jquery-3.2.1.min.js"></script>
  <script type="text/javascript" src="/lib/slideout/slideout-1.0.1.min.js"></script>




<script type="text/javascript" src="/js/main.638251f4230630f0335d8c6748e53a96f94b72670920b60c09a56fdc8bece214.js" integrity="sha256-Y4JR9CMGMPAzXYxnSOU6lvlLcmcJILYMCaVv3Ivs4hQ=" crossorigin="anonymous"></script>



  <script type="text/javascript">
    window.MathJax = {
      showProcessingMessages: false,
      messageStyle: 'none'
    };
  </script>
  <script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-MML-AM_CHTML' async></script>









  
    <script type="text/javascript" src="/js/load-photoswipe.js"></script>
    <script type="text/javascript" src="/lib/photoswipe/photoswipe.min.js"></script>
    <script type="text/javascript" src="/lib/photoswipe/photoswipe-ui-default.min.js"></script>
  















</body>
</html>
