<!DOCTYPE html>
<html lang="en" itemscope itemtype="http://schema.org/WebPage">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>R Notebook : Application Exercises - xiaoni&#39;s blog</title>
  

<meta name="renderer" content="webkit" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>

<meta name="MobileOptimized" content="width"/>
<meta name="HandheldFriendly" content="true"/>


<meta name="applicable-device" content="pc,mobile">

<meta name="theme-color" content="#f8f5ec" />
<meta name="msapplication-navbutton-color" content="#f8f5ec">
<meta name="apple-mobile-web-app-capable" content="yes">
<meta name="apple-mobile-web-app-status-bar-style" content="#f8f5ec">

<meta name="mobile-web-app-capable" content="yes">

<meta name="author" content="Xiaoni" />
  <meta name="description" content="Bioinformatics Exercises of Application with R.
" />

  <meta name="keywords" content="xiaonili, xiaonilee, xiaoni" />






<meta name="generator" content="Hugo 0.79.0" />


<link rel="canonical" href="https://xiaonilee.github.io/post/hmwk10/" />





<link rel="icon" href="/favicon.ico" />











<link rel="stylesheet" href="/sass/jane.min.f1e506a781bf25d33ffc18aa6b4e972a965c58049d27d4f92b7db2e9bf28e4bf.css" integrity="sha256-8eUGp4G/JdM//Biqa06XKpZcWASdJ9T5K32y6b8o5L8=" media="screen" crossorigin="anonymous">



<link href="https://cdn.bootcdn.net/ajax/libs/font-awesome/5.15.1/css/all.css" rel="stylesheet">
<script src="https://cdn.bootcdn.net/ajax/libs/font-awesome/5.15.1/js/all.js"></script>




<meta property="og:title" content="R Notebook : Application Exercises" />
<meta property="og:description" content="Bioinformatics Exercises of Application with R." />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://xiaonilee.github.io/post/hmwk10/" />
<meta property="article:published_time" content="2020-10-15T00:00:00+00:00" />
<meta property="article:modified_time" content="2020-10-16T00:00:00+00:00" />
<meta itemprop="name" content="R Notebook : Application Exercises">
<meta itemprop="description" content="Bioinformatics Exercises of Application with R.">
<meta itemprop="datePublished" content="2020-10-15T00:00:00+00:00" />
<meta itemprop="dateModified" content="2020-10-16T00:00:00+00:00" />
<meta itemprop="wordCount" content="838">



<meta itemprop="keywords" content="R,Bioinformatics,Plots,TCGA,OncoLnc," />
<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="R Notebook : Application Exercises"/>
<meta name="twitter:description" content="Bioinformatics Exercises of Application with R."/>

<!--[if lte IE 9]>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/classlist/1.1.20170427/classList.min.js"></script>
<![endif]-->

<!--[if lt IE 9]>
  <script src="https://cdn.jsdelivr.net/npm/html5shiv@3.7.3/dist/html5shiv.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/respond.js@1.4.2/dest/respond.min.js"></script>
<![endif]-->


<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'UA-177860608-1', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>



</head>
<body>
  <div id="mobile-navbar" class="mobile-navbar">
  <div class="mobile-header-logo">
    <a href="/" class="logo">Xiaoni's Blog</a>
  </div>
  <div class="mobile-navbar-icon">
    <span></span>
    <span></span>
    <span></span>
  </div>
</div>
<nav id="mobile-menu" class="mobile-menu slideout-menu">
  <ul class="mobile-menu-list">
    <li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="https://xiaonilee.github.io/">Home</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="https://xiaonilee.github.io/post/">Archives</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="https://xiaonilee.github.io/categories/">Categories</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="https://xiaonilee.github.io/tags/">Tags</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="https://xiaonilee.github.io/about/">About Me</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          <div class="mobile-menu-parent">
            <span class="mobile-submenu-open"></span>
            <a href="https://xiaonilee.github.io/">
              docs
            </a>
          </div>
          <ul class="mobile-submenu-list">
            
              <li>
                <a href="https://xiaonilee.github.io/post/rinactionmap/">Book: R in Action</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/firebrowse/">FireBrowse Notebook</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/gdc/">GDC Notebook</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/gscalite/">GSCALite: Gene Set Cancer Analysis</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/localxenadatahub/">Local UCSC Xena Data Hub</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/oncolnc/">OncoLnc: A straightforward Tool</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/hmwk10/">R Notebook : Application Exercises</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/ggplot2/">R Notebook : ggplot2</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/tricktcga/">R re-analysis in TCGA</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/tensorflow201/">TensorFlow2: Iris</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/tissgdb/">TissGDB: tissue-specific gene database in cancer</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/genomicsignatures/">UCSC Xena Features: Genomic Signatures</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/loaddata/">UCSC Xena: Load Data From Public Database</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/privatepublictogether/">UCSC Xena: Private and Public Data Put Together</a>
              </li>
            
          </ul>
        
      </li>
    

    
  </ul>
</nav>


  
    






  <link rel="stylesheet" href="/lib/photoswipe/photoswipe.min.css" />
  <link rel="stylesheet" href="/lib/photoswipe/default-skin/default-skin.min.css" />




<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

<div class="pswp__bg"></div>

<div class="pswp__scroll-wrap">
    
    <div class="pswp__container">
      <div class="pswp__item"></div>
      <div class="pswp__item"></div>
      <div class="pswp__item"></div>
    </div>
    
    <div class="pswp__ui pswp__ui--hidden">
    <div class="pswp__top-bar">
      
      <div class="pswp__counter"></div>
      <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>
      <button class="pswp__button pswp__button--share" title="Share"></button>
      <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>
      <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>
      
      
      <div class="pswp__preloader">
        <div class="pswp__preloader__icn">
          <div class="pswp__preloader__cut">
            <div class="pswp__preloader__donut"></div>
          </div>
        </div>
      </div>
    </div>
    <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
      <div class="pswp__share-tooltip"></div>
    </div>
    <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
    </button>
    <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
    </button>
    <div class="pswp__caption">
      <div class="pswp__caption__center"></div>
    </div>
    </div>
    </div>
</div>

  

  

  

  <header id="header" class="header container">
    <div class="logo-wrapper">
  <a href="/" class="logo">
    
      Xiaoni's Blog
    
  </a>
</div>

<nav class="site-navbar">
  <ul id="menu" class="menu">
    
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="https://xiaonilee.github.io/">Home</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="https://xiaonilee.github.io/post/">Archives</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="https://xiaonilee.github.io/categories/">Categories</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="https://xiaonilee.github.io/tags/">Tags</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="https://xiaonilee.github.io/about/">About Me</a>
          

        

      </li>
    
        <li class="menu-item active">
        
          
          <a class="menu-item-link menu-parent" href="https://xiaonilee.github.io/">docs</a>
          <ul class="submenu" >
            
              <li>
                <a href="https://xiaonilee.github.io/post/rinactionmap/">Book: R in Action</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/firebrowse/">FireBrowse Notebook</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/gdc/">GDC Notebook</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/gscalite/">GSCALite: Gene Set Cancer Analysis</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/localxenadatahub/">Local UCSC Xena Data Hub</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/oncolnc/">OncoLnc: A straightforward Tool</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/hmwk10/">R Notebook : Application Exercises</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/ggplot2/">R Notebook : ggplot2</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/tricktcga/">R re-analysis in TCGA</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/tensorflow201/">TensorFlow2: Iris</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/tissgdb/">TissGDB: tissue-specific gene database in cancer</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/genomicsignatures/">UCSC Xena Features: Genomic Signatures</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/loaddata/">UCSC Xena: Load Data From Public Database</a>
              </li>
            
              <li>
                <a href="https://xiaonilee.github.io/post/privatepublictogether/">UCSC Xena: Private and Public Data Put Together</a>
              </li>
            
          </ul>

        

      </li>
    

    
    

    
  </ul>
</nav>

  </header>

  <div id="mobile-panel">
    <main id="main" class="main bg-llight">
      <div class="content-wrapper">
        <div id="content" class="content container">
          <article class="post bg-white">
    
    <header class="post-header">
      <h1 class="post-title">R Notebook : Application Exercises</h1>
      
      <div class="post-meta">
        <time datetime="2020-10-15" class="post-time">
          2020-10-15
        </time>
        <div class="post-category">
            <a href="https://xiaonilee.github.io/categories/r/"> R </a>
            <a href="https://xiaonilee.github.io/categories/bioinformatics/"> Bioinformatics </a>
            
          </div>
        

        
        

        
        
      </div>
    </header>

    
    
<div class="post-toc" id="post-toc">
  <h2 class="post-toc-title">Table of Contents</h2>
  <div class="post-toc-content">
    <nav id="TableOfContents">
  <ul>
    <li>
      <ul>
        <li><a href="#in-summary">In summary</a></li>
        <li><a href="#exercise1-id-reverse-by-orghsegdb">Exercise1. id reverse by org.Hs.eg.db</a></li>
        <li><a href="#exercise2-id-reverse-by-hgu133adb">Exercise2. id reverse by hgu133a.db</a></li>
        <li><a href="#exercise3-get-gene-expression-by-cll-and-do-boxplot">Exercise3. get gene expression by CLL and do boxplot</a></li>
        <li><a href="#exercise4-get-data-via-tcga-and-r-analysis">Exercise4. get data via TCGA and r analysis</a></li>
        <li><a href="#exercise5-gene-in-clinical-survival">Exercise5. gene in clinical survival</a></li>
        <li><a href="#exercise6-download-gse17215-and-plot-heatmap">Exercise6. Download GSE17215 and plot heatmap</a></li>
        <li><a href="#exercise7-download-gse24673-to-do-correlation-and-plot-heatmap">Exercise7. Download GSE24673 to do correlation and plot heatmap</a></li>
        <li><a href="#exercise8-annotation-package-install">Exercise8. Annotation package install</a></li>
        <li><a href="#exercise9-download-gse42872-to-compute-avesdmad-of-probesgenes">Exercise9. Download GSE42872 to compute ave/sd/mad of probes/genes</a></li>
        <li><a href="#exercise10-download-gse42872-to-do-deg">Exercise10. Download GSE42872 to do DEG</a></li>
      </ul>
    </li>
  </ul>
</nav>
  </div>
</div>

    
    <div class="post-content">
      <p>Bioinformatics Exercises of Application with <a href="hmwk10.Rmd">R</a>.</p>
<h3 id="in-summary">In summary</h3>
<p>1.strsplit()</p>
<p>2.<code>lapply()</code> row 1, column 2</p>
<p>3.<code>library(stringr)</code> split character</p>
<p>4.merge()</p>
<p>5.match()</p>
<p>6.library(dplyr) delete column</p>
<p>7.library(org.Hs.eg.db) annotation <code>id reverse</code></p>
<p>8.library(hgu133a.db) annotation id reverse</p>
<p>9.?pData <code>group info</code></p>
<p>10.<code>library(ggstatsplot)</code></p>
<p>11.?survfit</p>
<p>12.<code>library(survival)</code></p>
<p>13.library(survminer)</p>
<p>14.?survfit</p>
<p>15.?ggsurvplot</p>
<p>16.<code>GEOquery</code> download dataset</p>
<p>17.<code>pheatmap::pheatmap</code></p>
<p>18.<code>?cor</code></p>
<p>19.<code>'%in%'</code></p>
<p>20.library(limma) <code>DEG</code></p>
<p>21.exprs()</p>
<h3 id="exercise1-id-reverse-by-orghsegdb">Exercise1. id reverse by org.Hs.eg.db</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">a</span><span class="o">=</span><span class="nf">read.table</span><span class="p">(</span><span class="s">&#39;ensembl_id2symbol.txt&#39;</span><span class="p">)</span>

<span class="nf">library</span><span class="p">(</span><span class="n">org.Hs.eg.db</span><span class="p">)</span>
<span class="n">g2s</span><span class="o">=</span><span class="nf">toTable</span><span class="p">(</span><span class="n">org.Hs.egSYMBOL</span><span class="p">)</span>
<span class="n">g2e</span><span class="o">=</span><span class="nf">toTable</span><span class="p">(</span><span class="n">org.Hs.egENSEMBL</span><span class="p">)</span>

<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>
<span class="n">a</span><span class="o">$</span><span class="n">ensembl_id</span><span class="o">=</span><span class="nf">str_split</span><span class="p">(</span><span class="n">a</span><span class="o">$</span><span class="n">V1</span><span class="p">,</span><span class="s">&#39;[.]&#39;</span><span class="p">,</span><span class="n">simplify</span><span class="o">=</span><span class="bp">T</span><span class="p">)</span><span class="n">[</span><span class="p">,</span><span class="m">1</span><span class="n">]</span>

<span class="n">b</span><span class="o">=</span><span class="nf">merge</span><span class="p">(</span><span class="n">a</span><span class="p">,</span><span class="n">g2e</span><span class="p">,</span><span class="n">by</span><span class="o">=</span><span class="s">&#39;ensembl_id&#39;</span><span class="p">,</span><span class="n">all.x</span><span class="o">=</span><span class="bp">T</span><span class="p">)</span>
<span class="n">d</span><span class="o">=</span><span class="nf">merge</span><span class="p">(</span><span class="n">b</span><span class="p">,</span><span class="n">g2s</span><span class="p">,</span><span class="n">by</span><span class="o">=</span><span class="s">&#39;gene_id&#39;</span><span class="p">,</span><span class="n">all.x</span><span class="o">=</span><span class="bp">T</span><span class="p">)</span>

<span class="n">d</span><span class="o">=</span><span class="n">d</span><span class="nf">[match</span><span class="p">(</span><span class="n">a</span><span class="o">$</span><span class="n">V1</span><span class="p">,</span><span class="n">d</span><span class="o">$</span><span class="n">V1</span><span class="p">),</span><span class="n">]</span>

<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="n">res</span><span class="o">=</span><span class="nf">select</span><span class="p">(</span><span class="n">d</span><span class="p">,</span><span class="o">-</span><span class="n">V1</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><h3 id="exercise2-id-reverse-by-hgu133adb">Exercise2. id reverse by hgu133a.db</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">a2</span><span class="o">=</span><span class="nf">read.table</span><span class="p">(</span><span class="s">&#39;probe_id2symbol.txt&#39;</span><span class="p">)</span>

<span class="c1"># BiocManager::install(&#34;hgu133a.db&#34;)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">hgu133a.db</span><span class="p">)</span>
<span class="nf">ls</span><span class="p">(</span><span class="s">&#34;package:hgu133a.db&#34;</span><span class="p">)</span>

<span class="n">id2s</span><span class="o">=</span><span class="nf">toTable</span><span class="p">(</span><span class="n">hgu133aSYMBOL</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">id2s</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">a2</span><span class="p">)</span><span class="o">=</span><span class="s">&#39;probe_id&#39;</span>

<span class="n">b2</span><span class="o">=</span><span class="nf">merge</span><span class="p">(</span><span class="n">a2</span><span class="p">,</span><span class="n">id2s</span><span class="p">,</span><span class="n">by</span><span class="o">=</span><span class="s">&#39;probe_id&#39;</span><span class="p">)</span>
<span class="n">res2</span><span class="o">=</span><span class="n">b2</span>
</code></pre></td></tr></table>
</div>
</div><h3 id="exercise3-get-gene-expression-by-cll-and-do-boxplot">Exercise3. get gene expression by CLL and do boxplot</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">library</span><span class="p">(</span><span class="n">CLL</span><span class="p">)</span>
<span class="nf">data</span><span class="p">(</span><span class="s">&#34;sCLLex&#34;</span><span class="p">)</span>
<span class="n">exprSet</span><span class="o">=</span><span class="nf">exprs</span><span class="p">(</span><span class="n">sCLLex</span><span class="p">)</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">exprSet</span><span class="p">)</span>
<span class="n">exprSet[1</span><span class="o">:</span><span class="m">6</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="m">6</span><span class="n">]</span>

<span class="n">pd</span><span class="o">=</span><span class="nf">pData</span><span class="p">(</span><span class="n">sCLLex</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">pd</span><span class="p">)</span>
<span class="nf">table</span><span class="p">(</span><span class="n">pd</span><span class="o">$</span><span class="n">Disease</span><span class="p">)</span>

<span class="nf">library</span><span class="p">(</span><span class="n">hgu95av2.db</span><span class="p">)</span>
<span class="nf">ls</span><span class="p">(</span><span class="s">&#39;package:hgu95av2.db&#39;</span><span class="p">)</span>
<span class="n">id2s</span><span class="o">=</span><span class="nf">toTable</span><span class="p">(</span><span class="n">hgu95av2SYMBOL</span><span class="p">)</span>
<span class="nf">filter</span><span class="p">(</span><span class="n">id2s</span><span class="p">,</span><span class="n">symbol</span><span class="o">==</span><span class="s">&#39;FSCN1&#39;</span><span class="p">)</span>

<span class="n">exprSet[</span><span class="s">&#39;39070_at&#39;</span><span class="p">,</span><span class="n">]</span>

<span class="n">pd[pd</span><span class="o">$</span><span class="n">Disease</span><span class="o">==</span><span class="s">&#39;progres.&#39;</span><span class="p">,</span><span class="n">]</span>

<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">boxplot</span><span class="p">(</span><span class="n">exprSet[</span><span class="s">&#39;39070_at&#39;</span><span class="p">,</span><span class="n">]</span> <span class="o">~</span> <span class="n">pd</span><span class="o">$</span><span class="n">Disease</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;brown&#39;</span><span class="p">,</span>
        <span class="n">main</span><span class="o">=</span><span class="s">&#39;fscn1 in CLL&#39;</span><span class="p">,</span>
        <span class="n">xlab</span><span class="o">=</span><span class="s">&#39;Disease&#39;</span><span class="p">,</span> <span class="n">ylab</span><span class="o">=</span><span class="s">&#39;expression of fscn1&#39;</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#39;fscn1boxplot.png&#39;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fscn1boxplot.png" alt="fscn1boxplot"></p>
<h3 id="exercise4-get-data-via-tcga-and-r-analysis">Exercise4. get data via TCGA and r analysis</h3>
<h4 id="setup-dataset-from-tcga-and-download">setup dataset from TCGA and download</h4>
<p>Breast&ndash;&gt; TCGA PanCancer Atlas&ndash;&gt;FSCN1&ndash;&gt;Plots</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">a</span><span class="o">=</span><span class="nf">read.table</span><span class="p">(</span><span class="s">&#39;fscn1.txt&#39;</span><span class="p">,</span><span class="n">sep</span> <span class="o">=</span> <span class="s">&#39;\t&#39;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">a</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">a</span><span class="p">)</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;id&#39;</span><span class="p">,</span><span class="s">&#39;subtype&#39;</span><span class="p">,</span><span class="s">&#39;expression&#39;</span><span class="p">,</span><span class="s">&#39;mut&#39;</span><span class="p">,</span><span class="s">&#39;CNA&#39;</span><span class="p">)</span>

<span class="nf">library</span><span class="p">(</span><span class="n">ggstatsplot</span><span class="p">)</span>
<span class="nf">ggbetweenstats</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">a</span><span class="p">,</span> <span class="n">x</span> <span class="o">=</span> <span class="n">subtype</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">expression</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#39;fscn1plot.png&#39;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="fscn1plot.png" alt="fscn1plot"></p>
<h3 id="exercise5-gene-in-clinical-survival">Exercise5. gene in clinical survival</h3>
<h4 id="get-data-from-oncolncorg">get data from oncoLnc.org</h4>
<ul>
<li>oncoLnc&ndash;&gt;TP53&ndash;&gt;BRCA&ndash;&gt;LOWER/UPPER Percentile of 50&ndash;&gt;Submit</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">a</span><span class="o">=</span><span class="nf">read.table</span><span class="p">(</span><span class="s">&#39;BRCA_7157_50_50.csv&#39;</span><span class="p">,</span><span class="n">sep</span> <span class="o">=</span> <span class="s">&#39;,&#39;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">a</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">a</span><span class="p">)</span>
<span class="nf">table</span><span class="p">(</span><span class="n">a</span><span class="o">$</span><span class="n">Status</span><span class="p">)</span>

<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">survival</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">survminer</span><span class="p">)</span>

<span class="n">a</span><span class="o">$</span><span class="n">Status</span><span class="o">=</span><span class="nf">ifelse</span><span class="p">(</span><span class="n">a</span><span class="o">$</span><span class="n">Status</span><span class="o">==</span><span class="s">&#39;Dead&#39;</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">0</span><span class="p">)</span>
<span class="n">fit</span><span class="o">=</span><span class="nf">survfit</span><span class="p">(</span><span class="nf">Surv</span><span class="p">(</span><span class="n">Days</span><span class="p">,</span><span class="n">Status</span><span class="p">)</span> <span class="o">~</span> <span class="n">Group</span><span class="p">,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">a</span><span class="p">)</span>
<span class="nf">str</span><span class="p">(</span><span class="n">fit</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">fit</span><span class="p">)</span>

<span class="nf">ggsurvplot</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span> <span class="n">conf.int</span> <span class="o">=</span> <span class="bp">F</span><span class="p">,</span> <span class="n">pval</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span>
           <span class="n">linetype</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span>
           <span class="n">ggtheme</span> <span class="o">=</span> <span class="nf">theme_survminer</span><span class="p">())</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#39;tp53_expression_Survival.png&#39;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="tp53_expression_Survival.png" alt="tp53_expression_Survival"></p>
<ul>
<li>considering the poor R value, re-analysis with subtype</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">b</span><span class="o">=</span><span class="nf">read.table</span><span class="p">(</span><span class="s">&#39;fscn1.txt&#39;</span><span class="p">,</span><span class="n">sep</span> <span class="o">=</span> <span class="s">&#39;\t&#39;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">b</span><span class="p">)</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;id&#39;</span><span class="p">,</span><span class="s">&#39;subtype&#39;</span><span class="p">,</span><span class="s">&#39;expression&#39;</span><span class="p">,</span><span class="s">&#39;mut&#39;</span><span class="p">,</span><span class="s">&#39;CNA&#39;</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">a</span><span class="p">)</span>

<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>
<span class="n">b</span><span class="o">$</span><span class="n">Patient</span> <span class="o">=</span><span class="nf">str_split</span><span class="p">(</span><span class="n">b</span><span class="o">$</span><span class="n">id</span><span class="p">,</span><span class="s">&#39;-01&#39;</span><span class="p">,</span><span class="n">simplify</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span><span class="n">[</span><span class="p">,</span><span class="m">1</span><span class="n">]</span>

<span class="n">d</span><span class="o">=</span><span class="nf">merge</span><span class="p">(</span><span class="n">a</span><span class="p">,</span> <span class="n">b</span><span class="p">,</span> <span class="n">by</span><span class="o">=</span><span class="s">&#39;Patient&#39;</span><span class="p">,</span> <span class="n">all.x</span><span class="o">=</span><span class="bp">T</span><span class="p">)</span>
<span class="n">fit2</span><span class="o">=</span><span class="nf">survfit</span><span class="p">(</span><span class="nf">Surv</span><span class="p">(</span><span class="n">Days</span><span class="p">,</span><span class="n">Status</span><span class="p">)</span> <span class="o">~</span> <span class="n">subtype</span><span class="p">,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">d</span><span class="p">)</span>
<span class="nf">str</span><span class="p">(</span><span class="n">fit2</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">fit2</span><span class="p">)</span>

<span class="nf">ggsurvplot</span><span class="p">(</span><span class="n">fit2</span><span class="p">,</span> <span class="n">conf.int</span> <span class="o">=</span> <span class="bp">F</span><span class="p">,</span> <span class="n">pval</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span>
           <span class="n">linetype</span> <span class="o">=</span> <span class="m">5</span><span class="p">,</span>
           <span class="n">ggtheme</span> <span class="o">=</span> <span class="nf">theme_survminer</span><span class="p">())</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#39;tp53_Subtype_Survival.png&#39;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="tp53_Subtype_Survival.png" alt="tp53_Subtype_Survival"></p>
<h3 id="exercise6-download-gse17215-and-plot-heatmap">Exercise6. Download GSE17215 and plot heatmap</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1"># download GEO dataset</span>
<span class="c1"># BiocManager::install(&#39;GEOquery&#39;)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">GEOquery</span><span class="p">)</span>
<span class="n">gse17215</span> <span class="o">=</span> <span class="nf">getGEO</span><span class="p">(</span><span class="s">&#39;GSE17215&#39;</span><span class="p">,</span> <span class="n">destdir</span> <span class="o">=</span> <span class="s">&#39;.&#39;</span><span class="p">,</span> <span class="n">AnnotGPL</span> <span class="o">=</span> <span class="bp">F</span><span class="p">,</span> <span class="n">getGPL</span> <span class="o">=</span> <span class="bp">F</span><span class="p">)</span>
<span class="nf">class</span><span class="p">(</span><span class="n">gse17215</span><span class="p">)</span>
<span class="nf">length</span><span class="p">(</span><span class="n">gse17215</span><span class="p">)</span>

<span class="n">a</span><span class="o">=</span><span class="n">gse17215[[1]]</span>
<span class="n">dat</span><span class="o">=</span><span class="nf">exprs</span><span class="p">(</span><span class="n">a</span><span class="p">)</span>

<span class="nf">dim</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>

<span class="nf">library</span><span class="p">(</span><span class="n">hgu133a.db</span><span class="p">)</span>
<span class="nf">library</span><span class="p">()</span>
<span class="n">id2s</span><span class="o">=</span><span class="nf">toTable</span><span class="p">(</span><span class="n">hgu133aSYMBOL</span><span class="p">)</span>

<span class="n">dat</span> <span class="o">=</span> <span class="n">dat[id2s</span><span class="o">$</span><span class="n">probe_id</span><span class="p">,</span><span class="n">]</span>
<span class="n">id2s</span><span class="o">$</span><span class="n">median</span><span class="o">=</span><span class="nf">apply</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="n">median</span><span class="p">)</span>
<span class="n">id2s</span><span class="o">=</span><span class="n">id2s</span><span class="nf">[order</span><span class="p">(</span><span class="n">id2s</span><span class="o">$</span><span class="n">symbol</span><span class="p">,</span><span class="n">id2s</span><span class="o">$</span><span class="n">median</span><span class="p">,</span><span class="n">decreasing</span> <span class="o">=</span> <span class="bp">T</span><span class="p">),</span><span class="n">]</span>
<span class="n">id2s</span><span class="o">=</span><span class="n">id2s[</span><span class="o">!</span><span class="nf">duplicated</span><span class="p">(</span><span class="n">id2s</span><span class="o">$</span><span class="n">symbol</span><span class="p">),</span><span class="n">]</span>
<span class="n">dat</span><span class="o">=</span><span class="n">dat[id2s</span><span class="o">$</span><span class="n">probe_id</span><span class="p">,</span><span class="n">]</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span><span class="o">=</span><span class="n">id2s</span><span class="o">$</span><span class="n">symbol</span>

<span class="c1"># gene list</span>
<span class="n">gl</span><span class="o">=</span><span class="s">&#39;ACTR3B ANLN BAG1 BCL2 BIRC5 BLVRA CCNB1 CCNE1 CDC20 CDC6 CDCA1 CDH3 CENPF CEP55 CXXC5 EGFR ERBB2 ESR1 EXO1 FGFR4 FOXA1 FOXC1 GPR160 GRB7 KIF2C KNTC2 KRT14 KRT17 KRT5 MAPT MDM2 MELK MIA MKI67 MLPH MMP11 MYBL2 MYC NAT1 ORC6L PGR PHGDH PTTG1 RRM2 SFRP1 SLC39A6 TMEM45B TYMS UBE2C UBE2T&#39;</span>

<span class="n">gl</span> <span class="o">=</span> <span class="nf">strsplit</span><span class="p">(</span><span class="n">gl</span><span class="p">,</span> <span class="s">&#39; &#39;</span><span class="p">)</span><span class="n">[[1]]</span>

<span class="nf">table</span><span class="p">(</span><span class="n">gl</span> <span class="o">%in%</span> <span class="nf">rownames</span><span class="p">(</span><span class="n">dat</span><span class="p">))</span>

<span class="n">gl</span><span class="o">=</span><span class="n">gl[gl</span> <span class="o">%in%</span> <span class="nf">rownames</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span><span class="n">]</span>
<span class="n">dat</span> <span class="o">=</span> <span class="n">dat[gl</span><span class="p">,</span><span class="n">]</span>
<span class="n">dat</span> <span class="o">=</span> <span class="nf">log2</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>

<span class="n">pheatmap</span><span class="o">::</span><span class="nf">pheatmap</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="n">scale</span> <span class="o">=</span> <span class="s">&#39;row&#39;</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#39;pheatmap_row.png&#39;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="pheatmap_row.png" alt="pheatmap_row"></p>
<h3 id="exercise7-download-gse24673-to-do-correlation-and-plot-heatmap">Exercise7. Download GSE24673 to do correlation and plot heatmap</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">library</span><span class="p">(</span><span class="n">GEOquery</span><span class="p">)</span>
<span class="n">gse24673</span><span class="o">=</span><span class="nf">getGEO</span><span class="p">(</span><span class="s">&#39;GSE24673&#39;</span><span class="p">,</span> <span class="n">destdir</span> <span class="o">=</span> <span class="s">&#39;.&#39;</span><span class="p">,</span> <span class="n">AnnotGPL</span> <span class="o">=</span> <span class="bp">F</span><span class="p">,</span> <span class="n">getGPL</span> <span class="o">=</span> <span class="bp">F</span><span class="p">)</span>

<span class="nf">class</span><span class="p">(</span><span class="n">gse24673</span><span class="p">)</span>
<span class="nf">length</span><span class="p">(</span><span class="n">gse24673</span><span class="p">)</span>

<span class="n">a</span> <span class="o">=</span> <span class="n">gse24673[[1]]</span>
<span class="n">dat</span><span class="o">=</span><span class="nf">exprs</span><span class="p">(</span><span class="n">a</span><span class="p">)</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>

<span class="n">pd</span><span class="o">=</span><span class="nf">pData</span><span class="p">(</span><span class="n">a</span><span class="p">)</span>

<span class="c1"># create group_list, according to pd &#39;source name&#39;.</span>
<span class="n">group_list</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;rbc&#39;</span><span class="p">,</span><span class="s">&#39;rbc&#39;</span><span class="p">,</span><span class="s">&#39;rbc&#39;</span><span class="p">,</span>
             <span class="s">&#39;rbn&#39;</span><span class="p">,</span><span class="s">&#39;rbn&#39;</span><span class="p">,</span><span class="s">&#39;rbn&#39;</span><span class="p">,</span>
             <span class="s">&#39;rbc&#39;</span><span class="p">,</span><span class="s">&#39;rbc&#39;</span><span class="p">,</span><span class="s">&#39;rbc&#39;</span><span class="p">,</span>
             <span class="s">&#39;normal&#39;</span><span class="p">,</span><span class="s">&#39;normal&#39;</span><span class="p">)</span>


<span class="n">M</span><span class="o">=</span><span class="nf">cor</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">M</span><span class="p">)</span>
<span class="n">pheatmap</span><span class="o">::</span><span class="nf">pheatmap</span><span class="p">(</span><span class="n">M</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#39;pheatmap2.png&#39;</span><span class="p">)</span>

<span class="n">tmp</span><span class="o">=</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">group</span><span class="o">=</span><span class="n">group_list</span><span class="p">)</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">tmp</span><span class="p">)</span><span class="o">=</span><span class="nf">colnames</span><span class="p">(</span><span class="n">M</span><span class="p">)</span>
<span class="n">pheatmap</span><span class="o">::</span><span class="nf">pheatmap</span><span class="p">(</span><span class="n">M</span><span class="p">,</span> <span class="n">annotation_col</span> <span class="o">=</span> <span class="n">tmp</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#39;pheatmap3.png&#39;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><p><img src="pheatmap3.png" alt="pheatmap3"></p>
<h3 id="exercise8-annotation-package-install">Exercise8. Annotation package install</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="c1"># ref:https://gist.github.com/xiaonilee/f582c75173122d3625afc34f638b7b30</span>
<span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&#39;hugene10sttranscriptcluster.db&#39;</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div><h3 id="exercise9-download-gse42872-to-compute-avesdmad-of-probesgenes">Exercise9. Download GSE42872 to compute ave/sd/mad of probes/genes</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">library</span><span class="p">(</span><span class="n">GEOquery</span><span class="p">)</span>
<span class="n">gse42872</span> <span class="o">=</span> <span class="nf">getGEO</span><span class="p">(</span><span class="s">&#39;GSE42872&#39;</span><span class="p">,</span> <span class="n">destdir</span> <span class="o">=</span> <span class="s">&#39;.&#39;</span><span class="p">,</span> <span class="n">AnnotGPL</span> <span class="o">=</span> <span class="bp">F</span><span class="p">,</span> <span class="n">getGPL</span> <span class="o">=</span> <span class="bp">F</span><span class="p">)</span>
<span class="n">a</span> <span class="o">=</span> <span class="n">gse42872[[1]]</span>
<span class="n">dat</span><span class="o">=</span><span class="nf">exprs</span><span class="p">(</span><span class="n">a</span><span class="p">)</span>
<span class="n">pd</span><span class="o">=</span><span class="nf">pData</span><span class="p">(</span><span class="n">a</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>

<span class="nf">sort</span><span class="p">(</span><span class="nf">apply</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="m">1</span><span class="p">,</span> <span class="n">mean</span><span class="p">),</span> <span class="n">decreasing</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span><span class="n">[1]</span>
<span class="c1"># output</span>
<span class="c1"># 7978905</span>
<span class="c1"># 14.53288</span>

<span class="nf">sort</span><span class="p">(</span><span class="nf">apply</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="n">sd</span><span class="p">),</span> <span class="n">decreasing</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span><span class="n">[1]</span>
<span class="c1"># output</span>
<span class="c1"># 8133876</span>
<span class="c1"># 3.166429</span>

<span class="nf">sort</span><span class="p">(</span><span class="nf">apply</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="m">1</span><span class="p">,</span> <span class="n">mad</span><span class="p">),</span> <span class="n">decreasing</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span><span class="n">[1]</span>
<span class="c1"># output</span>
<span class="c1"># 8133876</span>
<span class="c1"># 4.268561</span>


<span class="c1"># check annotation package https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE42872</span>
<span class="nf">library</span><span class="p">(</span><span class="n">hugene10sttranscriptcluster.db</span><span class="p">)</span>
<span class="nf">ls</span><span class="p">(</span><span class="s">&#39;package:hugene10sttranscriptcluster.db&#39;</span><span class="p">)</span>
<span class="n">id2s</span><span class="o">=</span><span class="nf">toTable</span><span class="p">(</span><span class="n">hugene10sttranscriptclusterSYMBOL</span><span class="p">)</span>


<span class="n">max_mean</span> <span class="o">=</span> <span class="n">id2s[id2s</span><span class="o">$</span><span class="n">probe_id</span> <span class="o">%in%</span> <span class="m">7978905</span><span class="p">,</span><span class="n">]</span>
<span class="nf">table</span><span class="p">(</span><span class="n">id2s</span><span class="o">$</span><span class="n">probe_id</span> <span class="o">%in%</span> <span class="m">7978905</span><span class="p">)</span>

<span class="n">max_sd</span> <span class="o">=</span> <span class="n">id2s[id2s</span><span class="o">$</span><span class="n">probe_id</span> <span class="o">%in%</span> <span class="m">8133876</span><span class="p">,</span><span class="n">]</span>
<span class="n">max_mad</span> <span class="o">=</span> <span class="n">id2s[id2s</span><span class="o">$</span><span class="n">probe_id</span> <span class="o">%in%</span> <span class="m">8133876</span><span class="p">,</span><span class="n">]</span>
</code></pre></td></tr></table>
</div>
</div><h3 id="exercise10-download-gse42872-to-do-deg">Exercise10. Download GSE42872 to do DEG</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="nf">library</span><span class="p">(</span><span class="n">GEOquery</span><span class="p">)</span>
<span class="n">gse42872</span> <span class="o">=</span> <span class="nf">getGEO</span><span class="p">(</span><span class="s">&#39;GSE42872&#39;</span><span class="p">,</span> <span class="n">destdir</span> <span class="o">=</span> <span class="s">&#39;.&#39;</span><span class="p">,</span> <span class="n">AnnotGPL</span> <span class="o">=</span> <span class="bp">F</span><span class="p">,</span> <span class="n">getGPL</span> <span class="o">=</span> <span class="bp">F</span><span class="p">)</span>
<span class="n">a</span> <span class="o">=</span> <span class="n">gse42872[[1]]</span>
<span class="n">dat</span><span class="o">=</span><span class="nf">exprs</span><span class="p">(</span><span class="n">a</span><span class="p">)</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">a</span><span class="p">)</span>
<span class="n">pd</span><span class="o">=</span><span class="nf">pData</span><span class="p">(</span><span class="n">a</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>

<span class="nf">colnames</span><span class="p">(</span><span class="n">pd</span><span class="p">)</span>
<span class="c1"># strsplit(pd$title,&#39; &#39;)</span>
<span class="c1"># str_split(pd$title,&#39; &#39;)</span>
<span class="c1"># strsplit(pd$title,&#39; &#39;)[[1]]</span>
<span class="c1"># strsplit(pd$title,&#39; &#39;)[[1]][4]</span>

<span class="n">group_list</span><span class="o">=</span><span class="nf">unlist</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">pd</span><span class="o">$</span><span class="n">title</span><span class="p">,</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
  <span class="nf">str_split</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="s">&#39; &#39;</span><span class="p">)</span><span class="n">[[1]][4]</span>
<span class="p">}))</span>
<span class="n">group_list</span>

<span class="c1"># design matrix</span>
<span class="nf">library</span><span class="p">(</span><span class="n">limma</span><span class="p">)</span> <span class="c1"># Linear Models for Microarray Data</span>
<span class="n">design</span> <span class="o">=</span> <span class="nf">model.matrix</span><span class="p">(</span><span class="o">~</span><span class="m">0</span><span class="o">+</span><span class="nf">factor</span><span class="p">(</span><span class="n">group_list</span><span class="p">))</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">design</span><span class="p">)</span><span class="o">=</span><span class="nf">levels</span><span class="p">(</span><span class="nf">factor</span><span class="p">(</span><span class="n">group_list</span><span class="p">))</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">design</span><span class="p">)</span><span class="o">=</span><span class="nf">colnames</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>

<span class="n">contrast.matric</span> <span class="o">=</span> <span class="nf">makeContrasts</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">group_list</span><span class="p">),</span> <span class="n">collapse</span> <span class="o">=</span> <span class="s">&#39;-&#39;</span><span class="p">),</span> <span class="n">levels</span> <span class="o">=</span> <span class="n">design</span><span class="p">)</span>

<span class="n">fit</span> <span class="o">=</span> <span class="nf">lmFit</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span><span class="n">design</span><span class="p">)</span> <span class="c1"># Linear Model for Series of Arrays</span>
<span class="n">fit</span> <span class="o">=</span> <span class="nf">contrasts.fit</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span><span class="n">contrast.matric</span><span class="p">)</span>
<span class="n">fit</span> <span class="o">=</span> <span class="nf">eBayes</span><span class="p">(</span><span class="n">fit</span><span class="p">)</span>

<span class="n">tempOutput</span> <span class="o">=</span> <span class="nf">topTable</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span><span class="n">coef</span> <span class="o">=</span> <span class="m">1</span><span class="p">,</span> <span class="n">n</span><span class="o">=</span><span class="kc">Inf</span><span class="p">)</span>
<span class="n">nrDEG</span> <span class="o">=</span> <span class="nf">na.omit</span><span class="p">(</span><span class="n">tempOutput</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">nrDEG</span><span class="p">)</span>
</code></pre></td></tr></table>
</div>
</div>
    </div>

    
    


    
    

    <footer class="post-footer">
      <div class="post-tags">
          <a href="https://xiaonilee.github.io/tags/r/">R</a>
          <a href="https://xiaonilee.github.io/tags/bioinformatics/">Bioinformatics</a>
          <a href="https://xiaonilee.github.io/tags/plots/">Plots</a>
          <a href="https://xiaonilee.github.io/tags/tcga/">TCGA</a>
          <a href="https://xiaonilee.github.io/tags/oncolnc/">OncoLnc</a>
          
        </div>

      
      <nav class="post-nav">
        
          <a class="prev" href="/post/rinaction1/">
            
            <i class="iconfont">
              <svg  class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="18" height="18">
  <path d="M691.908486 949.511495l75.369571-89.491197c10.963703-12.998035 10.285251-32.864502-1.499144-44.378743L479.499795 515.267417 757.434875 204.940602c11.338233-12.190647 11.035334-32.285311-0.638543-44.850487l-80.46666-86.564541c-11.680017-12.583596-30.356378-12.893658-41.662889-0.716314L257.233596 494.235404c-11.332093 12.183484-11.041474 32.266891 0.657986 44.844348l80.46666 86.564541c1.772366 1.910513 3.706415 3.533476 5.750981 4.877077l306.620399 321.703933C662.505829 963.726242 680.945807 962.528973 691.908486 949.511495z"></path>
</svg>

            </i>
            <span class="prev-text nav-default">Chapter 1. Introduction To R</span>
            <span class="prev-text nav-mobile">Prev</span>
          </a>
        
          <a class="next" href="/post/tricktcga/">
            <span class="next-text nav-default">R re-analysis in TCGA</span>
            <span class="prev-text nav-mobile">Next</span>
            
            <i class="iconfont">
              <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="18" height="18">
  <path d="M332.091514 74.487481l-75.369571 89.491197c-10.963703 12.998035-10.285251 32.864502 1.499144 44.378743l286.278095 300.375162L266.565125 819.058374c-11.338233 12.190647-11.035334 32.285311 0.638543 44.850487l80.46666 86.564541c11.680017 12.583596 30.356378 12.893658 41.662889 0.716314l377.434212-421.426145c11.332093-12.183484 11.041474-32.266891-0.657986-44.844348l-80.46666-86.564541c-1.772366-1.910513-3.706415-3.533476-5.750981-4.877077L373.270379 71.774697C361.493148 60.273758 343.054193 61.470003 332.091514 74.487481z"></path>
</svg>

            </i>
          </a>
      </nav>
    </footer>
  </article>

  
  

  
  

  

  
  

  

  

  

    

  

        </div>
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="icon-links">
  
  
    <a href="mailto:xiaoni0601@gmail.com" rel="me noopener" class="iconfont"
      title="email" >
      <svg class="icon" viewBox="0 0 1451 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="36" height="36">
  <path d="M664.781909 681.472759 0 97.881301C0 3.997201 71.046997 0 71.046997 0L474.477909 0 961.649408 0 1361.641813 0C1361.641813 0 1432.688811 3.997201 1432.688811 97.881301L771.345323 681.472759C771.345323 681.472759 764.482731 685.154773 753.594283 688.65053L753.594283 688.664858C741.602731 693.493018 729.424896 695.068979 718.077952 694.839748 706.731093 695.068979 694.553173 693.493018 682.561621 688.664858L682.561621 688.65053C671.644501 685.140446 664.781909 681.472759 664.781909 681.472759L664.781909 681.472759ZM718.063616 811.603883C693.779541 811.016482 658.879232 802.205449 619.10784 767.734955 542.989056 701.759633 0 212.052267 0 212.052267L0 942.809523C0 942.809523 0 1024 83.726336 1024L682.532949 1024 753.579947 1024 1348.948139 1024C1432.688811 1024 1432.688811 942.809523 1432.688811 942.809523L1432.688811 212.052267C1432.688811 212.052267 893.138176 701.759633 817.019477 767.734955 777.248 802.205449 742.347691 811.03081 718.063616 811.603883L718.063616 811.603883Z"></path>
</svg>

    </a>
  
    <a href="https://github.com/xiaonilee" rel="me noopener" class="iconfont"
      title="github"  target="_blank"
      >
      <svg class="icon" style="" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="36" height="36">
  <path d="M512 12.672c-282.88 0-512 229.248-512 512 0 226.261333 146.688 418.133333 350.08 485.76 25.6 4.821333 34.986667-11.008 34.986667-24.618667 0-12.16-0.426667-44.373333-0.64-87.04-142.421333 30.890667-172.458667-68.693333-172.458667-68.693333C188.672 770.986667 155.008 755.2 155.008 755.2c-46.378667-31.744 3.584-31.104 3.584-31.104 51.413333 3.584 78.421333 52.736 78.421333 52.736 45.653333 78.293333 119.850667 55.68 149.12 42.581333 4.608-33.109333 17.792-55.68 32.426667-68.48-113.706667-12.8-233.216-56.832-233.216-253.013333 0-55.893333 19.84-101.546667 52.693333-137.386667-5.76-12.928-23.04-64.981333 4.48-135.509333 0 0 42.88-13.738667 140.8 52.48 40.96-11.392 84.48-17.024 128-17.28 43.52 0.256 87.04 5.888 128 17.28 97.28-66.218667 140.16-52.48 140.16-52.48 27.52 70.528 10.24 122.581333 5.12 135.509333 32.64 35.84 52.48 81.493333 52.48 137.386667 0 196.693333-119.68 240-233.6 252.586667 17.92 15.36 34.56 46.762667 34.56 94.72 0 68.522667-0.64 123.562667-0.64 140.202666 0 13.44 8.96 29.44 35.2 24.32C877.44 942.592 1024 750.592 1024 524.672c0-282.752-229.248-512-512-512"></path>
</svg>

    </a>


<a href="https://xiaonilee.github.io/index.xml" rel="noopener alternate" type="application/rss&#43;xml"
    class="iconfont" title="rss" target="_blank">
    <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="30" height="30">
  <path d="M819.157333 1024C819.157333 574.592 449.408 204.8 0 204.8V0c561.706667 0 1024 462.293333 1024 1024h-204.842667zM140.416 743.04a140.8 140.8 0 0 1 140.501333 140.586667A140.928 140.928 0 0 1 140.074667 1024C62.72 1024 0 961.109333 0 883.626667s62.933333-140.544 140.416-140.586667zM678.784 1024h-199.04c0-263.210667-216.533333-479.786667-479.744-479.786667V345.173333c372.352 0 678.784 306.517333 678.784 678.826667z"></path>
</svg>

  </a>
   
</div>

<div class="copyright">

  <span class="copyright-year">
    &copy;
    2020
    <span class="heart">
      
      <i class="iconfont">
        <svg class="icon" viewBox="0 0 1025 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="14" height="14">
  <path d="M1000.1 247.9c-15.5-37.3-37.6-70.6-65.7-98.9-54.4-54.8-125.8-85-201-85-85.7 0-166 39-221.4 107.4C456.6 103 376.3 64 290.6 64c-75.1 0-146.5 30.4-201.1 85.6-28.2 28.5-50.4 61.9-65.8 99.3-16 38.8-24 79.9-23.6 122.2 0.7 91.7 40.1 177.2 108.1 234.8 3.1 2.6 6 5.1 8.9 7.8 14.9 13.4 58 52.8 112.6 102.7 93.5 85.5 209.9 191.9 257.5 234.2 7 6.1 15.8 9.5 24.9 9.5 9.2 0 18.1-3.4 24.9-9.5 34.5-30.7 105.8-95.9 181.4-165 74.2-67.8 150.9-138 195.8-178.2 69.5-57.9 109.6-144.4 109.9-237.3 0.1-42.5-8-83.6-24-122.2z"
   fill="#8a8a8a"></path>
</svg>

      </i>
    </span><span class="author">
        xiaonilee
        
      </span></span>

  
  

  
</div>

    </footer>

    <div class="back-to-top" id="back-to-top">
      <i class="iconfont">
        
        <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="35" height="35">
  <path d="M510.866688 227.694839 95.449397 629.218702l235.761562 0-2.057869 328.796468 362.40389 0L691.55698 628.188232l241.942331-3.089361L510.866688 227.694839zM63.840492 63.962777l894.052392 0 0 131.813095L63.840492 195.775872 63.840492 63.962777 63.840492 63.962777zM63.840492 63.962777"></path>
</svg>

      </i>
    </div>
  </div>
  
<script type="text/javascript" src="/lib/jquery/jquery-3.2.1.min.js"></script>
  <script type="text/javascript" src="/lib/slideout/slideout-1.0.1.min.js"></script>




<script type="text/javascript" src="/js/main.638251f4230630f0335d8c6748e53a96f94b72670920b60c09a56fdc8bece214.js" integrity="sha256-Y4JR9CMGMPAzXYxnSOU6lvlLcmcJILYMCaVv3Ivs4hQ=" crossorigin="anonymous"></script>



  <script type="text/javascript">
    window.MathJax = {
      showProcessingMessages: false,
      messageStyle: 'none'
    };
  </script>
  <script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-MML-AM_CHTML' async></script>









  
    <script type="text/javascript" src="/js/load-photoswipe.js"></script>
    <script type="text/javascript" src="/lib/photoswipe/photoswipe.min.js"></script>
    <script type="text/javascript" src="/lib/photoswipe/photoswipe-ui-default.min.js"></script>
  















</body>
</html>
